<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>E. coli K12 analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rare mutation detection</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="model.html">Model</a>
</li>
<li>
  <a href="ecoli.html">E coli BL21 results</a>
</li>
<li>
  <a href="ecoli_K12.html">E coli K12 results</a>
</li>
<li>
  <a href="compare_MGI_Illumina.html">Illumina vs. MGI</a>
</li>
<li>
  <a href="insilico_mixtures.html">in silico mixtures</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">E. coli K12 analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rare-mutation-detection/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontreee722dc0ffb9f2e47d0f6535f438d18be5b4026f4targetblanke722dc0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mcmero/rare-mutation-detection/tree/e722dc0ffb9f2e47d0f6535f438d18be5b4026f4" target="_blank">e722dc0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontreee722dc0ffb9f2e47d0f6535f438d18be5b4026f4targetblanke722dc0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mcmero/rare-mutation-detection/tree/e722dc0ffb9f2e47d0f6535f438d18be5b4026f4" target="_blank">e722dc0</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    scripts/

Untracked files:
    Untracked:  ._.DS_Store
    Untracked:  DOCNAME
    Untracked:  analysis/._.DS_Store
    Untracked:  analysis/cache/
    Untracked:  analysis/calc_nanoseq_metrics.Rmd
    Untracked:  data/._metrics.rds
    Untracked:  data/ecoli/
    Untracked:  data/metadata/
    Untracked:  data/metrics_efficiency_nossc.rds
    Untracked:  data/mixtures
    Untracked:  data/ref/
    Untracked:  prototype_code/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ecoli_K12.Rmd</code>) and HTML (<code>docs/ecoli_K12.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/e722dc0ffb9f2e47d0f6535f438d18be5b4026f4/analysis/ecoli_K12.Rmd" target="_blank">e722dc0</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-31
</td>
<td>
Fix typo, minor code update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/b5242382e07051bba0914bd6e3906623d0644a64/docs/ecoli_K12.html" target="_blank">b524238</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/afd79e57f54101b6546d69af08f398f1d12dc29e/analysis/ecoli_K12.Rmd" target="_blank">afd79e5</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-26
</td>
<td>
Added revised model, in silico mixtures redone with 1 supporting read, added input cell estimates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/491e97dd8a2e378752abda548fdd0ac8ed9a4ea8/docs/ecoli_K12.html" target="_blank">491e97d</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/434e8b9da30508bd132082864b42679824a086c3/analysis/ecoli_K12.Rmd" target="_blank">434e8b9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-19
</td>
<td>
Added in silico mixtures to navigation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/ecoli_K12.html" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/aacf423a01dfd4c2f8e4d8d701f3fc0469efd94e/analysis/ecoli_K12.Rmd" target="_blank">aacf423</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
<td>
Add coverage &amp; variant results without requiring SSC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/4da22442742ce535db365adc322853bc3bb8635d/docs/ecoli_K12.html" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/71e857b316760c9cd2a477592928f57b4bd34448/analysis/ecoli_K12.Rmd" target="_blank">71e857b</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
<td>
Summary plot with all experimental factors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/ecoli_K12.html" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/48b6d2e1a856afc28844fa0da74f9cfca6b09f88/analysis/ecoli_K12.Rmd" target="_blank">48b6d2e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
<td>
Added statistical tests with xGen rep 1 outlier removed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/7c4f40362875160205ac501db16a9417a27dece7/docs/ecoli_K12.html" target="_blank">7c4f403</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/30f532f9f8f97914a237df24fbcae4518c26f5e8/analysis/ecoli_K12.Rmd" target="_blank">30f532f</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-25
</td>
<td>
Include all samples in variant upset plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/ecoli_K12.html" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/6f2c2bbbb905f8c6561f46c3b8a8a015551320d9/analysis/ecoli_K12.Rmd" target="_blank">6f2c2bb</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
<td>
Add family stats, boxplot fixes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/a2f0a4a07b138167d30f1fb7c80004ee4f59937a/docs/ecoli_K12.html" target="_blank">a2f0a4a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bffbb7e47a044808e3459c496c93f7423760ba2d/analysis/ecoli_K12.Rmd" target="_blank">bffbb7e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-08
</td>
<td>
Repeat variant analysis without filtering strand bias
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/c246dc2fe2999e98e6daf86e38b31414df507d7e/docs/ecoli_K12.html" target="_blank">c246dc2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/e10a166c4f7b635f4465fc3c4e39784d8e5769e2/analysis/ecoli_K12.Rmd" target="_blank">e10a166</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-07
</td>
<td>
Added variant call upset plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/ecoli_K12.html" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5dcf0e9543d6e2b4f69ee2b72a08251a05f16bf7/analysis/ecoli_K12.Rmd" target="_blank">5dcf0e9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
<td>
Added relationship plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/ecoli_K12.html" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/43c95e35d2723cbfd48f8e6f3341ccc24957197a/analysis/ecoli_K12.Rmd" target="_blank">43c95e3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
<td>
Fix figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/ecoli_K12.html" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/db75aa72c4cec2f9bc956c1f5541663700601c63/analysis/ecoli_K12.Rmd" target="_blank">db75aa7</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
<td>
Added statistical tests
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/ecoli_K12.html" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/1e5e6960d0123c3c6041a5c848ee0faba5d34ed3/analysis/ecoli_K12.Rmd" target="_blank">1e5e696</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
<td>
Added descriptions for metrics. General plot improvements.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/953b83ede1e12bcd5c1708f773f0afc6c1e4de40/docs/ecoli_K12.html" target="_blank">953b83e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/05412f6103fbf7b3291006856596fcff0f1ae6ea/docs/ecoli_K12.html" target="_blank">05412f6</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/ea0ad826784acfbb0f6583ecf9073563694a4453/analysis/ecoli_K12.Rmd" target="_blank">ea0ad82</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-27
</td>
<td>
Added singleton comparison + facet summary plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/51aba0e4ca32d4fd0c3f083259d22b0f7e4dd7c6/docs/ecoli_K12.html" target="_blank">51aba0e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a3895f7de10a1553a4334b1569d89e8d1cb11698/analysis/ecoli_K12.Rmd" target="_blank">a3895f7</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-25
</td>
<td>
Bug fix
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/ea4faf4104730982537c3e8f26d2ccbf7129c558/docs/ecoli_K12.html" target="_blank">ea4faf4</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5964f141b12189eeff85b8d0cc66ace5cd10b5c2/analysis/ecoli_K12.Rmd" target="_blank">5964f14</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-25
</td>
<td>
Added more comparison plots for ecoli K12 data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/e5b39add336bbf4e248b3dd449b8363c00373c76/docs/ecoli_K12.html" target="_blank">e5b39ad</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/1926d3dc0acbe289fdff366b0becac83d9b3bc30/analysis/ecoli_K12.Rmd" target="_blank">1926d3d</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-25
</td>
<td>
added K12 ecoli metrics
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="metrics-for-e.-coli-k12-data" class="section level1">
<h1>Metrics for E. coli K12 data</h1>
<div id="multiqc-reports" class="section level2">
<h2>MultiQC reports:</h2>
<ul>
<li><a href="AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report.html">AGRF_CAGRF22029764_HJK2GDSX3 MultiQC pre-duplex</a></li>
<li><a href="AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report_consensus.html">AGRF_CAGRF22029764_HJK2GDSX3 MultiQC duplex</a></li>
<li><a href="AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report_consensus_nossc.html">AGRF_CAGRF22029764_HJK2GDSX3 MultiQC duplex without SSCs</a></li>
</ul>
<pre class="r"><code>library(ggplot2)
library(data.table)
library(dplyr)
library(here)
library(tibble)
library(stringr)
library(Rsamtools)
library(GenomicRanges)
library(seqinr)
library(parallel)
library(readxl)
library(patchwork)
library(RColorBrewer)
library(UpSetR)
library(vcfR)</code></pre>
<pre class="r"><code>source(here(&#39;code/load_data.R&#39;))
source(here(&#39;code/plot.R&#39;))
source(here(&#39;code/efficiency_nanoseq_functions.R&#39;))</code></pre>
<pre class="r"><code># Ecoli genome max size
# genome_max &lt;- 4528118
genome_max &lt;- c(&#39;2e914854fabb46b9_1&#39; = 4661751,
                &#39;2e914854fabb46b9_2&#39; = 67365)
cores = 8</code></pre>
<pre class="r"><code>genomeFile &lt;- here(&#39;data/ref/Escherichia_coli_ATCC_10798.fasta&#39;)
rinfo_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/read_info&#39;)
markdup_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/mark_duplicates&#39;)
qualimap_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/qualimap&#39;)
qualimap_cons_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/consensus/qualimap&#39;)
qualimap_cons_nossc_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/consensus/qualimap_nossc&#39;)
metadata_file &lt;- here(&#39;data/metadata/NovaSeq data E coli.xlsx&#39;)
variant_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/variants&#39;)
variant_nossc_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/variants_nossc&#39;)</code></pre>
<pre class="r"><code>sample_names &lt;- list.files(rinfo_dir) %&gt;%
                str_split(&#39;\\.txt.gz&#39;) %&gt;%
                lapply(., dplyr::first) %&gt;%
                unlist() %&gt;%
                str_split(&#39;_&#39;) %&gt;%
                lapply(., head, 2) %&gt;%
                lapply(., paste, collapse=&#39;-&#39;) %&gt;%
                unlist()

# load variant data
var_df &lt;- load_variants(variant_dir, sample_names)
var_df_nossc &lt;- load_variants(variant_nossc_dir, sample_names[-9])

# load and fetch duplicate rate from MarkDuplicates output
mdup &lt;- load_markdup_data(markdup_dir, sample_names)

# get mean coverage for pre and post-consensus reads
qmap_cov &lt;- get_qmap_coverage(qualimap_dir, sample_names)
qmap_cons_cov &lt;- get_qmap_coverage(qualimap_cons_dir, sample_names)
qmap_cons_cov_nossc &lt;- get_qmap_coverage(qualimap_cons_nossc_dir, sample_names[-9])

# uncomment below to calculate metrics
# # calculate metrics for nanoseq
# rlen &lt;- 151; skips &lt;- 5
# metrics_nano &lt;- calc_metrics_new_rbs(rinfo_dir, pattern = &#39;Nano&#39;, cores = cores)
# 
# # calculate metrics for xGen
# rlen &lt;- 151; skips &lt;- 8
# metrics_xgen &lt;- calc_metrics_new_rbs(rinfo_dir, pattern = &#39;xGEN&#39;, cores = cores)
# 
# metrics &lt;- c(metrics_nano, metrics_xgen) %&gt;% bind_rows()
# metrics$duplicate_rate &lt;- as.numeric(mdup)
# metrics$duplex_coverage_ratio &lt;- qmap_cov$coverage / qmap_cons_cov$coverage
# metrics$duplex_coverage_ratio[qmap_cons_cov$coverage &lt; 1] &lt;- 0 # fix when &lt; 1 duplex cov
# metrics$sample &lt;- gsub(&#39;-HJK2GDSX3&#39;, &#39;&#39;, sample_names)

# cache metrics object
# saveRDS(metrics, file = here(&#39;data/metrics.rds&#39;))
metrics &lt;- readRDS(here(&#39;data/metrics.rds&#39;))

# load metadata
metadata &lt;- read_excel(metadata_file)
metadata$`sample name` &lt;- gsub(&#39;_&#39;, &#39;-&#39;, metadata$`sample name`)

# prepare for plotting
mm &lt;- data.frame(melt(metrics))
mm$protocol &lt;- &#39;NanoSeq&#39;
mm$protocol[grep(&#39;xGEN&#39;, mm$sample)] &lt;- &#39;xGen&#39;

mm &lt;- inner_join(mm, metadata, by=c(&#39;sample&#39; = &#39;sample name&#39;))
colnames(mm)[2] &lt;- &#39;metric&#39;
mm$nuclease &lt;- paste(mm$`Mung bean unit`, mm$`S1 unit`, sep=&#39;+&#39;)</code></pre>
</div>
<div id="metric-comparison-plots" class="section level2">
<h2>Metric comparison plots</h2>
<div id="duplicate-rate" class="section level3">
<h3>Duplicate rate</h3>
<p>Fraction of duplicate reads calculated by <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-">Picard’s MarkDuplicates</a>. This is based on barcode-aware aligned duplicates mapping to the same 5’ positions for both read pairs. The <a href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis pipeline</a> states the optimal empirical duplicate rate is 75-76% (marked in the plot).</p>
<pre class="r"><code>metric &lt;- &#39;duplicate_rate&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.75, 0.76), alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/duplicate_rate-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-duplicate_rate-1">
Past versions of duplicate_rate-1.png
</button>
</p>
<div id="fig-duplicate_rate-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/duplicate_rate-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/duplicate_rate-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/duplicate_rate-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fraction-of-singleton-reads" class="section level3">
<h3>Fraction of singleton reads</h3>
<p>Shows the number of single-read families divided by the total number of reads. As suggested by <a href="https://doi.org/10.1186/s13059-016-1039-4">Stoler et al. 2016</a>, this metric can server as a proxy for error rate, as (uncorrected) barcode mismatches will manifest as single-read families. The lower the fraction of singletons, the better.</p>
<pre class="r"><code>metric &lt;- &#39;frac_singletons&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/singletons-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-singletons-1">
Past versions of singletons-1.png
</button>
</p>
<div id="fig-singletons-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/singletons-1.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/singletons-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/singletons-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/singletons-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="drop-out-rate" class="section level3">
<h3>Drop-out rate</h3>
<p>This is the same calculation as F-EFF in the <a href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis pipeline</a>:</p>
<p>“This shows the fraction of read bundles missing one of the two original strands beyond what would be expected under random sampling (assuming a binomial process). Good values are between 0.10-0.30, and larger values are likely due to DNA damage such as modified bases or internal nicks that prevent amplification of one of the two strands. Larger values do not impact the quality of the results, just reduce the efficiency of the protocol.”</p>
<p>This is similar to the singleton fraction, but taking into account loss of pairs due to sampling. The optimal range is shown by the lines.</p>
<pre class="r"><code>metric &lt;- &#39;drop_out_rate&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.1, 0.3), alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/drop_out_rate-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-drop_out_rate-1">
Past versions of drop_out_rate-1.png
</button>
</p>
<div id="fig-drop_out_rate-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/drop_out_rate-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/drop_out_rate-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/drop_out_rate-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="efficiency" class="section level3">
<h3>Efficiency</h3>
<p>Efficiency is the number of duplex bases divided by the number of sequenced bases. According the <a href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis pipeline</a>, this value is maximised at ~0.07 when duplicate rates and strand drop-outs are optimal.</p>
<pre class="r"><code>metric &lt;- &#39;efficiency&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.07), alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/efficiency-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-efficiency-1">
Past versions of efficiency-1.png
</button>
</p>
<div id="fig-efficiency-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/efficiency-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/efficiency-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/efficiency-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gc-deviation" class="section level3">
<h3>GC deviation</h3>
<p>GC deviation is the absolute difference between <code>GC_BOTH</code> and <code>GC_SINGLE</code> calculated by the <a href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis pipeline</a>. The lower this deviation, the better.</p>
<p>“GC_BOTH and GC_SINGLE: the GC content of RBs with both strands and with just one strand. The two values should be similar between them and similar to the genome average. If there are large deviations that is possibly due to biases during PCR amplification. If GC_BOTH is substantially larger than GC_SINGLE, DNA denaturation before dilution may have taken place.”</p>
<pre class="r"><code>metric &lt;- &#39;gc_deviation&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/gc_deviation-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gc_deviation-1">
Past versions of gc_deviation-1.png
</button>
</p>
<div id="fig-gc_deviation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7c4f40362875160205ac501db16a9417a27dece7/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">7c4f403</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a2f0a4a07b138167d30f1fb7c80004ee4f59937a/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">a2f0a4a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/c246dc2fe2999e98e6daf86e38b31414df507d7e/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">c246dc2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/gc_deviation-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="duplex-coverage-ratio" class="section level3">
<h3>Duplex Coverage ratio</h3>
<p>The mean sequence (pre-duplex) coverage divided by mean duplex coverage. Indicates the yield of how much duplex coverage we get at each sample’s sequence coverage. <a href="https://doi.org/10.1038/s41586-021-03477-4">Abascal et al.</a> report that their yield was approximately 30x (marked on the plot).</p>
<pre class="r"><code>metric &lt;- &#39;duplex_coverage_ratio&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = 30, alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/duplex_coverage_ratio-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-duplex_coverage_ratio-1">
Past versions of duplex_coverage_ratio-1.png
</button>
</p>
<div id="fig-duplex_coverage_ratio-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/duplex_coverage_ratio-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/duplex_coverage_ratio-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/duplex_coverage_ratio-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="family-statistics" class="section level2">
<h2>Family statistics</h2>
<p>Comparison of family pair sizes between samples (these are calculated from total reads of paired AB and BA families).</p>
<pre class="r"><code>ggplot(mm[mm$metric %like% &#39;family&#39;, ], aes(value, sample, colour = metric)) +
        geom_point() +
        coord_trans(x=&#39;log2&#39;) +
        scale_x_continuous(breaks=seq(0, 94, 8)) +
        theme(axis.text.x = element_text(size=5)) +
        theme_bw() +
        ggtitle(&#39;Family pair sizes&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/family_sizes-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-family_sizes-1">
Past versions of family_sizes-1.png
</button>
</p>
<div id="fig-family_sizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/family_sizes-1.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The following plot shows:</p>
<ul>
<li>families_gt1: number of family pairs where at least one family (AB or BA) has &gt; 1 reads.</li>
<li>paired_families: number of family pairs where both families (AB and BA) have &gt; 0 reads.</li>
<li>paired_and_gt1: number of family pairs where both families (AB and BA) have &gt; 1 reads.</li>
</ul>
<pre class="r"><code>ggplot(mm[mm$metric %like% &#39;pair|gt1&#39;, ], aes(value, sample, fill = metric)) +
        geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
        theme_bw() +
        ggtitle(&#39;Family statistics&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/family_stats-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-family_stats-1">
Past versions of family_stats-1.png
</button>
</p>
<div id="fig-family_stats-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/family_stats-1.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="compare-metrics-side-by-side" class="section level2">
<h2>Compare metrics side-by-side</h2>
<p>Compare protocols and nucleases directly.</p>
<pre class="r"><code>gg_prot &lt;- list(geom_boxplot(outlier.shape = NA),
               geom_jitter(width = 0.1, size = 2, aes(colour = nuclease, shape = nuclease)),
               theme_bw(),
               theme(legend.position = &#39;left&#39;))
gg_nuc &lt;- list(geom_boxplot(outlier.shape = NA),
               geom_jitter(width = 0.1, size = 2, aes(colour = protocol, shape = protocol)),
               theme_bw(),
               theme(legend.position = &#39;right&#39;))

# duplicate rate
metric &lt;- &#39;duplicate_rate&#39;
p1 &lt;- ggplot(mm[mm$metric == metric,], aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = 0.81) +
        ggtitle(&#39;Duplicate rate (line = optimal)&#39;)

p2 &lt;- ggplot(mm[mm$metric == metric,], aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = 0.81) +
        ggtitle(&#39;Duplicate rate (line = optimal)&#39;)
show(p1 + p2)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/metrics_comp-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-metrics_comp-1">
Past versions of metrics_comp-1.png
</button>
</p>
<div id="fig-metrics_comp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/metrics_comp-1.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/metrics_comp-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/metrics_comp-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/metrics_comp-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># singletons
metric &lt;- &#39;frac_singletons&#39;
p1 &lt;- ggplot(mm[mm$metric == metric,], aes(protocol, value)) +
        gg_prot +
        ggtitle(&#39;Fraction of singleton reads&#39;)

p2 &lt;- ggplot(mm[mm$metric == metric,], aes(nuclease, value)) +
        gg_nuc + 
        ggtitle(&#39;Fraction of singleton reads&#39;)
show(p1 + p2)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/metrics_comp-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-metrics_comp-2">
Past versions of metrics_comp-2.png
</button>
</p>
<div id="fig-metrics_comp-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/metrics_comp-2.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/metrics_comp-2.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/metrics_comp-2.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/metrics_comp-2.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/metrics_comp-2.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># drop-out rate
metric &lt;- &#39;drop_out_rate&#39;
p1 &lt;- ggplot(mm[mm$metric == metric,], aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = c(0.1, 0.3)) + ylim(c(0,1)) +
        ggtitle(&#39;Drop-out fraction\n(lines = optimal range)&#39;)

p2 &lt;- ggplot(mm[mm$metric == metric,], aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = c(0.1, 0.3)) + ylim(c(0,1)) +
        ggtitle(&#39;Drop-out fraction\n(lines = optimal range)&#39;)
show(p1 + p2)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/metrics_comp-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-metrics_comp-3">
Past versions of metrics_comp-3.png
</button>
</p>
<div id="fig-metrics_comp-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/metrics_comp-3.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/metrics_comp-3.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/metrics_comp-3.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/metrics_comp-3.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># efficiency
metric &lt;- &#39;efficiency&#39;
p1 &lt;- ggplot(mm[mm$metric == metric,], aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = 0.07) +
        ggtitle(&#39;Efficiency\n(line = optimal)&#39;)

p2 &lt;- ggplot(mm[mm$metric == metric,], aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = 0.07) +
        ggtitle(&#39;Efficiency\n(line = optomal)&#39;)
show(p1 + p2)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/metrics_comp-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-metrics_comp-4">
Past versions of metrics_comp-4.png
</button>
</p>
<div id="fig-metrics_comp-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/metrics_comp-4.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/metrics_comp-4.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/metrics_comp-4.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/metrics_comp-4.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/metrics_comp-4.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># GC deviation
metric &lt;- &#39;gc_deviation&#39;
p1 &lt;- ggplot(mm[mm$metric == metric,], aes(protocol, value)) +
        gg_prot +
        ggtitle(&#39;GC deviation\n(both strands vs. one)&#39;)

p2 &lt;- ggplot(mm[mm$metric == metric,], aes(nuclease, value)) +
        gg_nuc + 
        ggtitle(&#39;GC deviation\n(both strands vs. one)&#39;)
show(p1 + p2)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/metrics_comp-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-metrics_comp-5">
Past versions of metrics_comp-5.png
</button>
</p>
<div id="fig-metrics_comp-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7c4f40362875160205ac501db16a9417a27dece7/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">7c4f403</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a2f0a4a07b138167d30f1fb7c80004ee4f59937a/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">a2f0a4a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/c246dc2fe2999e98e6daf86e38b31414df507d7e/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">c246dc2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/metrics_comp-5.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># duplex coverage ratio
metric &lt;- &#39;duplex_coverage_ratio&#39;
p1 &lt;- ggplot(mm[mm$metric == metric,], aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = 30) +
        ggtitle(&#39;Duplex coverage ratio\n(total cov / duplex cov)&#39;)

p2 &lt;- ggplot(mm[mm$metric == metric,], aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = 30) +
        ggtitle(&#39;Duplex coverage ratio\n(total cov / duplex cov)&#39;)
show(p1 + p2)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/metrics_comp-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-metrics_comp-6">
Past versions of metrics_comp-6.png
</button>
</p>
<div id="fig-metrics_comp-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/metrics_comp-6.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/metrics_comp-6.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/metrics_comp-6.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/metrics_comp-6.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="facet-summary-plots" class="section level2">
<h2>Facet summary plots</h2>
<p>Facet boxplots by nuclease and protocol to show overall results.</p>
<pre class="r"><code>ggplot(mm, aes(protocol, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = &#39;free&#39;) +
    ggtitle(&#39;by protocol&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/facet_summary-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-facet_summary-1">
Past versions of facet_summary-1.png
</button>
</p>
<div id="fig-facet_summary-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/facet_summary-1.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/facet_summary-1.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/facet_summary-1.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(mm, aes(nuclease, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = &#39;free&#39;) +
    ggtitle(&#39;by nuclease&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/facet_summary-2.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-facet_summary-2">
Past versions of facet_summary-2.png
</button>
</p>
<div id="fig-facet_summary-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/facet_summary-2.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/facet_summary-2.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/facet_summary-2.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Plots again removing the outlier xGEN rep 1.</p>
<pre class="r"><code>mmo &lt;- mm[mm$sample != &#39;xGEN-xGENRep1&#39;,]
mmo$replicate &lt;- str_split(mmo$sample, &#39;Rep&#39;) %&gt;% lapply(., dplyr::last) %&gt;% unlist() %&gt;% as.numeric()
mmo$sample &lt;- str_split(mmo$sample, &#39;Rep&#39;) %&gt;% lapply(., dplyr::first) %&gt;% unlist()

ggplot(mmo, aes(protocol, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = &#39;free&#39;) +
    ggtitle(&#39;by protocol&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/facet_summary_remove_outlier-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-facet_summary_remove_outlier-1">
Past versions of facet_summary_remove_outlier-1.png
</button>
</p>
<div id="fig-facet_summary_remove_outlier-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/facet_summary_remove_outlier-1.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/facet_summary_remove_outlier-1.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/facet_summary_remove_outlier-1.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(mmo, aes(nuclease, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = &#39;free&#39;) +
    ggtitle(&#39;by nuclease&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/facet_summary_remove_outlier-2.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-facet_summary_remove_outlier-2">
Past versions of facet_summary_remove_outlier-2.png
</button>
</p>
<div id="fig-facet_summary_remove_outlier-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/facet_summary_remove_outlier-2.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/facet_summary_remove_outlier-2.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/facet_summary_remove_outlier-2.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Summary plot including separated by all experimental factors.</p>
<pre class="r"><code>ggplot(mmo, aes(sample, value, colour = protocol, shape = nuclease)) + 
    geom_point() +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90)) +
    facet_wrap(~metric, scales = &#39;free&#39;) +
    scale_colour_brewer(palette = &#39;Dark2&#39;) +
    ggtitle(&#39;by protocol&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/plot_by_sample-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_by_sample-1">
Past versions of plot_by_sample-1.png
</button>
</p>
<div id="fig-plot_by_sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/plot_by_sample-1.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/4da22442742ce535db365adc322853bc3bb8635d/docs/figure/ecoli_K12.Rmd/plot_by_sample-1.png" target="_blank">4da2244</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="statistical-test-results-by-protocol" class="section level2">
<h2>Statistical test results by protocol</h2>
<p>For each metric, take the average of each replicate and perform a two-sided, unpaired T-test between protocols.</p>
<pre class="r"><code>stats &lt;- NULL
metric_names &lt;- unique(mm$metric) %&gt;% as.character()
for(metric_name in metric_names) {
    nano &lt;- mm[mm$metric == metric_name &amp; mm$protocol == &#39;NanoSeq&#39;,]
    xgen &lt;- mm[mm$metric == metric_name &amp; mm$protocol == &#39;xGen&#39;,]
    nano_vals &lt;- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals &lt;- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest &lt;- t.test(nano_vals, xgen_vals)
    stats &lt;- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant &lt;- stats$pvalue &lt; 0.05
print(stats)</code></pre>
<pre><code>                  metric       pvalue significant
1        frac_singletons 0.3509205995       FALSE
2             efficiency 0.0016455448        TRUE
3          drop_out_rate 0.0016563300        TRUE
4           gc_deviation 0.0195493871        TRUE
5            family_mean 0.2739786956       FALSE
6          family_median 0.3816330944       FALSE
7             family_max 0.4142481163       FALSE
8           families_gt1 0.6326687504       FALSE
9        paired_families 0.0006261990        TRUE
10        paired_and_gt1 0.0003304955        TRUE
11        duplicate_rate 0.9225366816       FALSE
12 duplex_coverage_ratio 0.3166974439       FALSE</code></pre>
<p>Rerun tests removing outlier (xGEN rep1). The results are similar.</p>
<pre class="r"><code>stats &lt;- NULL
for(metric_name in metric_names) {
    nano &lt;- mmo[mmo$metric == metric_name &amp; mmo$protocol == &#39;NanoSeq&#39;,]
    xgen &lt;- mmo[mmo$metric == metric_name &amp; mmo$protocol == &#39;xGen&#39;,]
    nano_vals &lt;- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals &lt;- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest &lt;- t.test(nano_vals, xgen_vals)
    stats &lt;- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant &lt;- stats$pvalue &lt; 0.05
print(stats)</code></pre>
<pre><code>                  metric      pvalue significant
1        frac_singletons 0.885668764       FALSE
2             efficiency 0.017624387        TRUE
3          drop_out_rate 0.010803300        TRUE
4           gc_deviation 0.002863331        TRUE
5            family_mean 0.408145893       FALSE
6          family_median 0.617803978       FALSE
7             family_max 0.517858472       FALSE
8           families_gt1 0.914651102       FALSE
9        paired_families 0.001878931        TRUE
10        paired_and_gt1 0.002903205        TRUE
11        duplicate_rate 0.469189762       FALSE
12 duplex_coverage_ratio 0.214507629       FALSE</code></pre>
</div>
<div id="two-way-anova-analysis" class="section level2">
<h2>Two-way ANOVA analysis</h2>
<p>We consider a two-way ANOVA, modelling the protocol, Mung Bean Unit and S1 Unit variables, as well as the interaction effect between the units and the protocol.</p>
<pre class="r"><code>stats &lt;- NULL
metric_names &lt;- unique(mm$metric) %&gt;% as.character()
for(metric_name in metric_names) {
    x &lt;- mm[mm$metric == metric_name,]
    x$MungBeanUnit &lt;- as.factor(x$`Mung bean unit`)
    x$S1Unit &lt;- as.factor(x$`S1 unit`)
    x &lt;- x[,c(&#39;MungBeanUnit&#39;, &#39;S1Unit&#39;, &#39;protocol&#39;, &#39;nuclease&#39;, &#39;value&#39;)]
    x_aov &lt;- aov(value ~ MungBeanUnit * protocol + S1Unit * protocol, data = x) %&gt;% summary() %&gt;% dplyr::first()
    stats &lt;- rbind(stats,
                   data.frame(metric = metric_name,
                              variable = rownames(x_aov)[1:5],
                              pvalue = x_aov[[&#39;Pr(&gt;F)&#39;]][1:5]))
}
stats$significant &lt;- stats$pvalue &lt; 0.05
print(stats)</code></pre>
<pre><code>                  metric              variable       pvalue significant
1        frac_singletons MungBeanUnit          0.3179447536       FALSE
2        frac_singletons protocol              0.9702278541       FALSE
3        frac_singletons S1Unit                0.8553539457       FALSE
4        frac_singletons MungBeanUnit:protocol 0.9858376372       FALSE
5        frac_singletons protocol:S1Unit       0.8540792709       FALSE
6             efficiency MungBeanUnit          0.6743588660       FALSE
7             efficiency protocol              0.0033776091        TRUE
8             efficiency S1Unit                0.4674622674       FALSE
9             efficiency MungBeanUnit:protocol 0.8509682597       FALSE
10            efficiency protocol:S1Unit       0.2278366157       FALSE
11         drop_out_rate MungBeanUnit          0.4118682293       FALSE
12         drop_out_rate protocol              0.0002566346        TRUE
13         drop_out_rate S1Unit                0.0904262242       FALSE
14         drop_out_rate MungBeanUnit:protocol 0.8387881526       FALSE
15         drop_out_rate protocol:S1Unit       0.3182162279       FALSE
16          gc_deviation MungBeanUnit          0.5321833114       FALSE
17          gc_deviation protocol              0.0087584727        TRUE
18          gc_deviation S1Unit                0.7094314628       FALSE
19          gc_deviation MungBeanUnit:protocol 0.4019325303       FALSE
20          gc_deviation protocol:S1Unit       0.9145680596       FALSE
21           family_mean MungBeanUnit          0.3721341384       FALSE
22           family_mean protocol              0.2541550650       FALSE
23           family_mean S1Unit                0.2914711924       FALSE
24           family_mean MungBeanUnit:protocol 0.2723545197       FALSE
25           family_mean protocol:S1Unit       0.1501250836       FALSE
26         family_median MungBeanUnit          0.6347857947       FALSE
27         family_median protocol              0.4810155294       FALSE
28         family_median S1Unit                0.3250056081       FALSE
29         family_median MungBeanUnit:protocol 0.4997581250       FALSE
30         family_median protocol:S1Unit       0.3250056081       FALSE
31            family_max MungBeanUnit          0.3849414781       FALSE
32            family_max protocol              0.5270991535       FALSE
33            family_max S1Unit                0.1424841525       FALSE
34            family_max MungBeanUnit:protocol 0.9819905871       FALSE
35            family_max protocol:S1Unit       0.0605658200       FALSE
36          families_gt1 MungBeanUnit          0.1793349230       FALSE
37          families_gt1 protocol              0.9876271265       FALSE
38          families_gt1 S1Unit                0.6881757434       FALSE
39          families_gt1 MungBeanUnit:protocol 0.6170001142       FALSE
40          families_gt1 protocol:S1Unit       0.0301632029        TRUE
41       paired_families MungBeanUnit          0.3217761699       FALSE
42       paired_families protocol              0.0002319573        TRUE
43       paired_families S1Unit                0.1990511038       FALSE
44       paired_families MungBeanUnit:protocol 0.9226482412       FALSE
45       paired_families protocol:S1Unit       0.8092463937       FALSE
46        paired_and_gt1 MungBeanUnit          0.6527042939       FALSE
47        paired_and_gt1 protocol              0.0007082361        TRUE
48        paired_and_gt1 S1Unit                0.8872835419       FALSE
49        paired_and_gt1 MungBeanUnit:protocol 0.5304733713       FALSE
50        paired_and_gt1 protocol:S1Unit       0.2688705912       FALSE
51        duplicate_rate MungBeanUnit          0.3209743574       FALSE
52        duplicate_rate protocol              0.6617113407       FALSE
53        duplicate_rate S1Unit                0.4855983121       FALSE
54        duplicate_rate MungBeanUnit:protocol 0.8160371321       FALSE
55        duplicate_rate protocol:S1Unit       0.5516726318       FALSE
56 duplex_coverage_ratio MungBeanUnit          0.0515969116       FALSE
57 duplex_coverage_ratio protocol              0.0059993368        TRUE
58 duplex_coverage_ratio S1Unit                0.4875101164       FALSE
59 duplex_coverage_ratio MungBeanUnit:protocol 0.0621558351       FALSE
60 duplex_coverage_ratio protocol:S1Unit       0.0041291841        TRUE</code></pre>
<p>We remove the outlier xGEN rep 1 and test again.</p>
<pre class="r"><code>stats &lt;- NULL
metric_names &lt;- unique(mmo$metric) %&gt;% as.character()
for(metric_name in metric_names) {
    x &lt;- mmo[mmo$metric == metric_name,]
    x$MungBeanUnit &lt;- as.factor(x$`Mung bean unit`)
    x$S1Unit &lt;- as.factor(x$`S1 unit`)
    x &lt;- x[,c(&#39;MungBeanUnit&#39;, &#39;S1Unit&#39;, &#39;protocol&#39;, &#39;nuclease&#39;, &#39;value&#39;)]
    x_aov &lt;- aov(value ~ MungBeanUnit * protocol + S1Unit * protocol, data = x) %&gt;% summary() %&gt;% dplyr::first()
    stats &lt;- rbind(stats,
                   data.frame(metric = metric_name,
                              variable = rownames(x_aov)[1:5],
                              pvalue = x_aov[[&#39;Pr(&gt;F)&#39;]][1:5]))
}
stats$significant &lt;- stats$pvalue &lt; 0.05
print(stats)</code></pre>
<pre><code>                  metric              variable       pvalue significant
1        frac_singletons MungBeanUnit          3.747242e-01       FALSE
2        frac_singletons protocol              6.061218e-01       FALSE
3        frac_singletons S1Unit                2.820185e-02        TRUE
4        frac_singletons MungBeanUnit:protocol 1.145001e-01       FALSE
5        frac_singletons protocol:S1Unit       2.714409e-02        TRUE
6             efficiency MungBeanUnit          2.943575e-02        TRUE
7             efficiency protocol              8.567087e-07        TRUE
8             efficiency S1Unit                4.375525e-02        TRUE
9             efficiency MungBeanUnit:protocol 2.583454e-01       FALSE
10            efficiency protocol:S1Unit       3.175014e-03        TRUE
11         drop_out_rate MungBeanUnit          4.996962e-04        TRUE
12         drop_out_rate protocol              2.459532e-09        TRUE
13         drop_out_rate S1Unit                2.501322e-05        TRUE
14         drop_out_rate MungBeanUnit:protocol 9.115253e-02       FALSE
15         drop_out_rate protocol:S1Unit       1.679681e-03        TRUE
16          gc_deviation MungBeanUnit          5.973605e-02       FALSE
17          gc_deviation protocol              1.140481e-04        TRUE
18          gc_deviation S1Unit                4.637927e-01       FALSE
19          gc_deviation MungBeanUnit:protocol 5.772604e-02       FALSE
20          gc_deviation protocol:S1Unit       8.311211e-01       FALSE
21           family_mean MungBeanUnit          4.646379e-01       FALSE
22           family_mean protocol              8.429941e-02       FALSE
23           family_mean S1Unit                1.077790e-01       FALSE
24           family_mean MungBeanUnit:protocol 6.294463e-02       FALSE
25           family_mean protocol:S1Unit       3.389191e-02        TRUE
26         family_median MungBeanUnit          4.629868e-01       FALSE
27         family_median protocol              3.164774e-01       FALSE
28         family_median S1Unit                1.678507e-01       FALSE
29         family_median MungBeanUnit:protocol 2.615312e-01       FALSE
30         family_median protocol:S1Unit       1.678507e-01       FALSE
31            family_max MungBeanUnit          8.985047e-01       FALSE
32            family_max protocol              4.901973e-01       FALSE
33            family_max S1Unit                1.144701e-01       FALSE
34            family_max MungBeanUnit:protocol 9.783491e-01       FALSE
35            family_max protocol:S1Unit       4.537604e-02        TRUE
36          families_gt1 MungBeanUnit          4.921202e-01       FALSE
37          families_gt1 protocol              9.709211e-01       FALSE
38          families_gt1 S1Unit                3.554965e-01       FALSE
39          families_gt1 MungBeanUnit:protocol 1.121914e-01       FALSE
40          families_gt1 protocol:S1Unit       2.152397e-04        TRUE
41       paired_families MungBeanUnit          1.158679e-02        TRUE
42       paired_families protocol              2.650844e-08        TRUE
43       paired_families S1Unit                1.825979e-03        TRUE
44       paired_families MungBeanUnit:protocol 4.727076e-01       FALSE
45       paired_families protocol:S1Unit       4.522724e-01       FALSE
46        paired_and_gt1 MungBeanUnit          3.715875e-01       FALSE
47        paired_and_gt1 protocol              4.877137e-07        TRUE
48        paired_and_gt1 S1Unit                7.110938e-01       FALSE
49        paired_and_gt1 MungBeanUnit:protocol 4.055773e-02        TRUE
50        paired_and_gt1 protocol:S1Unit       1.317773e-02        TRUE
51        duplicate_rate MungBeanUnit          2.028567e-01       FALSE
52        duplicate_rate protocol              9.553308e-02       FALSE
53        duplicate_rate S1Unit                1.519542e-02        TRUE
54        duplicate_rate MungBeanUnit:protocol 7.348826e-02       FALSE
55        duplicate_rate protocol:S1Unit       3.151836e-02        TRUE
56 duplex_coverage_ratio MungBeanUnit          2.480884e-01       FALSE
57 duplex_coverage_ratio protocol              6.668694e-04        TRUE
58 duplex_coverage_ratio S1Unit                3.194763e-01       FALSE
59 duplex_coverage_ratio MungBeanUnit:protocol 1.030421e-02        TRUE
60 duplex_coverage_ratio protocol:S1Unit       4.306854e-04        TRUE</code></pre>
</div>
<div id="relationships-between-variables" class="section level2">
<h2>Relationships between variables</h2>
<pre class="r"><code>mm$replicate &lt;- str_split(mm$sample, &#39;Rep&#39;) %&gt;% lapply(., dplyr::last) %&gt;% unlist() %&gt;% as.numeric()
mm$sample &lt;- str_split(mm$sample, &#39;Rep&#39;) %&gt;% lapply(., dplyr::first) %&gt;% unlist()

mm &lt;- mm[,c(&#39;sample&#39;, &#39;metric&#39;, &#39;value&#39;, &#39;protocol&#39;, &#39;nuclease&#39;, &#39;replicate&#39;)]
dm &lt;- reshape2::dcast(mm, sample + protocol + nuclease + replicate ~ metric)

cols &lt;- c(brewer.pal(5, &#39;Greens&#39;)[2:5],
          brewer.pal(6, &#39;Blues&#39;)[2:6])
names(cols) &lt;- as.factor(dm$sample) %&gt;% levels()

ggplot(dm, aes(frac_singletons, drop_out_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle(&#39;Singletons vs. drop-out rate&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/relationships-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relationships-1">
Past versions of relationships-1.png
</button>
</p>
<div id="fig-relationships-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/relationships-1.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(dm, aes(efficiency, duplicate_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle(&#39;Efficiency vs. duplicate rate&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/relationships-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relationships-2">
Past versions of relationships-2.png
</button>
</p>
<div id="fig-relationships-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/relationships-2.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(dm, aes(efficiency, drop_out_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle(&#39;Efficiency vs. drop-out rate&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/relationships-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relationships-3">
Past versions of relationships-3.png
</button>
</p>
<div id="fig-relationships-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/relationships-3.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(dm, aes(efficiency, duplex_coverage_ratio, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle(&#39;Efficiency vs. duplex coverage ratio&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/relationships-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relationships-4">
Past versions of relationships-4.png
</button>
</p>
<div id="fig-relationships-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/relationships-4.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(dm, aes(duplicate_rate, duplex_coverage_ratio, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle(&#39;Duplicate rate vs. duplex coverage ratio&#39;)</code></pre>
<p><img src="figure/ecoli_K12.Rmd/relationships-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-relationships-5">
Past versions of relationships-5.png
</button>
</p>
<div id="fig-relationships-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/relationships-5.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="variant-calls" class="section level2">
<h2>Variant calls</h2>
<p>Upset plot showing duplex variant calls. Variants were called in areas with at least 4x coverage with at least 2 supporting reads and a VAF of <span class="math inline">\(\geq2\)</span>.</p>
<pre class="r"><code>ulist &lt;- NULL
for(sample in sample_names) {
    ids &lt;- var_df[var_df$sample %in% sample,]$id
    if (length(ids) &gt; 0) {
        ulist[[sample]] &lt;- ids
    }
}

upset(fromList(ulist), order.by=&#39;freq&#39;, nsets=length(sample_names))</code></pre>
<p><img src="figure/ecoli_K12.Rmd/variants-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-variants-1">
Past versions of variants-1.png
</button>
</p>
<div id="fig-variants-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7c4f40362875160205ac501db16a9417a27dece7/docs/figure/ecoli_K12.Rmd/variants-1.png" target="_blank">7c4f403</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a2f0a4a07b138167d30f1fb7c80004ee4f59937a/docs/figure/ecoli_K12.Rmd/variants-1.png" target="_blank">a2f0a4a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/c246dc2fe2999e98e6daf86e38b31414df507d7e/docs/figure/ecoli_K12.Rmd/variants-1.png" target="_blank">c246dc2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="duplex-coverage-without-requiring-ssc" class="section level2">
<h2>Duplex coverage without requiring SSC</h2>
<p>The pipeline was run only requiring a single read on each strand. Here we plot the difference in mean coverage. As we would expect, skipping SSC step increases duplex coverage. For some samples with disproportionately higher single-read families (NanoMB-S1), this increases duplex coverage significantly more.</p>
<pre class="r"><code>ccov &lt;- inner_join(qmap_cons_cov,
                   qmap_cons_cov_nossc,
                   by = &#39;Sample&#39;,
                   suffix = c(&#39;_ssc&#39;, &#39;_nossc&#39;)) %&gt;%
          inner_join(., qmap_cov, by = &#39;Sample&#39;)
ccov$sample &lt;- str_split(ccov$Sample, &#39;Rep&#39;) %&gt;% lapply(., dplyr::first) %&gt;% unlist()
ccov$duplex_cov_ratio &lt;- ccov$coverage / ccov$coverage_ssc
ccov$duplex_cov_ratio_noscc &lt;- ccov$coverage / ccov$coverage_nossc

p1 &lt;- ggplot(ccov, aes(coverage_ssc, coverage_nossc, colour = sample)) +
  geom_point() +
  theme_bw() +
  xlim(0, 550) +
  ylim(0, 550) +
  xlab(&#39;with SSC&#39;) +
  ylab(&#39;without SSC&#39;) +
  geom_abline(slope = 1) +
  theme(legend.position = &#39;left&#39;) +
  scale_colour_brewer(palette = &#39;Dark2&#39;) +
  ggtitle(&#39;Mean duplex coverage&#39;)

p2 &lt;- ggplot(ccov, aes(duplex_cov_ratio, duplex_cov_ratio_noscc, colour = sample)) +
  geom_point() +
  theme_bw() +
  xlim(0, 100) +
  ylim(0, 100) +
  xlab(&#39;with SSC&#39;) +
  ylab(&#39;without SSC&#39;) +
  geom_abline(slope = 1) +
  theme(legend.position = &#39;right&#39;) +
  scale_colour_brewer(palette = &#39;Dark2&#39;) +
  ggtitle(&#39;Duplex coverage ratio&#39;)

p1 + p2</code></pre>
<p><img src="figure/ecoli_K12.Rmd/coverage_without_ssc-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-coverage_without_ssc-1">
Past versions of coverage_without_ssc-1.png
</button>
</p>
<div id="fig-coverage_without_ssc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/coverage_without_ssc-1.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="variant-calls-without-ssc" class="section level2">
<h2>Variant calls without SSC</h2>
<p>Here we show the variant calls from the duplex sequences without SSC in the same Upset plot format.</p>
<pre class="r"><code>ulist &lt;- NULL
for(sample in sample_names) {
    ids &lt;- var_df_nossc[var_df_nossc$sample %in% sample,]$id
    if (length(ids) &gt; 0) {
        ulist[[sample]] &lt;- ids
    }
}

upset(fromList(ulist), order.by=&#39;freq&#39;, nsets=length(sample_names))</code></pre>
<p><img src="figure/ecoli_K12.Rmd/variants_nossc-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-variants_nossc-1">
Past versions of variants_nossc-1.png
</button>
</p>
<div id="fig-variants_nossc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/faf9130e9a2175627a5e566185d9557a491a8a5e/docs/figure/ecoli_K12.Rmd/variants_nossc-1.png" target="_blank">faf9130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="input-cells" class="section level2">
<h2>Input cells</h2>
<p>Estimate the number of input cells using formula <span class="math inline">\(d / e / c = n\)</span> where d = mean duplex coverage, e = duplex efficiency, c = coverage per genome equivalent and n = number of cells.</p>
<pre class="r"><code>qcc &lt;- qmap_cons_cov
coverage_per_genome &lt;- 10
qcc$Sample &lt;- gsub(&#39;-HJK2GDSX3&#39;, &#39;&#39;, qcc$Sample)
mc &lt;- inner_join(metrics, qcc, by = c(&#39;sample&#39; = &#39;Sample&#39;))
mc$estimated_cells &lt;- mc$coverage / mc$efficiency / coverage_per_genome

ggplot(mc[!mc$sample %in% &#39;xGEN-xGENRep1&#39;,], aes(sample, estimated_cells)) +
    geom_bar(stat = &#39;identity&#39;) + 
    theme_minimal() +
    coord_flip()</code></pre>
<p><img src="figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/b5242382e07051bba0914bd6e3906623d0644a64/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">b524238</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cc380cc389100b3ecdc5c9cdf5616276fb4d1fd0/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">cc380cc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7c4f40362875160205ac501db16a9417a27dece7/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">7c4f403</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/fcb6578c014536c59339c72554c3a9f7844e0dde/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">fcb6578</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a2f0a4a07b138167d30f1fb7c80004ee4f59937a/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">a2f0a4a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/c246dc2fe2999e98e6daf86e38b31414df507d7e/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">c246dc2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/a8601010ee4d8b48191aeedf7be6bb95ae0b1234/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">a860101</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/81272b2356f01b49a7d0c32f4e62d9d5d8d17f9e/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">81272b2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f13e13a048f85249dd4b1e77fc34ba066446ea6e/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">f13e13a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/def213010fd1a838e8e423b2662c6d5cd49b7ac7/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">def2130</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/953b83ede1e12bcd5c1708f773f0afc6c1e4de40/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">953b83e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/05412f6103fbf7b3291006856596fcff0f1ae6ea/docs/figure/ecoli_K12.Rmd/unnamed-chunk-1-1.png" target="_blank">05412f6</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRblas.so
LAPACK: /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] vcfR_1.12.0          UpSetR_1.4.0         RColorBrewer_1.1-2  
 [4] patchwork_1.1.1      readxl_1.3.1         seqinr_4.2-8        
 [7] Rsamtools_2.6.0      Biostrings_2.58.0    XVector_0.30.0      
[10] GenomicRanges_1.42.0 GenomeInfoDb_1.26.7  IRanges_2.24.1      
[13] S4Vectors_0.28.1     BiocGenerics_0.36.1  stringr_1.4.0       
[16] tibble_3.1.5         here_1.0.1           dplyr_1.0.7         
[19] data.table_1.14.0    ggplot2_3.3.5        workflowr_1.6.2     

loaded via a namespace (and not attached):
 [1] sass_0.4.0             viridisLite_0.4.0      splines_4.0.5         
 [4] jsonlite_1.7.2         bslib_0.3.0            assertthat_0.2.1      
 [7] memuse_4.2-1           highr_0.9              GenomeInfoDbData_1.2.4
[10] cellranger_1.1.0       yaml_2.2.1             pillar_1.6.4          
[13] lattice_0.20-44        glue_1.4.2             digest_0.6.27         
[16] promises_1.2.0.1       colorspace_2.0-0       Matrix_1.3-2          
[19] htmltools_0.5.2        httpuv_1.6.3           plyr_1.8.6            
[22] pkgconfig_2.0.3        zlibbioc_1.36.0        purrr_0.3.4           
[25] scales_1.1.1           whisker_0.4            later_1.3.0           
[28] BiocParallel_1.24.1    git2r_0.28.0           mgcv_1.8-35           
[31] farver_2.1.0           generics_0.1.1         ellipsis_0.3.2        
[34] withr_2.4.2            magrittr_2.0.1         crayon_1.4.2          
[37] evaluate_0.14          fs_1.5.0               fansi_0.5.0           
[40] nlme_3.1-152           MASS_7.3-53.1          vegan_2.5-7           
[43] tools_4.0.5            lifecycle_1.0.1        munsell_0.5.0         
[46] cluster_2.1.2          ade4_1.7-18            compiler_4.0.5        
[49] jquerylib_0.1.4        rlang_0.4.12           grid_4.0.5            
[52] RCurl_1.98-1.3         labeling_0.4.2         bitops_1.0-7          
[55] rmarkdown_2.11         gtable_0.3.0           DBI_1.1.1             
[58] reshape2_1.4.4         R6_2.5.1               gridExtra_2.3         
[61] knitr_1.33             pinfsc50_1.2.0         fastmap_1.1.0         
[64] utf8_1.2.2             rprojroot_2.0.2        permute_0.9-5         
[67] ape_5.5                stringi_1.7.5          Rcpp_1.0.7            
[70] vctrs_0.3.8            tidyselect_1.1.1       xfun_0.22             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
