<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>E coli spike-in experiments</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rare mutation detection</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="model.html">Model</a>
</li>
<li>
  <a href="ecoli.html">E coli BL21 results</a>
</li>
<li>
  <a href="ecoli_K12.html">E coli K12 results</a>
</li>
<li>
  <a href="compare_MGI_Illumina.html">Illumina vs. MGI</a>
</li>
<li>
  <a href="insilico_mixtures.html">in silico mixtures</a>
</li>
<li>
  <a href="ecoli_spikeins.html">E coli spike-ins</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">E coli spike-in experiments</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rare-mutation-detection/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210916code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210916code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210916)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontree2ec3680ce8599168a5afeca9dae2cb60e4bb2d62targetblank2ec3680a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mcmero/rare-mutation-detection/tree/2ec3680ce8599168a5afeca9dae2cb60e4bb2d62" target="_blank">2ec3680</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontree2ec3680ce8599168a5afeca9dae2cb60e4bb2d62targetblank2ec3680a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mcmero/rare-mutation-detection/tree/2ec3680ce8599168a5afeca9dae2cb60e4bb2d62" target="_blank">2ec3680</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    scripts/

Untracked files:
    Untracked:  ._.DS_Store
    Untracked:  DOCNAME
    Untracked:  analysis/._.DS_Store
    Untracked:  analysis/cache/
    Untracked:  analysis/calc_nanoseq_metrics.Rmd
    Untracked:  data/._metrics.rds
    Untracked:  data/ecoli/
    Untracked:  data/metadata/
    Untracked:  data/metrics_efficiency_nossc.rds
    Untracked:  data/mixtures
    Untracked:  data/ref/
    Untracked:  prototype_code/

Unstaged changes:
    Modified:   analysis/ecoli_K12.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/ecoli_spikeins.Rmd</code>) and
HTML (<code>docs/ecoli_spikeins.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/2ec3680ce8599168a5afeca9dae2cb60e4bb2d62/analysis/ecoli_spikeins.Rmd" target="_blank">2ec3680</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-07-12
</td>
<td>
Added preliminary QC results/metrics from spike-in experiments
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="e-coli-spike-in-experiment-results" class="section level1">
<h1>E coli spike-in experiment results</h1>
<p>E coli K12 strain was spiked into E coli BL21 with different
proportions:</p>
<table>
<thead>
<tr class="header">
<th>Lib Name</th>
<th>Spike in %</th>
<th>~Cell equivalent*</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0-K12Rep1</td>
<td>0%K12Rep1(BL2 only)</td>
<td>318</td>
</tr>
<tr class="even">
<td>0-K12Rep2</td>
<td>0%K12Rep2 (BL2 only)</td>
<td>202</td>
</tr>
<tr class="odd">
<td>1-K12Rep1</td>
<td>1%K12Rep1</td>
<td>601</td>
</tr>
<tr class="even">
<td>1-K12Rep2</td>
<td>1%K12Rep2</td>
<td>585</td>
</tr>
<tr class="odd">
<td>10-K12Rep1</td>
<td>10%K12Rep1</td>
<td>86</td>
</tr>
<tr class="even">
<td>10-K12Rep2</td>
<td>10%K12Rep2</td>
<td>74</td>
</tr>
<tr class="odd">
<td>1_10-K12Rep1</td>
<td>0.1%K12Rep1</td>
<td>11,139</td>
</tr>
<tr class="even">
<td>5-K12Rep1</td>
<td>5%K12Rep1</td>
<td>188</td>
</tr>
<tr class="odd">
<td>5-K12Rep2</td>
<td>5%K12Rep2</td>
<td>228</td>
</tr>
</tbody>
</table>
<p>*based on R1 unique read number.</p>
<p>The 1_10-K12Rep1 sample is currently omitted in this analysis as it
is too large to process with the existing script.</p>
<div id="multiqc-reports" class="section level2">
<h2>MultiQC reports:</h2>
<ul>
<li><a
href="AGRF_CAGRF220410419_HFVGHDSX3_multiqc/multiqc_report.html">AGRF_CAGRF220410419_HFVGHDSX3
MultiQC pre-duplex</a></li>
<li><a
href="AGRF_CAGRF220410419_HFVGHDSX3_multiqc/multiqc_report_consensus.html">AGRF_CAGRF220410419_HFVGHDSX3
MultiQC duplex</a></li>
</ul>
<pre class="r"><code>library(ggplot2)
library(data.table)
library(dplyr)
library(here)
library(tibble)
library(stringr)
library(Rsamtools)
library(GenomicRanges)
library(seqinr)
library(parallel)
library(readxl)
library(patchwork)
library(RColorBrewer)
library(UpSetR)
library(vcfR)</code></pre>
<pre class="r"><code>source(here(&#39;code/load_data.R&#39;))
source(here(&#39;code/plot.R&#39;))
source(here(&#39;code/efficiency_nanoseq_functions.R&#39;))</code></pre>
<pre class="r"><code>genome_max &lt;- 4528118
cores &lt;- 8</code></pre>
<pre class="r"><code>genomeFile &lt;- here(&#39;data/ref/Escherichia_coli_strain_BL21_TaKaRa.fasta&#39;)
rinfo_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF220410419_HFVGHDSX3/QC/read_info&#39;)
markdup_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF220410419_HFVGHDSX3/QC/mark_duplicates&#39;)
qualimap_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF220410419_HFVGHDSX3/QC/qualimap&#39;)
qualimap_cons_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF220410419_HFVGHDSX3/QC/consensus/qualimap&#39;)
variant_dir &lt;- here(&#39;data/ecoli/AGRF_CAGRF220410419_HFVGHDSX3/variants&#39;)</code></pre>
<pre class="r"><code>sample_names &lt;- list.files(rinfo_dir) %&gt;%
                str_split(&#39;\\.txt.gz&#39;) %&gt;%
                lapply(., dplyr::first) %&gt;%
                unlist() %&gt;%
                str_split(&#39;_&#39;) %&gt;%
                lapply(., head, 2) %&gt;%
                lapply(., paste, collapse=&#39;-&#39;) %&gt;%
                unlist()

# load variant data
var_df &lt;- load_variants(variant_dir, sample_names)

# load and fetch duplicate rate from MarkDuplicates output
mdup &lt;- load_markdup_data(markdup_dir, sample_names)

# get mean coverage for pre and post-consensus reads
# filter out 1-10-K12Rep1 for now
qmap_cov &lt;- list.files(
        qualimap_dir,
        full.names = TRUE,
        recursive = TRUE,
        pattern = &#39;genome_results.txt&#39;) %&gt;%
        lapply(., function(x){if(!x %like% &#39;1-10-K12Rep1&#39;){x}}) %&gt;% unlist() %&gt;%
        lapply(., read.delim)
for (i in seq(length(sample_names))) {
        qmap_cov[[i]]$Sample &lt;- sample_names[i]
}
qmap_cov &lt;- rbindlist(qmap_cov)
qmap_cov &lt;- qmap_cov[qmap_cov$BamQC.report %like% &#39;mean cov&#39;,]
qmap_cov$coverage &lt;-
        str_split(qmap_cov$BamQC.report, &#39; = &#39;) %&gt;%
        lapply(., last) %&gt;%
        unlist() %&gt;%
        gsub(&#39;X|,&#39;, &#39;&#39;, .) %&gt;%
        lapply(., as.numeric) %&gt;%
        unlist()
qmap_cov &lt;- qmap_cov[,c(&#39;Sample&#39;, &#39;coverage&#39;)]


qmap_cov_cons &lt;- list.files(
        qualimap_cons_dir,
        full.names = TRUE,
        recursive = TRUE,
        pattern = &#39;genome_results.txt&#39;) %&gt;%
        lapply(., function(x){if(!x %like% &#39;1-10-K12Rep1&#39;){x}}) %&gt;% unlist() %&gt;%
        lapply(., read.delim)
for (i in seq(length(sample_names))) {
        qmap_cov_cons[[i]]$Sample &lt;- sample_names[i]
}
qmap_cov_cons &lt;- rbindlist(qmap_cov_cons)
qmap_cov_cons &lt;- qmap_cov_cons[qmap_cov_cons$BamQC.report %like% &#39;mean cov&#39;,]
qmap_cov_cons$coverage &lt;-
        str_split(qmap_cov_cons$BamQC.report, &#39; = &#39;) %&gt;%
        lapply(., last) %&gt;%
        unlist() %&gt;%
        gsub(&#39;X|,&#39;, &#39;&#39;, .) %&gt;%
        lapply(., as.numeric) %&gt;%
        unlist()
qmap_cov_cons &lt;- qmap_cov_cons[,c(&#39;Sample&#39;, &#39;coverage&#39;)]

# uncomment below to calculate metrics
# calculate metrics for nanoseq
rlen &lt;- 151; skips &lt;- 5
metrics &lt;- calc_metrics_new_rbs(rinfo_dir, cores = cores) %&gt;% bind_rows()

metrics$duplicate_rate &lt;- as.numeric(mdup)
metrics$duplex_coverage_ratio &lt;- qmap_cov$coverage / qmap_cov_cons$coverage
metrics$duplex_coverage_ratio[qmap_cov_cons$coverage &lt; 1] &lt;- 0 # fix when &lt; 1 duplex cov
metrics$sample &lt;- gsub(&#39;-HFVGHDSX3&#39;, &#39;&#39;, sample_names)

# cache metrics object
# saveRDS(metrics, file = here(&#39;data/metrics.rds&#39;))

# prepare for plotting
mm &lt;- data.frame(melt(metrics))
colnames(mm)[2] &lt;- &#39;metric&#39;</code></pre>
<pre class="r"><code>ggplot(mm, aes(sample, value)) + 
    geom_point() +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90)) +
    facet_wrap(~metric, scales = &#39;free&#39;) +
    scale_colour_brewer(palette = &#39;Dark2&#39;)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/plot_by_sample-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="metric-comparison-plots" class="section level2">
<h2>Metric comparison plots</h2>
<div id="duplicate-rate" class="section level3">
<h3>Duplicate rate</h3>
<p>Fraction of duplicate reads calculated by <a
href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-">Picard’s
MarkDuplicates</a>. This is based on barcode-aware aligned duplicates
mapping to the same 5’ positions for both read pairs. The <a
href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis pipeline</a>
states the optimal empirical duplicate rate is 75-76% (marked in the
plot).</p>
<pre class="r"><code>metric &lt;- &#39;duplicate_rate&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.75, 0.76), alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/duplicate_rate-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fraction-of-singleton-reads" class="section level3">
<h3>Fraction of singleton reads</h3>
<p>Shows the number of single-read families divided by the total number
of reads. As suggested by <a
href="https://doi.org/10.1186/s13059-016-1039-4">Stoler et al. 2016</a>,
this metric can server as a proxy for error rate, as (uncorrected)
barcode mismatches will manifest as single-read families. The lower the
fraction of singletons, the better.</p>
<pre class="r"><code>metric &lt;- &#39;frac_singletons&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/singletons-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="drop-out-rate" class="section level3">
<h3>Drop-out rate</h3>
<p>This is the same calculation as F-EFF in the <a
href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis
pipeline</a>:</p>
<p>“This shows the fraction of read bundles missing one of the two
original strands beyond what would be expected under random sampling
(assuming a binomial process). Good values are between 0.10-0.30, and
larger values are likely due to DNA damage such as modified bases or
internal nicks that prevent amplification of one of the two strands.
Larger values do not impact the quality of the results, just reduce the
efficiency of the protocol.”</p>
<p>This is similar to the singleton fraction, but taking into account
loss of pairs due to sampling. The optimal range is shown by the
lines.</p>
<pre class="r"><code>metric &lt;- &#39;drop_out_rate&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.1, 0.3), alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/drop_out_rate-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="efficiency" class="section level3">
<h3>Efficiency</h3>
<p>Efficiency is the number of duplex bases divided by the number of
sequenced bases. According the <a
href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis
pipeline</a>, this value is maximised at ~0.07 when duplicate rates and
strand drop-outs are optimal.</p>
<pre class="r"><code>metric &lt;- &#39;efficiency&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.07), alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/efficiency-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gc-deviation" class="section level3">
<h3>GC deviation</h3>
<p>GC deviation is the absolute difference between <code>GC_BOTH</code>
and <code>GC_SINGLE</code> calculated by the <a
href="https://github.com/cancerit/NanoSeq">NanoSeq Analysis
pipeline</a>. The lower this deviation, the better.</p>
<p>“GC_BOTH and GC_SINGLE: the GC content of RBs with both strands and
with just one strand. The two values should be similar between them and
similar to the genome average. If there are large deviations that is
possibly due to biases during PCR amplification. If GC_BOTH is
substantially larger than GC_SINGLE, DNA denaturation before dilution
may have taken place.”</p>
<pre class="r"><code>metric &lt;- &#39;gc_deviation&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/gc_deviation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="duplex-coverage-ratio" class="section level3">
<h3>Duplex Coverage ratio</h3>
<p>The mean sequence (pre-duplex) coverage divided by mean duplex
coverage. Indicates the yield of how much duplex coverage we get at each
sample’s sequence coverage. <a
href="https://doi.org/10.1038/s41586-021-03477-4">Abascal et al.</a>
report that their yield was approximately 30x (marked on the plot).</p>
<pre class="r"><code>metric &lt;- &#39;duplex_coverage_ratio&#39;
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = 30, alpha = 0.4)  +
        ggtitle(metric)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/duplex_coverage_ratio-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="family-statistics" class="section level2">
<h2>Family statistics</h2>
<p>Comparison of family pair sizes between samples (these are calculated
from total reads of paired AB and BA families).</p>
<pre class="r"><code>ggplot(mm[mm$metric %like% &#39;family&#39;, ], aes(value, sample, colour = metric)) +
        geom_point() +
        coord_trans(x=&#39;log2&#39;) +
        scale_x_continuous(breaks=seq(0, 94, 8)) +
        theme(axis.text.x = element_text(size=5)) +
        theme_bw() +
        ggtitle(&#39;Family pair sizes&#39;)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/family_sizes-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>The following plot shows:</p>
<ul>
<li>families_gt1: number of family pairs where at least one family (AB
or BA) has &gt; 1 reads.</li>
<li>paired_families: number of family pairs where both families (AB and
BA) have &gt; 0 reads.</li>
<li>paired_and_gt1: number of family pairs where both families (AB and
BA) have &gt; 1 reads.</li>
</ul>
<pre class="r"><code>ggplot(mm[mm$metric %like% &#39;pair|gt1&#39;, ], aes(value, sample, fill = metric)) +
        geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
        theme_bw() +
        ggtitle(&#39;Family statistics&#39;)</code></pre>
<p><img src="figure/ecoli_spikeins.Rmd/family_stats-1.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRblas.so
LAPACK: /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] vcfR_1.12.0          UpSetR_1.4.0         RColorBrewer_1.1-3  
 [4] patchwork_1.1.1      readxl_1.3.1         seqinr_4.2-8        
 [7] Rsamtools_2.6.0      Biostrings_2.58.0    XVector_0.30.0      
[10] GenomicRanges_1.42.0 GenomeInfoDb_1.26.7  IRanges_2.24.1      
[13] S4Vectors_0.28.1     BiocGenerics_0.36.1  stringr_1.4.0       
[16] tibble_3.1.7         here_1.0.1           dplyr_1.0.7         
[19] data.table_1.14.0    ggplot2_3.3.6        workflowr_1.6.2     

loaded via a namespace (and not attached):
 [1] nlme_3.1-152           bitops_1.0-7           fs_1.5.0              
 [4] rprojroot_2.0.2        tools_4.0.5            bslib_0.3.0           
 [7] utf8_1.2.2             R6_2.5.1               vegan_2.5-7           
[10] DBI_1.1.1              mgcv_1.8-35            colorspace_2.0-3      
[13] permute_0.9-5          ade4_1.7-18            withr_2.5.0           
[16] tidyselect_1.1.1       gridExtra_2.3          compiler_4.0.5        
[19] git2r_0.28.0           cli_3.3.0              labeling_0.4.2        
[22] sass_0.4.0             scales_1.2.0           digest_0.6.29         
[25] rmarkdown_2.11         pkgconfig_2.0.3        htmltools_0.5.2       
[28] highr_0.9              fastmap_1.1.0          rlang_1.0.2           
[31] rstudioapi_0.13        jquerylib_0.1.4        generics_0.1.1        
[34] farver_2.1.0           jsonlite_1.7.2         BiocParallel_1.24.1   
[37] RCurl_1.98-1.3         magrittr_2.0.3         GenomeInfoDbData_1.2.4
[40] Matrix_1.3-2           Rcpp_1.0.7             munsell_0.5.0         
[43] fansi_1.0.3            ape_5.5                lifecycle_1.0.1       
[46] stringi_1.7.5          whisker_0.4            yaml_2.2.1            
[49] MASS_7.3-53.1          zlibbioc_1.36.0        plyr_1.8.6            
[52] pinfsc50_1.2.0         grid_4.0.5             promises_1.2.0.1      
[55] crayon_1.5.1           lattice_0.20-44        splines_4.0.5         
[58] knitr_1.33             pillar_1.7.0           reshape2_1.4.4        
[61] glue_1.6.2             evaluate_0.14          memuse_4.2-1          
[64] vctrs_0.4.1            httpuv_1.6.3           cellranger_1.1.0      
[67] gtable_0.3.0           purrr_0.3.4            assertthat_0.2.1      
[70] xfun_0.22              later_1.3.0            viridisLite_0.4.0     
[73] cluster_2.1.2          ellipsis_0.3.2        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
