<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Compare sequencing metrics from jafarJ_201021 E coli data</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rare mutation detection</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="model.html">Model</a>
</li>
<li>
  <a href="ecoli.html">E coli BL21 results</a>
</li>
<li>
  <a href="ecoli_K12.html">E coli K12 results</a>
</li>
<li>
  <a href="compare_MGI_Illumina.html">Illumina vs. MGI</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Compare sequencing metrics from jafarJ_201021 E coli data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rare-mutation-detection/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontree1926d3dc0acbe289fdff366b0becac83d9b3bc30targetblank1926d3da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mcmero/rare-mutation-detection/tree/1926d3dc0acbe289fdff366b0becac83d9b3bc30" target="_blank">1926d3d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontree1926d3dc0acbe289fdff366b0becac83d9b3bc30targetblank1926d3da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mcmero/rare-mutation-detection/tree/1926d3dc0acbe289fdff366b0becac83d9b3bc30" target="_blank">1926d3d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    scripts/

Untracked files:
    Untracked:  ._.DS_Store
    Untracked:  DOCNAME
    Untracked:  analysis/._.DS_Store
    Untracked:  analysis/cache/
    Untracked:  analysis/calc_nanoseq_metrics.Rmd
    Untracked:  data/
    Untracked:  prototype_code/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   code/efficiency_nanoseq_functions.R
    Modified:   code/load_data.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ecoli.Rmd</code>) and HTML (<code>docs/ecoli.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/5fd8f9aedcec59b1a05ddc0f1ed2fd772ba2b3d4/docs/ecoli.html" target="_blank">5fd8f9a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/b257d52745fd15568405a5d1ea50a52a99d140eb/analysis/ecoli.Rmd" target="_blank">b257d52</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-03-10
</td>
<td>
Added MGI vs. Illumina comparison; refactoring ecoli.Rmd code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/ecoli.html" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7cfe3e492e9c8473cd39878ceacf3e0d256ef992/analysis/ecoli.Rmd" target="_blank">7cfe3e4</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
<td>
Updated/simplified model
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/ecoli.html" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/ac440bc83c779bdd24667c207eee43eaaf8a677a/analysis/ecoli.Rmd" target="_blank">ac440bc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
<td>
Added NanoSeq stats &amp; comparison plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/ecoli.html" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/47858f3e84cacaa296c3f68f97077bac9d453746/analysis/ecoli.Rmd" target="_blank">47858f3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
<td>
Added genotype plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/02e8cedcebfbefc90470f68ac925107d637147e1/docs/ecoli.html" target="_blank">02e8ced</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/ce8d35de59ffe4c1e3589464a0a90d54274d9cde/analysis/ecoli.Rmd" target="_blank">ce8d35d</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
<td>
discordant variant analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/ecoli.html" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7664b094d18fbb1bbd95c9c0ade9d179777673dc/analysis/ecoli.Rmd" target="_blank">7664b09</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
<td>
Updated to handle VCF output
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/e5ed9a72a7c15c0b9078b6ae9c7d9b3f2376d27c/docs/ecoli.html" target="_blank">e5ed9a7</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/ed42fa9c24ae607540808bca260924f57cc823d2/analysis/ecoli.Rmd" target="_blank">ed42fa9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
<td>
Added multiQC reports
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/de277d979b1bd9622005ea47797cf6792201fd97/docs/ecoli.html" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/e610f975ca124e971530a78f8577de1ac001d0c7/analysis/ecoli.Rmd" target="_blank">e610f97</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
<td>
Added ecoli analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="compare-sequencing-metrics-from-jafarj_201021-e-coli-data" class="section level1">
<h1>Compare sequencing metrics from jafarJ_201021 E coli data</h1>
<p>These are extra stats that are not available in the MultiQC reports. These reports can be found below:</p>
<ul>
<li><a href="jafarJ_201021_multiqc/multiqc_report.html">jafarJ_201021 MultiQC pre-duplex</a></li>
<li><a href="jafarJ_201021_multiqc/multiqc_report_consensus.html">jafarJ_201021 MultiQC duplex</a></li>
</ul>
<pre class="r"><code>library(ggplot2)
library(data.table)
library(dplyr)
library(R.utils)
library(UpSetR)
library(here)
library(vcfR)
library(tibble)
library(stringr)
library(patchwork)</code></pre>
<pre class="r"><code>source(here(&#39;code/load_data.R&#39;))
source(here(&#39;code/plot.R&#39;))</code></pre>
<pre class="r"><code>qualimap_dir &lt;- here(&#39;data/ecoli/jafarJ_201021/QC/qualimap/&#39;)
qualimap_cons_dir &lt;- here(&#39;data/ecoli/jafarJ_201021/QC/consensus/qualimap/&#39;)
variant_dir &lt;- here(&#39;data/ecoli/jafarJ_201021/variants&#39;)
family_size_stats &lt;- here(&#39;data/ecoli/jafarJ_201021_family_sizes.txt&#39;)
nanoseq_dir &lt;- here(&#39;data/ecoli/jafarJ_201021_nanoseq_results&#39;)

samples &lt;- list.files(qualimap_dir)

cov &lt;- load_data(qualimap_dir, &#39;coverage_across_reference.txt&#39;, samples)
ccov &lt;- load_data(qualimap_cons_dir, &#39;coverage_across_reference.txt&#39;, samples)
clip &lt;- load_data(qualimap_dir, &#39;mapped_reads_clipping_profile&#39;, samples)
cclip &lt;- load_data(qualimap_cons_dir, &#39;mapped_reads_clipping_profile&#39;, samples)
vars &lt;- load_data(variant_dir, &#39;.vcf&#39;, samples, read.table)
cov_stats &lt;- load_cov_stats(cov, qualimap_dir, samples)
ccov_stats &lt;- load_cov_stats(ccov, qualimap_cons_dir, samples)
fam &lt;- read.delim(family_size_stats, sep=&#39;\t&#39;)

# TODO: replace these stats with new functions used in compare_MGI_Illumina.Rmd
tsvs &lt;- load_nanoseq_stats(nanoseq_dir)</code></pre>
<div id="coverage-boxplot" class="section level2">
<h2>Coverage boxplot</h2>
<p>Using coverage summary data from Qualimap (I assume these are summarised to 10kb windows, though I couldn’t find this in the documentation).</p>
<pre class="r"><code># order by median coverage
median_cov &lt;- data.table(cov)[,median(Coverage), by=Sample]
sample_order &lt;- median_cov[order(median_cov$V1)]$Sample
cov$Sample &lt;- factor(cov$Sample, levels = sample_order)
p1 &lt;- ggplot(cov, aes(Coverage, Sample, colour = protocol)) + geom_boxplot() + theme_bw() + ggtitle(&#39;Pre-duplex coverage&#39;)

median_cov &lt;- data.table(ccov)[,median(Coverage), by=Sample]
ccov$Sample &lt;- factor(ccov$Sample, levels = sample_order)
p2 &lt;- ggplot(ccov, aes(Coverage, Sample, colour = protocol)) + geom_boxplot() + theme_bw() + ggtitle(&#39;Duplex coverage&#39;)

p1 + p2</code></pre>
<p><img src="figure/ecoli.Rmd/coverage_boxplots-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-coverage_boxplots-1">
Past versions of coverage_boxplots-1.png
</button>
</p>
<div id="fig-coverage_boxplots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/figure/ecoli.Rmd/coverage_boxplots-1.png" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/coverage_boxplots-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/coverage_boxplots-1.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/coverage_boxplots-1.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="coverage-standard-deviation-bar-plot" class="section level2">
<h2>Coverage standard deviation bar plot</h2>
<pre class="r"><code>ggplot(melt(cov_stats), aes(value, Sample, fill=variable)) +
    geom_bar(stat=&#39;identity&#39;, position = &#39;dodge&#39;) +
    theme_bw() +
    ggtitle(&#39;Coverage std &amp; mean (pre-duplex)&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/cov_std-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cov_std-1">
Past versions of cov_std-1.png
</button>
</p>
<div id="fig-cov_std-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/cov_std-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/cov_std-1.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/cov_std-1.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(melt(ccov_stats), aes(value, Sample, fill=variable)) +
    geom_bar(stat=&#39;identity&#39;, position = &#39;dodge&#39;) +
    theme_bw() +
    ggtitle(&#39;Coverage std &amp; mean (duplex)&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/cov_std-2.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cov_std-2">
Past versions of cov_std-2.png
</button>
</p>
<div id="fig-cov_std-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/cov_std-2.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/cov_std-2.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/cov_std-2.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cov_stats$cov_cv &lt;- cov_stats$cov_std / cov_stats$cov_mean
ggplot(cov_stats, aes(cov_cv, Sample)) +
    geom_bar(stat=&#39;identity&#39;, position = &#39;dodge&#39;) +
    theme_bw() +
    ggtitle(&#39;Coverage CV (pre-duplex)&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/cov_std-3.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cov_std-3">
Past versions of cov_std-3.png
</button>
</p>
<div id="fig-cov_std-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/cov_std-3.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/cov_std-3.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/cov_std-3.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ccov_stats$cov_cv &lt;- ccov_stats$cov_std / ccov_stats$cov_mean
ggplot(ccov_stats, aes(cov_cv, Sample)) +
    geom_bar(stat=&#39;identity&#39;, position = &#39;dodge&#39;) +
    theme_bw() +
    ggtitle(&#39;Coverage CV (duplex)&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/cov_std-4.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cov_std-4">
Past versions of cov_std-4.png
</button>
</p>
<div id="fig-cov_std-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/cov_std-4.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/cov_std-4.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/cov_std-4.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="clipping-profile" class="section level2">
<h2>Clipping profile</h2>
<p>Pre-duplex reads prior to overlap clipping, but post-UMI removal.</p>
<pre class="r"><code>ggplot(clip, aes(X.Read.position..bp., Clipping.profile)) +
    geom_line() +
    theme_bw() +
    xlab(&#39;Read position&#39;) +
    facet_wrap(~Sample) +
    ggtitle(&#39;Pre-duplex clipping profile&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/clipping_profile-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clipping_profile-1">
Past versions of clipping_profile-1.png
</button>
</p>
<div id="fig-clipping_profile-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/clipping_profile-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/clipping_profile-1.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/clipping_profile-1.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Duplex reads have been clipped to remove read overlap.</p>
<pre class="r"><code>ggplot(cclip, aes(X.Read.position..bp., Clipping.profile)) +
    geom_line() +
    theme_bw() +
    xlab(&#39;Read position&#39;) +
    facet_wrap(~Sample) +
    ggtitle(&#39;Duplex clipping profile&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/clipping_profile_duplex-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clipping_profile_duplex-1">
Past versions of clipping_profile_duplex-1.png
</button>
</p>
<div id="fig-clipping_profile_duplex-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/clipping_profile_duplex-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/clipping_profile_duplex-1.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/clipping_profile_duplex-1.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="family-size-stats-duplex-statistics" class="section level2">
<h2>Family size stats (duplex statistics)</h2>
<ul>
<li>frac_fam_gt1: the fraction of families where family size is greater than one.</li>
<li>frac_fam_paired: the fraction of paired families (a family on each strand with the same UMI).</li>
<li>frac_fam_paired_gt1: the fraction of families that are greater than one in size, and also paired.</li>
</ul>
<pre class="r"><code>ggplot(fam, aes(len, sample)) +
    geom_bar(stat=&#39;identity&#39;) +
    theme_bw() +
    ggtitle(&#39;Total family count&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/family_size-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-family_size-1">
Past versions of family_size-1.png
</button>
</p>
<div id="fig-family_size-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/family_size-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/family_size-1.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/family_size-1.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mfam &lt;- reshape2::melt(fam[,c(&#39;sample&#39;, &#39;min&#39;, &#39;max&#39;, &#39;mean&#39;, &#39;median&#39;)])
ggplot(mfam, aes(value, sample, colour=variable)) +
    geom_point() +
    theme_bw() +
    coord_trans(x=&#39;log2&#39;) +
    scale_x_continuous(breaks=seq(0, 25, 2)) +
    theme(axis.text.x = element_text(size=6)) +
    ggtitle(&#39;Family sizes&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/family_size-2.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-family_size-2">
Past versions of family_size-2.png
</button>
</p>
<div id="fig-family_size-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/family_size-2.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/family_size-2.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/family_size-2.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mfam &lt;- reshape2::melt(fam[,colnames(fam) %like% &#39;frac|sample&#39;])
ggplot(mfam, aes(value, sample, fill=variable)) +
    geom_bar(stat=&#39;identity&#39;, position=&#39;dodge&#39;) +
    theme_bw() +
    ggtitle(&#39;Family reads + paired statistics&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/family_size-3.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-family_size-3">
Past versions of family_size-3.png
</button>
</p>
<div id="fig-family_size-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/family_size-3.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f3d146c8d97fa2d9e3cc3db4b845ccd88494d56e/docs/figure/ecoli.Rmd/family_size-3.png" target="_blank">f3d146c</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de277d979b1bd9622005ea47797cf6792201fd97/docs/figure/ecoli.Rmd/family_size-3.png" target="_blank">de277d9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="compare-single-read-fraction" class="section level3">
<h3>Compare single-read fraction</h3>
<pre class="r"><code># extract protocol and nuclease labels
fam$protocol &lt;- &#39;NanoSeq&#39;
fam$protocol[grep(&#39;Nux&#39;, fam$sample)] &lt;- &#39;xGen&#39;
fam$nuclease &lt;- str_split(fam$sample, regex(&quot;N(uxg|an)|_&quot;)) %&gt;%
                    lapply(., tail, 2) %&gt;% lapply(., dplyr::first) %&gt;% unlist()

mfam &lt;- reshape2::melt(fam[,colnames(fam) %like% &#39;fam_gt1|sample|protocol|nuc&#39;])
ggplot(mfam, aes(protocol, value)) +
    geom_boxplot() +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    theme_bw() +
    ggtitle(&#39;Fraction families with size &gt; 1 by protocol&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/compare_famsize-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_famsize-1">
Past versions of compare_famsize-1.png
</button>
</p>
<div id="fig-compare_famsize-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_famsize-1.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(mfam, aes(nuclease, value)) +
    geom_boxplot() +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    theme_bw() +
    ggtitle(&#39;Fraction families with size &gt; 1 by nuclease&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/compare_famsize-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_famsize-2">
Past versions of compare_famsize-2.png
</button>
</p>
<div id="fig-compare_famsize-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_famsize-2.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="variants" class="section level2">
<h2>Variants</h2>
<pre class="r"><code>vars$Sample &lt;- strsplit(vars$Sample, &#39;\\.&#39;) %&gt;% lapply(., head, 1) %&gt;% unlist()

ulist &lt;- NULL
for(sample in samples) {
    ulist[[sample]] &lt;- vars[vars$Sample %in% sample,]$V2
}

upset(fromList(ulist), order.by=&#39;freq&#39;, nsets=8)</code></pre>
<p><img src="figure/ecoli.Rmd/variant_overlap-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-variant_overlap-1">
Past versions of variant_overlap-1.png
</button>
</p>
<div id="fig-variant_overlap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/variant_overlap-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/02e8cedcebfbefc90470f68ac925107d637147e1/docs/figure/ecoli.Rmd/variant_overlap-1.png" target="_blank">02e8ced</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Check how many variants have at least one reference base called.</p>
<pre class="r"><code>vcfs &lt;- list.files(variant_dir,
                  full.names = TRUE) %&gt;%
        lapply(., read.vcfR, verbose = FALSE) %&gt;%
        lapply(., vcfR2tidy,
               format_fields = c(&#39;GT&#39;, &#39;AD&#39;, &#39;RD&#39;),
               verbose = FALSE)

disc &lt;- lapply(vcfs, function(vcf) {
          has_ref &lt;- vcf$gt$gt_RD &gt; 0
          row &lt;- c(sum(has_ref), nrow(vcf$gt))
          return(row)
        })

disc &lt;- data.frame(disc, row.names = c(&#39;has_ref_bases&#39;,&#39;total_variants&#39;)) %&gt;% t() %&gt;%
          data.frame(row.names = 1:nrow(.)) %&gt;%
          add_column(sample=samples)

ggplot(melt(disc), aes(value, sample, fill = variable)) +
  geom_histogram(stat = &#39;identity&#39;, position = &#39;dodge&#39;) +
  theme_bw()</code></pre>
<p><img src="figure/ecoli.Rmd/discordant_variants-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-discordant_variants-1">
Past versions of discordant_variants-1.png
</button>
</p>
<div id="fig-discordant_variants-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/discordant_variants-1.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/02e8cedcebfbefc90470f68ac925107d637147e1/docs/figure/ecoli.Rmd/discordant_variants-1.png" target="_blank">02e8ced</a>
</td>
<td>
mcmero
</td>
<td>
2021-12-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gts &lt;- NULL
for(i in 1:length(samples)) {
    vcfs[[i]]$gt$sample &lt;- samples[i]
    gts &lt;- rbind(gts, vcfs[[i]]$gt)
}

ggplot(gts, aes(sample, fill = gt_GT)) +
    geom_histogram(stat = &#39;count&#39;, position = &#39;dodge&#39;) +
    theme_bw() +
    coord_flip() +
    ggtitle(&#39;Variant genotype&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/discordant_variants-2.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-discordant_variants-2">
Past versions of discordant_variants-2.png
</button>
</p>
<div id="fig-discordant_variants-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/bacece1c5156523ae03ba580852f4e1bf04cea4e/docs/figure/ecoli.Rmd/discordant_variants-2.png" target="_blank">bacece1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="statistics-from-nanoseq-pipeline" class="section level2">
<h2>Statistics from NanoSeq pipeline</h2>
<p>Stats obtained from <a href="https://github.com/cancerit/NanoSeq">NanoSeq</a> pipeline.</p>
<pre class="r"><code>plot_metric(tsvs, &#39;READS$&#39;, &#39;Num reads&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-1">
Past versions of nanoseq_stats-1.png
</button>
</p>
<div id="fig-nanoseq_stats-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-1.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;DUPLICATE&#39;, &#39;Duplicate rate (line = optimal)&#39;) +
    geom_hline(yintercept = 0.81)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-2.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-2">
Past versions of nanoseq_stats-2.png
</button>
</p>
<div id="fig-nanoseq_stats-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-2.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;TOTAL&#39;, &#39;Total RBs (read barcodes)&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-3.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-3">
Past versions of nanoseq_stats-3.png
</button>
</p>
<div id="fig-nanoseq_stats-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-3.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;PER&#39;, &#39;Reads per RB&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-4.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-4">
Past versions of nanoseq_stats-4.png
</button>
</p>
<div id="fig-nanoseq_stats-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-4.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;OK&#39;, &#39;OK RBs (2 on each strand)&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-5.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-5">
Past versions of nanoseq_stats-5.png
</button>
</p>
<div id="fig-nanoseq_stats-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-5.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;F-EFF&#39;,
             &#39;F-EFF (drop out fraction, lines show optimal range)&#39;) +
    geom_hline(yintercept = c(0.1, 0.3))</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-6.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-6">
Past versions of nanoseq_stats-6.png
</button>
</p>
<div id="fig-nanoseq_stats-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-6.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;EFFICIENCY&#39;,&#39;Efficiency (line shows maximised value)&#39;) +
    geom_hline(yintercept = 0.07)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-7.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-7">
Past versions of nanoseq_stats-7.png
</button>
</p>
<div id="fig-nanoseq_stats-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-7.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric(tsvs, &#39;(GC_BOTH|GC_SINGLE)&#39;, &#39;GC per strand&#39;)</code></pre>
<p><img src="figure/ecoli.Rmd/nanoseq_stats-8.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-nanoseq_stats-8">
Past versions of nanoseq_stats-8.png
</button>
</p>
<div id="fig-nanoseq_stats-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/nanoseq_stats-8.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="comparison-of-key-stats" class="section level3">
<h3>Comparison of key stats</h3>
<pre class="r"><code># duplicate rate
plot_metric_boxplot(tsvs, &#39;protocol&#39;, &#39;DUPLICATE&#39;, &#39;Duplicate rate (line = optimal&#39;) +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    geom_hline(yintercept = 0.81)</code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-1">
Past versions of compare_nanoseq-1.png
</button>
</p>
<div id="fig-compare_nanoseq-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-1.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric_boxplot(tsvs, &#39;nuclease&#39;, &#39;DUPLICATE&#39;, &#39;Duplicate rate (line = optimal&#39;) +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    geom_hline(yintercept = 0.81)</code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-2">
Past versions of compare_nanoseq-2.png
</button>
</p>
<div id="fig-compare_nanoseq-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-2.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># efficiency
plot_metric_boxplot(tsvs, &#39;protocol&#39;, &#39;EFFICIENCY&#39;, &#39;Efficiency (line = optimal)&#39;) +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    geom_hline(yintercept = 0.07) </code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-3">
Past versions of compare_nanoseq-3.png
</button>
</p>
<div id="fig-compare_nanoseq-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-3.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric_boxplot(tsvs, &#39;nuclease&#39;, &#39;EFFICIENCY&#39;, &#39;Efficiency (line = optimal)&#39;) +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    geom_hline(yintercept = 0.07) </code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-4">
Past versions of compare_nanoseq-4.png
</button>
</p>
<div id="fig-compare_nanoseq-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-4.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># drop out rate
plot_metric_boxplot(tsvs, &#39;protocol&#39;, &#39;F-EFF&#39;, &#39;Strand drop-out fraction (lines = optimal range)&#39;) +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    geom_hline(yintercept = c(0.1, 0.3))</code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-5.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-5">
Past versions of compare_nanoseq-5.png
</button>
</p>
<div id="fig-compare_nanoseq-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-5.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric_boxplot(tsvs, &#39;nuclease&#39;, &#39;F-EFF&#39;, &#39;Strand drop-out fraction (lines = optimal range)&#39;) +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    geom_hline(yintercept = c(0.1, 0.3))</code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-6.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-6">
Past versions of compare_nanoseq-6.png
</button>
</p>
<div id="fig-compare_nanoseq-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-6.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># GC deviation between strands
plot_metric_boxplot(tsvs, &#39;protocol&#39;, &#39;GC_DEV&#39;, &#39;GC deviation (both strands vs. one)&#39;) +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease))</code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-7.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-7">
Past versions of compare_nanoseq-7.png
</button>
</p>
<div id="fig-compare_nanoseq-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-7.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_metric_boxplot(tsvs, &#39;nuclease&#39;, &#39;GC_DEV&#39;, &#39;GC deviation (both strands vs. one)&#39;) +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol))</code></pre>
<p><img src="figure/ecoli.Rmd/compare_nanoseq-8.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-compare_nanoseq-8">
Past versions of compare_nanoseq-8.png
</button>
</p>
<div id="fig-compare_nanoseq-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/36b6bf3ba3d288812ba9a108d364dbb52c63f0c2/docs/figure/ecoli.Rmd/compare_nanoseq-8.png" target="_blank">36b6bf3</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-01-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRblas.so
LAPACK: /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.1.1   stringr_1.4.0     tibble_3.1.5      vcfR_1.12.0      
 [5] here_1.0.1        UpSetR_1.4.0      R.utils_2.11.0    R.oo_1.24.0      
 [9] R.methodsS3_1.8.1 dplyr_1.0.7       data.table_1.14.0 ggplot2_3.3.5    
[13] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        ape_5.5           lattice_0.20-44   assertthat_0.2.1 
 [5] rprojroot_2.0.2   digest_0.6.27     utf8_1.2.2        R6_2.5.1         
 [9] plyr_1.8.6        evaluate_0.14     highr_0.9         pillar_1.6.4     
[13] rlang_0.4.12      whisker_0.4       vegan_2.5-7       jquerylib_0.1.4  
[17] Matrix_1.3-2      rmarkdown_2.11    labeling_0.4.2    splines_4.0.5    
[21] pinfsc50_1.2.0    munsell_0.5.0     compiler_4.0.5    httpuv_1.6.3     
[25] xfun_0.22         pkgconfig_2.0.3   mgcv_1.8-35       htmltools_0.5.2  
[29] tidyselect_1.1.1  gridExtra_2.3     memuse_4.2-1      fansi_0.5.0      
[33] permute_0.9-5     viridisLite_0.4.0 crayon_1.4.2      withr_2.4.2      
[37] later_1.3.0       MASS_7.3-53.1     grid_4.0.5        nlme_3.1-152     
[41] jsonlite_1.7.2    gtable_0.3.0      lifecycle_1.0.1   DBI_1.1.1        
[45] git2r_0.28.0      magrittr_2.0.1    scales_1.1.1      stringi_1.7.5    
[49] reshape2_1.4.4    farver_2.1.0      fs_1.5.0          promises_1.2.0.1 
[53] bslib_0.3.0       ellipsis_0.3.2    generics_0.1.1    vctrs_0.3.8      
[57] tools_4.0.5       glue_1.4.2        purrr_0.3.4       parallel_4.0.5   
[61] fastmap_1.1.0     yaml_2.2.1        colorspace_2.0-0  cluster_2.1.2    
[65] knitr_1.33        sass_0.4.0       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
