<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Human spike-in experiments</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rare mutation detection</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="model.html">Model</a>
</li>
<li>
  <a href="ecoli.html">E coli BL21 results</a>
</li>
<li>
  <a href="ecoli_K12.html">E coli K12 results</a>
</li>
<li>
  <a href="compare_MGI_Illumina.html">Illumina vs. MGI</a>
</li>
<li>
  <a href="insilico_mixtures.html">in silico mixtures</a>
</li>
<li>
  <a href="ecoli_spikeins.html">E coli spike-ins</a>
</li>
<li>
  <a href="downsampling.html">Downsampling</a>
</li>
<li>
  <a href="human_mixtures.html">Human mixtures</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Human spike-in experiments</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>G000204_duplex/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210916code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210916code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210916)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontree0c24851a5270327aaf5ec9d8bde3625fa2eb3d29targetblank0c24851a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mcmero/rare-mutation-detection/tree/0c24851a5270327aaf5ec9d8bde3625fa2eb3d29" target="_blank">0c24851</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontree0c24851a5270327aaf5ec9d8bde3625fa2eb3d29targetblank0c24851a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mcmero/rare-mutation-detection/tree/0c24851a5270327aaf5ec9d8bde3625fa2eb3d29" target="_blank">0c24851</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/cache/
    Ignored:    data/.DS_Store
    Ignored:    scripts/

Untracked files:
    Untracked:  ._.DS_Store
    Untracked:  ._rare-mutation-detection.Rproj
    Untracked:  DOCNAME
    Untracked:  analysis/._.DS_Store
    Untracked:  analysis/._ecoli_spikeins.Rmd
    Untracked:  analysis/calc_nanoseq_metrics.Rmd
    Untracked:  data/._.DS_Store
    Untracked:  data/._metrics.rds
    Untracked:  data/ecoli/
    Untracked:  data/ecoli_k12_metrics.rds
    Untracked:  data/human_mixture_capture_region.bed
    Untracked:  data/human_mixture_refs
    Untracked:  data/human_mixture_vars
    Untracked:  data/metadata/
    Untracked:  data/metrics_efficiency_nossc.rds
    Untracked:  data/metrics_spikeins.rds
    Untracked:  data/mixtures
    Untracked:  data/natasha_anstee_metrics.rds
    Untracked:  data/ref/
    Untracked:  drop_out_rate.pdf
    Untracked:  efficiency.pdf
    Untracked:  prototype_code/
    Untracked:  stats.csv

Unstaged changes:
    Modified:   analysis/model.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/human_mixtures.Rmd</code>) and
HTML (<code>docs/human_mixtures.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/0c24851a5270327aaf5ec9d8bde3625fa2eb3d29/analysis/human_mixtures.Rmd" target="_blank">0c24851</a>
</td>
<td>
mcmero
</td>
<td>
2023-08-08
</td>
<td>
Added human mixture variant analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Human samples are duplicate of 1% spike-in of 8393 (son of Chinese
ancestry HG-005) in 8391 (son of Eastern European Ashkenazi Jewish
ancestry HG-0020). <a
href="https://www.coriell.org/1/NIGMS/Collections/NIST-Reference-Materials">Reference</a>.</p>
<pre class="r"><code>library(ggplot2)
library(data.table)
library(dplyr)
library(here)
library(tibble)
library(stringr)
library(Rsamtools)
library(GenomicRanges)
library(seqinr)
library(parallel)
library(readxl)
library(patchwork)
library(RColorBrewer)
library(UpSetR)
library(vcfR)
library(R.utils)
library(knitr)</code></pre>
<pre class="r"><code>source(here(&#39;code/load_data.R&#39;))
source(here(&#39;code/plot.R&#39;))
source(here(&#39;code/efficiency_nanoseq_functions.R&#39;))</code></pre>
<pre class="r"><code>variant_dir &lt;- here(&#39;data/human_mixture_vars&#39;)
region_bed &lt;- here(&#39;data/human_mixture_capture_region.bed&#39;)
hg002_var_file &lt;- here(&#39;data/human_mixture_refs/HG002_GRCh38_1_22_v4.2.1_benchmark.vcf.gz&#39;)
hg005_var_file &lt;- here(&#39;data/human_mixture_refs/HG005_GRCh38_1_22_v4.2.1_benchmark.vcf.gz&#39;)</code></pre>
<pre class="r"><code>sample_names &lt;- c(&#39;Human1pR1&#39;, &#39;Human1pR2&#39;)

# load variant data
var_df &lt;- load_variants(variant_dir, sample_names)
hg002_vars &lt;- read.vcfR(hg002_var_file, verbose = FALSE)
hg005_vars &lt;- read.vcfR(hg005_var_file, verbose = FALSE)

hg002v &lt;- data.frame(hg002_vars@fix)
hg005v &lt;- data.frame(hg005_vars@fix)

# get capture regions
regions &lt;- read.delim(region_bed, sep = &#39;\t&#39;, header = FALSE)</code></pre>
<div id="variant-upset-plot" class="section level2">
<h2>Variant Upset plot</h2>
<p>Here we remove any “N” variant calls and INDELs and compare the
overlaps for on- and off-target variant calls.</p>
<pre class="r"><code># remove any N calls and INDELs
var_df &lt;- filter(var_df, ALT != &quot;N&quot;) %&gt;%
            filter(., (ALT %&gt;% str_split(&quot;&quot;) %&gt;% lapply(., length) %&gt;% unlist) == 1) %&gt;%
            filter(., (REF %&gt;% str_split(&quot;&quot;) %&gt;% lapply(., length) %&gt;% unlist) == 1) %&gt;%
            calculate_vafs_nvc(.)

# get capture region range
grx &lt;- GRanges(seqnames = paste0(&quot;chr&quot;, regions$V1),
               ranges = IRanges(start = regions$V2, end = regions$V3))
vrx &lt;- GRanges(seqnames = var_df$CHROM,
               ranges = IRanges(start = as.numeric(var_df$POS),
                                end = as.numeric(var_df$POS) + 1))
var_df$on_target &lt;- overlapsAny(vrx, grx)

# make upsetplot
ulist &lt;- NULL
for(sample in sample_names) {
    ont_ids &lt;- var_df[var_df$sample %in% sample &amp; var_df$on_target,]$id
    oft_ids &lt;- var_df[var_df$sample %in% sample &amp; !var_df$on_target,]$id
    ulist[[paste(sample, &quot;ontarget&quot;)]] &lt;- ont_ids
    ulist[[paste(sample, &quot;offtarget&quot;)]] &lt;- oft_ids
    
}

upset(fromList(ulist), order.by=&#39;freq&#39;, nsets=4)</code></pre>
<p><img src="figure/human_mixtures.Rmd/variant_filtering-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="variant-allele-frequencies" class="section level2">
<h2>Variant allele frequencies</h2>
<p>Here we plot the allelic frequencies per-replicate in three
plots:</p>
<ul>
<li>all filtered variant calls: no frequency or target filtering</li>
<li>on-target variant calls: only on-target variant calls (in capture
region)</li>
<li>VAF-filtered on-target variant calls: all on-target variants under
&lt;0.3% VAF</li>
</ul>
<pre class="r"><code>ggplot(var_df, aes(VAF)) +
    geom_histogram(binwidth = 0.05) +
    facet_grid(~sample) +
    theme_minimal() +
    ggtitle(&quot;All filtered variant calls&quot;)</code></pre>
<p><img src="figure/human_mixtures.Rmd/vaf_plots-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(var_df[var_df$on_target,], aes(VAF)) +
    geom_histogram(binwidth = 0.05) +
    facet_grid(~sample) +
    theme_minimal() +
    ggtitle(&quot;On-target variant calls&quot;)</code></pre>
<p><img src="figure/human_mixtures.Rmd/vaf_plots-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(var_df[var_df$on_target &amp; var_df$VAF &lt; 0.3,], aes(VAF)) +
    geom_histogram(binwidth = 0.01) +
    facet_grid(~sample) +
    theme_minimal() +
    ggtitle(&quot;On-target variant calls &lt; 0.3 VAF&quot;)</code></pre>
<p><img src="figure/human_mixtures.Rmd/vaf_plots-3.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="variant-comparison" class="section level2">
<h2>Variant comparison</h2>
<p>Given the referene information, we check how many SNPs are present in
the capture area. We also filter out any variants that are common
between the two samples (if the same variant appears in both samples, we
can’t differentiate the calls without some kind of phasing).</p>
<pre class="r"><code># filter out any INDELs
hg005v &lt;- filter(hg005v, (ALT %&gt;% str_split(&quot;&quot;) %&gt;% lapply(., length) %&gt;% unlist) == 1) %&gt;%
          filter(., (REF %&gt;% str_split(&quot;&quot;) %&gt;% lapply(., length) %&gt;% unlist) == 1)

# construct Granges for hg005 SNPs and keep only SNPs in capture area
hg5x &lt;- GRanges(seqnames = hg005v$CHROM,
                ranges = IRanges(start = as.numeric(hg005v$POS),
                                 end = as.numeric(hg005v$POS) + 1),
                variant = hg005v$ALT)
hg5x &lt;- hg5x[overlapsAny(hg5x, grx) %&gt;% suppressWarnings()]

# construct Granges for hg002
hg2x &lt;- GRanges(seqnames = hg002v$CHROM,
                ranges = IRanges(start = as.numeric(hg002v$POS),
                                 end = as.numeric(hg002v$POS) + 1),
                variant = hg002v$ALT)

# check variants that overlap, we will keep these if they call a different base
unique_vars &lt;- hg5x[overlapsAny(hg5x, hg2x)]$variant != hg2x[overlapsAny(hg2x, hg5x)]$variant
hg5x &lt;- c(hg5x[!overlapsAny(hg5x, hg2x)], hg5x[overlapsAny(hg5x, hg2x)][unique_vars])

kable(hg5x)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">width</th>
<th align="left">strand</th>
<th align="left">variant</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr1</td>
<td align="right">114714012</td>
<td align="right">114714013</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">G</td>
</tr>
<tr class="even">
<td align="left">chr3</td>
<td align="right">128486108</td>
<td align="right">128486109</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">T</td>
</tr>
<tr class="odd">
<td align="left">chr5</td>
<td align="right">177516672</td>
<td align="right">177516673</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">T</td>
</tr>
<tr class="even">
<td align="left">chr9</td>
<td align="right">5069837</td>
<td align="right">5069838</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">A</td>
</tr>
<tr class="odd">
<td align="left">chr10</td>
<td align="right">87970403</td>
<td align="right">87970404</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">T</td>
</tr>
<tr class="even">
<td align="left">chr11</td>
<td align="right">32396399</td>
<td align="right">32396400</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left">chr11</td>
<td align="right">64805130</td>
<td align="right">64805131</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">A</td>
</tr>
<tr class="even">
<td align="left">chr11</td>
<td align="right">64810148</td>
<td align="right">64810149</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left">chr17</td>
<td align="right">7676301</td>
<td align="right">7676302</td>
<td align="right">2</td>
<td align="left">*</td>
<td align="left">T</td>
</tr>
</tbody>
</table>
<pre class="r"><code>kable(var_df[var_df$POS %in% start(hg5x) &amp; var_df$CHROM %in% seqnames(hg5x),])</code></pre>
<table>
<colgroup>
<col width="2%" />
<col width="3%" />
<col width="6%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="3%" />
<col width="4%" />
<col width="15%" />
<col width="7%" />
<col width="20%" />
<col width="6%" />
<col width="9%" />
<col width="6%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">CHROM</th>
<th align="left">POS</th>
<th align="left">ID</th>
<th align="left">REF</th>
<th align="left">ALT</th>
<th align="left">QUAL</th>
<th align="left">FILTER</th>
<th align="left">INFO</th>
<th align="left">FORMAT</th>
<th align="left">Sample1</th>
<th align="left">sample</th>
<th align="left">id</th>
<th align="right">VAF</th>
<th align="left">on_target</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">127</td>
<td align="left">chr5</td>
<td align="left">177516672</td>
<td align="left">NA</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AC=2;AF=0.0104712041885</td>
<td align="left">GT:AC:AF:NC</td>
<td align="left">0:2:0.0104712041885:C=189,T=2,</td>
<td align="left">Human1pR1</td>
<td align="left">chr5_177516672</td>
<td align="right">0.0104712</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">231</td>
<td align="left">chr10</td>
<td align="left">87970403</td>
<td align="left">NA</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AC=2;AF=0.00393700787402</td>
<td align="left">GT:AC:AF:NC</td>
<td align="left">0:2:0.00393700787402:C=506,T=2,</td>
<td align="left">Human1pR1</td>
<td align="left">chr10_87970403</td>
<td align="right">0.0039370</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">260</td>
<td align="left">chr11</td>
<td align="left">32396399</td>
<td align="left">NA</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AC=4;AF=0.0121951219512</td>
<td align="left">GT:AC:AF:NC</td>
<td align="left">0:4:0.0121951219512:C=4,T=324,</td>
<td align="left">Human1pR1</td>
<td align="left">chr11_32396399</td>
<td align="right">0.0121951</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">517</td>
<td align="left">chr3</td>
<td align="left">128486108</td>
<td align="left">NA</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AC=3;AF=0.0260869565217</td>
<td align="left">GT:AC:AF:NC</td>
<td align="left">0:3:0.0260869565217:C=112,T=3,</td>
<td align="left">Human1pR2</td>
<td align="left">chr3_128486108</td>
<td align="right">0.0260870</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">732</td>
<td align="left">chr11</td>
<td align="left">32396399</td>
<td align="left">NA</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">AC=3;AF=0.0125</td>
<td align="left">GT:AC:AF:NC</td>
<td align="left">0:3:0.0125:C=3,T=237,</td>
<td align="left">Human1pR2</td>
<td align="left">chr11_32396399</td>
<td align="right">0.0125000</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.3.0/lib64/R/lib/libRblas.so 
LAPACK: /stornext/System/data/apps/R/R-4.3.0/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Melbourne
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] knitr_1.43           R.utils_2.12.2       R.oo_1.25.0         
 [4] R.methodsS3_1.8.2    vcfR_1.14.0          UpSetR_1.4.0        
 [7] RColorBrewer_1.1-3   patchwork_1.1.2      readxl_1.4.3        
[10] seqinr_4.2-30        Rsamtools_2.16.0     Biostrings_2.68.1   
[13] XVector_0.40.0       GenomicRanges_1.52.0 GenomeInfoDb_1.36.1 
[16] IRanges_2.34.1       S4Vectors_0.38.1     BiocGenerics_0.46.0 
[19] stringr_1.5.0        tibble_3.2.1         here_1.0.1          
[22] dplyr_1.1.2          data.table_1.14.8    ggplot2_3.4.2       
[25] workflowr_1.7.0     

loaded via a namespace (and not attached):
 [1] ade4_1.7-22             tidyselect_1.2.0        viridisLite_0.4.2      
 [4] farver_2.1.1            bitops_1.0-7            fastmap_1.1.1          
 [7] RCurl_1.98-1.12         promises_1.2.0.1        digest_0.6.33          
[10] lifecycle_1.0.3         cluster_2.1.4           processx_3.8.2         
[13] magrittr_2.0.3          compiler_4.3.0          rlang_1.1.1            
[16] sass_0.4.7              tools_4.3.0             utf8_1.2.3             
[19] yaml_2.3.7              labeling_0.4.2          plyr_1.8.8             
[22] BiocParallel_1.34.2     memuse_4.2-3            withr_2.5.0            
[25] grid_4.3.0              fansi_1.0.4             git2r_0.32.0           
[28] colorspace_2.1-0        scales_1.2.1            MASS_7.3-58.4          
[31] cli_3.6.1               rmarkdown_2.23          vegan_2.6-4            
[34] crayon_1.5.2            generics_0.1.3          rstudioapi_0.15.0      
[37] httr_1.4.6              ape_5.7-1               cachem_1.0.8           
[40] zlibbioc_1.46.0         splines_4.3.0           cellranger_1.1.0       
[43] vctrs_0.6.3             Matrix_1.5-4            jsonlite_1.8.7         
[46] callr_3.7.3             jquerylib_0.1.4         glue_1.6.2             
[49] codetools_0.2-19        ps_1.7.5                stringi_1.7.12         
[52] gtable_0.3.3            later_1.3.1             munsell_0.5.0          
[55] pillar_1.9.0            htmltools_0.5.5         GenomeInfoDbData_1.2.10
[58] R6_2.5.1                pinfsc50_1.2.0          rprojroot_2.0.3        
[61] evaluate_0.21           lattice_0.21-8          highr_0.10             
[64] httpuv_1.6.11           bslib_0.5.0             Rcpp_1.0.11            
[67] gridExtra_2.3           nlme_3.1-162            permute_0.9-7          
[70] mgcv_1.8-42             whisker_0.4.1           xfun_0.39              
[73] fs_1.6.3                getPass_0.2-2           pkgconfig_2.0.3        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
