<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Rare mutation detection model</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rare mutation detection</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="model.html">Model</a>
</li>
<li>
  <a href="ecoli.html">E coli BL21 results</a>
</li>
<li>
  <a href="ecoli_K12.html">E coli K12 results</a>
</li>
<li>
  <a href="compare_MGI_Illumina.html">Illumina vs. MGI</a>
</li>
<li>
  <a href="insilico_mixtures.html">in silico mixtures</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Rare mutation detection model</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rare-mutation-detection/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontreee722dc0ffb9f2e47d0f6535f438d18be5b4026f4targetblanke722dc0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mcmero/rare-mutation-detection/tree/e722dc0ffb9f2e47d0f6535f438d18be5b4026f4" target="_blank">e722dc0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommcmeroraremutationdetectiontreee722dc0ffb9f2e47d0f6535f438d18be5b4026f4targetblanke722dc0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mcmero/rare-mutation-detection/tree/e722dc0ffb9f2e47d0f6535f438d18be5b4026f4" target="_blank">e722dc0</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    scripts/

Untracked files:
    Untracked:  ._.DS_Store
    Untracked:  DOCNAME
    Untracked:  analysis/._.DS_Store
    Untracked:  analysis/cache/
    Untracked:  analysis/calc_nanoseq_metrics.Rmd
    Untracked:  data/._metrics.rds
    Untracked:  data/ecoli/
    Untracked:  data/metadata/
    Untracked:  data/metrics_efficiency_nossc.rds
    Untracked:  data/mixtures
    Untracked:  data/ref/
    Untracked:  prototype_code/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/model.Rmd</code>) and HTML (<code>docs/model.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/e722dc0ffb9f2e47d0f6535f438d18be5b4026f4/analysis/model.Rmd" target="_blank">e722dc0</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-31
</td>
<td>
Fix typo, minor code update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/b5242382e07051bba0914bd6e3906623d0644a64/docs/model.html" target="_blank">b524238</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/afd79e57f54101b6546d69af08f398f1d12dc29e/analysis/model.Rmd" target="_blank">afd79e5</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-26
</td>
<td>
Added revised model, in silico mixtures redone with 1 supporting read, added input cell estimates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/0b0a0ddc024ce75022175dee5b3fd468ad0da894/docs/model.html" target="_blank">0b0a0dd</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/3c84b8f379002b262e0b511267279204f655f1aa/analysis/model.Rmd" target="_blank">3c84b8f</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-12
</td>
<td>
update model for haploid genome
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/0e464647426b6b69aaf40bbc707efb2d6fb5d315/docs/model.html" target="_blank">0e46464</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/f5ca0c02f999dabd8bd8c330159e99063cca2e00/analysis/model.Rmd" target="_blank">f5ca0c0</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-04-11
</td>
<td>
added mixture experiment calculation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/model.html" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/7cfe3e492e9c8473cd39878ceacf3e0d256ef992/analysis/model.Rmd" target="_blank">7cfe3e4</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
<td>
Updated/simplified model
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/e5ed9a72a7c15c0b9078b6ae9c7d9b3f2376d27c/docs/model.html" target="_blank">e5ed9a7</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/5a3c6b14c160037eb2498bc9f1cdd069bca68630/docs/model.html" target="_blank">5a3c6b1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/cbb488669c3100da94c593d291e1b2cb06e1f2d9/docs/model.html" target="_blank">cbb4886</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cf87608a18d2617db10636d202913a1cdf9130e6/analysis/model.Rmd" target="_blank">cf87608</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
Added equation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5cb4f4dd62f2466c17aa720887e56605ba9c95c9/analysis/model.Rmd" target="_blank">5cb4f4d</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
Fix probability of selecting mutant fragment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/5cb4f4dd62f2466c17aa720887e56605ba9c95c9/docs/model.html" target="_blank">5cb4f4d</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
Fix probability of selecting mutant fragment
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/2f677c24ba1c15f37208066f5e54186e58037430/docs/model.html" target="_blank">2f677c2</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de62c5c7a87a259f517eaf6b99676351f539ecae/analysis/model.Rmd" target="_blank">de62c5c</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
<ul>
<li>Fix error in probability calculation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/de62c5c7a87a259f517eaf6b99676351f539ecae/docs/model.html" target="_blank">de62c5c</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
<td>
<ul>
<li>Fix error in probability calculation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/b680ba04eb31c2ee8ff74999358db02cf37df4a8/docs/model.html" target="_blank">b680ba0</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/ebb9d74647432ec0b35bec27ab7245915b35caa3/analysis/model.Rmd" target="_blank">ebb9d74</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-17
</td>
<td>
initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mcmero/rare-mutation-detection/ebb9d74647432ec0b35bec27ab7245915b35caa3/docs/model.html" target="_blank">ebb9d74</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-17
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div></li>
</ul></li>
</ul>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ggplot2)</code></pre>
<div id="model" class="section level1">
<h1>Model</h1>
<p>Q: how many cells (or “cell equivalents”) do we need to sequence to detect variants down to a VAF of X%</p>
<div id="assumptions" class="section level2">
<h2>Assumptions</h2>
<ul>
<li>We consider a haploid genome without SCNAs</li>
<li>We’re not considering DNA extraction efficiency or ligation efficiency</li>
</ul>
</div>
<div id="probability-of-sequencing-a-mutant-cell" class="section level2">
<h2>Probability of sequencing a mutant cell</h2>
<p>The probability of selecting a cell with a given variant from a pool of <span class="math inline">\(n\)</span> cells is the variant allele frequency (VAF) <span class="math inline">\(v\)</span>. This indicates that a VAF of 0.01 means 1 cell in 100 is a mutant (<span class="math inline">\(nv\)</span>).</p>
<p>We assume the probability of selecting a mutant cell is binomially distributed. We want to know the probability of selecting at least one mutant:</p>
<p><span class="math inline">\(P(Bin(n, v)) &gt; 0)\)</span> = 0.95</p>
<p>This is equivalent to:</p>
<p><span class="math inline">\(P(Bin(n, v)) = 0)\)</span> = 0.05</p>
<p>Where <span class="math inline">\(n\)</span> is the number of sequenced cells.</p>
<p>Let’s consider how many input cells (<span class="math inline">\(n\)</span>) are required to select at least one mutant cell at 1% VAF (95% confidence).</p>
<p>We’ll consider 100 cell increments from <span class="math inline">\(n = \{100, 200..2000\}\)</span>. Using these values, we can plot the probability selecting of sequencing at least one cell.</p>
<pre class="r"><code>v = 0.01
n = seq(100, 2000, 100)

vafs &lt;- data.frame(vaf=v,
                   p=pbinom(0, n, v),
                   input_cells=n,
                   mutant_cells=(n * v))

ggplot(vafs, aes(input_cells, p)) +
    geom_point() +
    theme_bw() +
    geom_hline(yintercept=0.05, alpha=0.4)</code></pre>
<p><img src="figure/model.Rmd/cells_plot-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cells_plot-1">
Past versions of cells_plot-1.png
</button>
</p>
<div id="fig-cells_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/0b0a0ddc024ce75022175dee5b3fd468ad0da894/docs/figure/model.Rmd/cells_plot-1.png" target="_blank">0b0a0dd</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/figure/model.Rmd/cells_plot-1.png" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can also plot this as mutant cells instead of VAF:</p>
<pre class="r"><code>ggplot(vafs, aes(mutant_cells, p)) +
    geom_point() +
    theme_bw() +
    geom_hline(yintercept=0.05, alpha=0.4)</code></pre>
<p><img src="figure/model.Rmd/mutant_cells_plot-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mutant_cells_plot-1">
Past versions of mutant_cells_plot-1.png
</button>
</p>
<div id="fig-mutant_cells_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/0b0a0ddc024ce75022175dee5b3fd468ad0da894/docs/figure/model.Rmd/mutant_cells_plot-1.png" target="_blank">0b0a0dd</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/figure/model.Rmd/mutant_cells_plot-1.png" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5a3c6b14c160037eb2498bc9f1cdd069bca68630/docs/figure/model.Rmd/mutant_cells_plot-1.png" target="_blank">5a3c6b1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5cb4f4dd62f2466c17aa720887e56605ba9c95c9/docs/figure/model.Rmd/mutant_cells_plot-1.png" target="_blank">5cb4f4d</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de62c5c7a87a259f517eaf6b99676351f539ecae/docs/figure/model.Rmd/mutant_cells_plot-1.png" target="_blank">de62c5c</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This shows that for a VAF target of 1%, we should sequence at least 300 cells (3 are mutants).</p>
<pre class="r"><code>deviation &lt;- abs(0.05 - vafs$p)
print(vafs[which(deviation == min(deviation)),])</code></pre>
<pre><code>   vaf          p input_cells mutant_cells
3 0.01 0.04904089         300            3</code></pre>
</div>
<div id="varying-the-vaf" class="section level2">
<h2>Varying the VAF</h2>
<p>If we select 1000 input cells, what’s the lowest VAF we can sequence down to, where <span class="math inline">\(v = {0.0002, 0.0004..0.01}\)</span>.</p>
<pre class="r"><code>v = seq(0.0002, 0.01, 0.0002)
n = 1000

vafs &lt;- data.frame(vaf=v,
                   p=pbinom(0, n, v),
                   input_cells=n,
                   mutant_cells=(n * v))

ggplot(vafs, aes(vaf, p)) +
    geom_point() +
    theme_bw() +
    geom_hline(yintercept=0.05, alpha=0.4)</code></pre>
<p><img src="figure/model.Rmd/varying_vaf-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-varying_vaf-1">
Past versions of varying_vaf-1.png
</button>
</p>
<div id="fig-varying_vaf-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/0b0a0ddc024ce75022175dee5b3fd468ad0da894/docs/figure/model.Rmd/varying_vaf-1.png" target="_blank">0b0a0dd</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/figure/model.Rmd/varying_vaf-1.png" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can sequence down to approx 0.3% VAF.</p>
<pre class="r"><code>deviation &lt;- abs(0.05 - vafs$p)
print(vafs[which(deviation == min(deviation)),])</code></pre>
<pre><code>     vaf          p input_cells mutant_cells
15 0.003 0.04956308        1000            3</code></pre>
<p>We can look at the relationship between input cells and allele frequency:</p>
<pre class="r"><code>cells_vs_vaf = NULL
n = seq(100, 10000, 100)
V = seq(0.001, 0.01, 0.001)
for (v in V) {
    toadd &lt;- data.frame(
        vaf=as.factor(v),
        p=pbinom(0, n, v),
        total_cells=n
    )
    cells_vs_vaf &lt;- rbind(cells_vs_vaf, toadd)
}

ggplot(cells_vs_vaf, aes(total_cells, p, colour=vaf)) +
    geom_line() +
    theme_bw() +
    theme(legend.position = &#39;bottom&#39;) +
    geom_hline(yintercept=0.05, alpha=0.4)</code></pre>
<p><img src="figure/model.Rmd/cells_vs_vaf_plot-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cells_vs_vaf_plot-1">
Past versions of cells_vs_vaf_plot-1.png
</button>
</p>
<div id="fig-cells_vs_vaf_plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/0b0a0ddc024ce75022175dee5b3fd468ad0da894/docs/figure/model.Rmd/cells_vs_vaf_plot-1.png" target="_blank">0b0a0dd</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/figure/model.Rmd/cells_vs_vaf_plot-1.png" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5a3c6b14c160037eb2498bc9f1cdd069bca68630/docs/figure/model.Rmd/cells_vs_vaf_plot-1.png" target="_blank">5a3c6b1</a>
</td>
<td>
Marek Cmero
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/5cb4f4dd62f2466c17aa720887e56605ba9c95c9/docs/figure/model.Rmd/cells_vs_vaf_plot-1.png" target="_blank">5cb4f4d</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/de62c5c7a87a259f517eaf6b99676351f539ecae/docs/figure/model.Rmd/cells_vs_vaf_plot-1.png" target="_blank">de62c5c</a>
</td>
<td>
mcmero
</td>
<td>
2021-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can define an equation based on the binomial probability calculation, of obtaining the number of cells to sequence, to be 95% confident of sequencing the mutation based on the target VAF:</p>
<p><span class="math inline">\((1 – v) ^ n = 0.05\)</span></p>
</div>
<div id="coverage" class="section level2">
<h2>Coverage</h2>
<p>The above thought experiment won’t tell us the probability of sequencing the variant, as sequencing involves a whole host of other factors. Coverage is one of the most important. Given that approximately 30x raw coverage is required to yield 1x of high quality duplex coverage, we can look at the relationship between coverage and VAF (need the variant to be at least 1x to detect).</p>
<p>Here we can see that at VAF = 0.01 (line), we need to sequence to approx 3000x, and this gets exponentially deeper as we go down in target VAF.</p>
<pre class="r"><code>v = seq(0.001, 0.1, 0.001)
ccov &lt;- data.frame(d = 30 / v,
                   v = v)
ggplot(ccov, aes(d, v)) +
    geom_line() +
    theme_bw() +
    geom_hline(yintercept=0.01, alpha=0.4)</code></pre>
<p><img src="figure/model.Rmd/coverage-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-coverage-1">
Past versions of coverage-1.png
</button>
</p>
<div id="fig-coverage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/cde303ec36dbeac572d8071b93f6cb506bfc1b81/docs/figure/model.Rmd/coverage-1.png" target="_blank">cde303e</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-02-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mixture-experiments" class="section level2">
<h2>Mixture experiments</h2>
<p>How much total coverage will we need, in order to detect a mixture of <span class="math inline">\(X\%?\)</span> We define this as successfully calling at least 50% of the variants at VAF <span class="math inline">\(v = \{0.1, 0.01, 0.001\}\)</span> with at least one supporting duplex read. We assume our duplex yield is 30 (i.e., 30x coverage yields 1x duplex coverage).</p>
<p>Our binomial calculation takes the form of:</p>
<p><span class="math inline">\(P(Bin(n, v)) = 0)\)</span> = 0.5 where <span class="math inline">\(n =\)</span> duplex coverage.</p>
<pre class="r"><code>df &lt;- NULL
min_reads &lt;- 1
duplex_yield &lt;- 30
duplex_seq_range &lt;- seq(1, 10000, 1)
fraction_we_can_miss &lt;- 0.5

for(vaf in c(0.1, 0.01, 0.001)) {
    tmp &lt;- data.frame(
        pval=pbinom(min_reads - 1, duplex_seq_range, vaf),
        vaf=vaf,
        duplex_cov=duplex_seq_range,
        total_cov=duplex_seq_range * duplex_yield
    )
    tmp &lt;- tmp[tmp$pval &lt; fraction_we_can_miss,]
    tmp &lt;- tmp[tmp$pval == max(tmp$pval),]
    df &lt;- rbind(df, tmp)
}

print(df)</code></pre>
<pre><code>         pval   vaf duplex_cov total_cov
7   0.4782969 0.100          7       210
69  0.4998370 0.010         69      2070
693 0.4999002 0.001        693     20790</code></pre>
</div>
<div id="revised-model" class="section level2">
<h2>Revised model</h2>
<p>Attempting to combine both coverage and cell input, we can take the following formula:</p>
<p><span class="math inline">\((1 - v) ^ {(n * c * d)} = p\)</span></p>
<p>Where <span class="math inline">\(v\)</span> = target VAF, <span class="math inline">\(n\)</span> = number of input cells, <span class="math inline">\(c\)</span> = coverage per input cell, <span class="math inline">\(d\)</span> = duplex efficiency and <span class="math inline">\(p\)</span> = probability of missing a variant.</p>
<p>We set:</p>
<ul>
<li><span class="math inline">\(c = 10\)</span> (coverage we are aiming for for each genome equivalent)</li>
<li><span class="math inline">\(d = 0.057\)</span> (average efficiency for NanoSeq MB2)</li>
<li><span class="math inline">\(p = 0.05\)</span> (fraction of variants we are okay to miss)</li>
<li><span class="math inline">\(v = \{0.001, 0.01, 0.05, 0.1\}\)</span> (target VAFs)</li>
<li><span class="math inline">\(n = \{100, 200..10000\}\)</span> (range of input cells)</li>
</ul>
<p>We also need to check whether the number of genome equivalents we take forward for sequencing exceeds the lane capacity (600GB). Our sequenced bases will be:</p>
<p><span class="math inline">\(b = g * c * n\)</span> where <span class="math inline">\(g\)</span> = genome size.</p>
<pre class="r"><code>lane_cap &lt;- 6e11
g &lt;- 5e6
pmiss &lt;- 0.05
c &lt;- 10
e &lt;- 0.057

vafs &lt;- c(0.001, 0.01, 0.05, 0.1)
n &lt;- seq(100, 10000, 100)

# calculate duplex coverage
dcov &lt;- n * c * e

# calculate total sequenced bases
bases &lt;- g * c * n

probs &lt;- NULL
min_cells &lt;- NULL
for(vaf in vafs) {
    prob &lt;- data.frame(ncells = n,
                        dcov = dcov,
                        p = pbinom(0, dcov, vaf),
                        vaf = vaf,
                        bases = bases,
                        within_capacity = bases &lt; lane_cap)
    
    tmp &lt;- prob[prob$p &lt; pmiss,]
    tmp &lt;- tmp[tmp$p == max(tmp$p),]
    min_cells &lt;- rbind(tmp, min_cells)

    probs &lt;- rbind(probs, prob)
}

print(min_cells)</code></pre>
<pre><code>   ncells dcov           p   vaf    bases within_capacity
1     100   57 0.002465035 0.100 5.00e+09            TRUE
2     200  114 0.002887294 0.050 1.00e+10            TRUE
6     600  342 0.032154114 0.010 3.00e+10            TRUE
53   5300 3021 0.048678808 0.001 2.65e+11            TRUE</code></pre>
<pre class="r"><code>ggplot(probs, aes(ncells, p, colour = factor(vaf))) +
    geom_point() +
    theme_minimal() +
    geom_hline(yintercept = pmiss, alpha=0.4) +
    scale_color_brewer(palette = &#39;Dark2&#39;)</code></pre>
<p><img src="figure/model.Rmd/unnamed-chunk-1-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mcmero/rare-mutation-detection/blob/b5242382e07051bba0914bd6e3906623d0644a64/docs/figure/model.Rmd/unnamed-chunk-1-1.png" target="_blank">b524238</a>
</td>
<td>
Marek Cmero
</td>
<td>
2022-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRblas.so
LAPACK: /stornext/System/data/apps/R/R-4.0.5/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.5   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7         RColorBrewer_1.1-2 highr_0.9          pillar_1.6.4      
 [5] compiler_4.0.5     bslib_0.3.0        later_1.3.0        jquerylib_0.1.4   
 [9] git2r_0.28.0       tools_4.0.5        digest_0.6.27      jsonlite_1.7.2    
[13] evaluate_0.14      lifecycle_1.0.1    tibble_3.1.5       gtable_0.3.0      
[17] pkgconfig_2.0.3    rlang_0.4.12       DBI_1.1.1          yaml_2.2.1        
[21] xfun_0.22          fastmap_1.1.0      withr_2.4.2        dplyr_1.0.7       
[25] stringr_1.4.0      knitr_1.33         generics_0.1.1     fs_1.5.0          
[29] vctrs_0.3.8        sass_0.4.0         tidyselect_1.1.1   rprojroot_2.0.2   
[33] grid_4.0.5         glue_1.4.2         R6_2.5.1           fansi_0.5.0       
[37] rmarkdown_2.11     farver_2.1.0       purrr_0.3.4        magrittr_2.0.1    
[41] whisker_0.4        scales_1.1.1       promises_1.2.0.1   ellipsis_0.3.2    
[45] htmltools_0.5.2    assertthat_0.2.1   colorspace_2.0-0   httpuv_1.6.3      
[49] labeling_0.4.2     utf8_1.2.2         stringi_1.7.5      munsell_0.5.0     
[53] crayon_1.4.2      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
