---
title: "E. coli K12 analysis"
output:
  workflowr::wflow_html:
      code_folding: hide
---

# Metrics for E. coli K12 data

## MultiQC reports:
* [AGRF_CAGRF22029764_HJK2GDSX3 MultiQC pre-duplex](AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report.html)
* [AGRF_CAGRF22029764_HJK2GDSX3 MultiQC duplex](AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report_consensus.html)
* [AGRF_CAGRF22029764_HJK2GDSX3 MultiQC duplex without SSCs](AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report_consensus_nossc.html)

```{r setup, include=FALSE}
DOCNAME = "Analyse E. coli K12 results"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      cache.lazy     = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 7,
                      fig.height     = 5,
                      dev            = c("png"),
                      message        = FALSE,
                      warning        = FALSE)
```


```{r libraries, cache=FALSE, message=FALSE}
library(ggplot2)
library(data.table)
library(dplyr)
library(here)
library(tibble)
library(stringr)
library(Rsamtools)
library(GenomicRanges)
library(seqinr)
library(parallel)
library(readxl)
library(patchwork)
library(RColorBrewer)
library(UpSetR)
library(vcfR)
```


```{r source}
source(here('code/load_data.R'))
source(here('code/plot.R'))
source(here('code/efficiency_nanoseq_functions.R'))
```


```{r variables}
# Ecoli genome max size
# genome_max <- 4528118
genome_max <- c('2e914854fabb46b9_1' = 4661751,
                '2e914854fabb46b9_2' = 67365)
cores = 8
```


```{r paths}
genomeFile <- here('data/ref/Escherichia_coli_ATCC_10798.fasta')
rinfo_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/read_info')
markdup_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/mark_duplicates')
qualimap_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/qualimap')
qualimap_cons_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/consensus/qualimap')
qualimap_cons_nossc_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/consensus/qualimap_nossc')
metadata_file <- here('data/metadata/NovaSeq data E coli.xlsx')
variant_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/variants')
variant_nossc_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/variants_nossc')
```


```{r calculate_metrics, cache=FALSE}
sample_names <- list.files(rinfo_dir) %>%
                str_split('\\.txt.gz') %>%
                lapply(., dplyr::first) %>%
                unlist() %>%
                str_split('_') %>%
                lapply(., head, 2) %>%
                lapply(., paste, collapse='-') %>%
                unlist()

# load variant data
var_df <- load_variants(variant_dir, sample_names)
var_df_nossc <- load_variants(variant_nossc_dir, sample_names[-9])

# load and fetch duplicate rate from MarkDuplicates output
mdup <- load_markdup_data(markdup_dir, sample_names)

# get mean coverage for pre and post-consensus reads
qmap_cov <- get_qmap_coverage(qualimap_dir, sample_names)
qmap_cons_cov <- get_qmap_coverage(qualimap_cons_dir, sample_names)
qmap_cons_cov_nossc <- get_qmap_coverage(qualimap_cons_nossc_dir, sample_names[-9])

# # uncomment below to calculate metrics
# # calculate metrics for nanoseq
# rlen <- 151; skips <- 5
# metrics_nano <- calc_metrics_new_rbs(rinfo_dir, pattern = 'Nano', cores = cores)
# 
# # calculate metrics for xGen
# rlen <- 151; skips <- 8
# metrics_xgen <- calc_metrics_new_rbs(rinfo_dir, pattern = 'xGEN', cores = cores)
# 
# metrics <- c(metrics_nano, metrics_xgen) %>% bind_rows()
# metrics$duplicate_rate <- as.numeric(mdup)
# metrics$duplex_coverage_ratio <- qmap_cov$coverage / qmap_cons_cov$coverage
# metrics$duplex_coverage_ratio[qmap_cons_cov$coverage < 1] <- 0 # fix when < 1 duplex cov
# metrics$sample <- gsub('-HJK2GDSX3', '', sample_names)

# cache metrics object
# saveRDS(metrics, file = here('data/ecoli_k12_metrics.rds'))
metrics <- readRDS(here('data/ecoli_k12_metrics.rds'))
metrics$single_family_fraction <- metrics$single_families / metrics$total_families

# load metadata
metadata <- read_excel(metadata_file)
metadata$`sample name` <- gsub('_', '-', metadata$`sample name`)

# prepare for plotting
mm <- data.frame(melt(metrics))
mm$protocol <- 'NanoSeq'
mm$protocol[grep('xGEN', mm$sample)] <- 'xGen'

mm <- inner_join(mm, metadata, by=c('sample' = 'sample name'))
colnames(mm)[2] <- 'metric'
mm$nuclease <- paste(mm$`Mung bean unit`, mm$`S1 unit`, sep='+')
```

## Metric comparison plots

### Duplicate rate

Fraction of duplicate reads calculated by [Picard's MarkDuplicates](https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-). This is based on barcode-aware aligned duplicates mapping to the same 5' positions for both read pairs. The [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq) states the optimal empirical duplicate rate is 75-76\% (marked in the plot).

```{r duplicate_rate}
metric <- 'duplicate_rate'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = 0.81, alpha = 0.4)  +
        ggtitle(metric)
```

### Fraction of singleton reads

Shows the number of single-read families divided by the total number of reads. As suggested by [Stoler et al. 2016](https://doi.org/10.1186/s13059-016-1039-4), this metric can server as a proxy for error rate, as (uncorrected) barcode mismatches will manifest as single-read families. The lower the fraction of singletons, the better.

```{r singletons}
metric <- 'frac_singletons'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)
```

### Single family fraction

Similar to traction of singletons, this is the number of single read families, divided by the total families.

```{r}
metric <- 'single_family_fraction'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)
```


### Drop-out rate

This is the same calculation as F-EFF in the [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq):

"This shows the fraction of read bundles missing one of the two original strands beyond what would be expected under random sampling (assuming a binomial process). Good values are between 0.10-0.30, and larger values are likely due to DNA damage such as modified bases or internal nicks that prevent amplification of one of the two strands. Larger values do not impact the quality of the results, just reduce the efficiency of the protocol."

This is similar to the singleton fraction, but taking into account loss of pairs due to sampling. The optimal range is shown by the lines.

```{r drop_out_rate}
metric <- 'drop_out_rate'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.1, 0.3), alpha = 0.4)  +
        ggtitle(metric)
```


### Efficiency

Efficiency is the number of duplex bases divided by the number of sequenced bases. According the [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq), this value is maximised at ~0.07 when duplicate rates and strand drop-outs are optimal.

```{r efficiency}
metric <- 'efficiency'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.07), alpha = 0.4)  +
        ggtitle(metric)
```

### GC deviation

GC deviation is the absolute difference between `GC_BOTH` and `GC_SINGLE` calculated by the  [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq). The lower this deviation, the better.

"GC_BOTH and GC_SINGLE: the GC content of RBs with both strands and with just one strand. The two values should be similar between them and similar to the genome average. If there are large deviations that is possibly due to biases during PCR amplification. If GC_BOTH is substantially larger than GC_SINGLE, DNA denaturation before dilution may have taken place."


```{r gc_deviation}
metric <- 'gc_deviation'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)
```


### Duplex Coverage ratio

The mean sequence (pre-duplex) coverage divided by mean duplex coverage. Indicates the yield of how much duplex coverage we get at each sample's sequence coverage. [Abascal et al.](https://doi.org/10.1038/s41586-021-03477-4) report that their yield was approximately 30x (marked on the plot).


```{r duplex_coverage_ratio}
metric <- 'duplex_coverage_ratio'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = 30, alpha = 0.4)  +
        ggtitle(metric)
```


## Family statistics

Comparison of family pair sizes between samples (these are calculated from total reads of paired AB and BA families).

```{r family_sizes, fig.width = 8, fig.height = 5}
ggplot(mm[mm$metric %like% 'family', ], aes(value, sample, colour = metric)) +
        geom_point() +
        coord_trans(x='log2') +
        scale_x_continuous(breaks=seq(0, 94, 8)) +
        theme(axis.text.x = element_text(size=5)) +
        theme_bw() +
        ggtitle('Family pair sizes')
```


The following plot shows:

- families_gt1: number of family pairs where at least one family (AB or BA) has > 1 reads.
- single_families: number of single-read families.
- paired_families: number of family pairs where both families (AB and BA) have > 0 reads.
- paired_and_gt1: number of family pairs where both families (AB and BA) have > 1 reads.

```{r family_stats, fig.width = 9, fig.height = 6}
tmp <- data.table(mm)[,list(meanval = mean(value), minval = min(value), maxval= max(value)), by=c('sample', 'metric')] %>% data.frame()
tmp <- left_join(mm, tmp, by = c('sample', 'metric'))

ggplot(tmp[tmp$metric %like% 'pair|gt1|single_families', ], aes(sample, meanval, fill = metric)) +
        geom_bar(stat='identity', position='dodge') +
        geom_errorbar( aes(x = sample, ymin = minval, ymax = maxval), position = 'dodge', colour = 'grey') +
        theme_bw() +
        coord_flip() +
        scale_fill_brewer(palette = 'Dark2') +
        theme(legend.position = 'right')
```

## Compare metrics side-by-side

Compare protocols and nucleases directly, the first plot includes the outlier sample and the second removes it.

```{r metrics_comp, fig.width=12, fig.height=4.5}
metric_optimals <- list('duplicate_rate' = 0.81,
                        'frac_singletons' = 0,
                        'drop_out_rate' = c(0.1, 0.3),
                        'efficiency' = 0.07,
                        'gc_deviation' = 0,
                        'duplex_coverage_ratio' = 30)
                       
gg_prot <- list(geom_boxplot(outlier.shape = NA),
                geom_jitter(width = 0.1, size = 2, aes(colour = nuclease, shape = nuclease)),
                theme_bw(),
                theme(legend.position = 'bottom'))
gg_nuc <- list(geom_boxplot(outlier.shape = NA),
               geom_jitter(width = 0.1, size = 2, aes(colour = protocol, shape = protocol)),
               theme_bw(),
               theme(legend.position = 'bottom'))

mmt <- mm
mmt$replicate <- str_split(mmt$sample, 'Rep') %>% lapply(., dplyr::last) %>% unlist() %>% as.numeric()
mmt$sample <- str_split(mmt$sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()

for(metric in names(metric_optimals)) {
    # plot all samples
    threshold <- metric_optimals[metric][[1]]
    tmp <- mmt[mmt$metric %in% metric,]
    p1 <- ggplot(tmp, aes(sample, value)) +
        geom_point() +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = threshold, alpha = 0.4) +
        ggtitle(paste(metric, '(line = optimal)'))
    
    p2 <- ggplot(tmp, aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    p3 <- ggplot(tmp, aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    show(p1 + p2 + p3)
    
    # repeat with removed outlier
    tmp <- mmt[mmt$metric %in% metric & !(mmt$sample %in% 'xGEN-xGEN' & mmt$replicate == 1),]
    p1 <- ggplot(tmp, aes(sample, value)) +
        geom_point() +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = threshold, alpha = 0.4) +
        ggtitle(paste(metric, '(line = optimal)'))
    
    p2 <- ggplot(tmp, aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    p3 <- ggplot(tmp, aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    show(p1 + p2 + p3)
}
```

## Facet summary plots

Facet boxplots by nuclease and protocol to show overall results.

```{r facet_summary, fig.width=9, fig.height=6}
ggplot(mm, aes(protocol, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by protocol')

ggplot(mm, aes(nuclease, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by nuclease')
```

Plots again removing the outlier xGEN rep 1.

```{r facet_summary_remove_outlier, fig.width=9, fig.height=6}
mmo <- mm[mm$sample != 'xGEN-xGENRep1',]
mmo$replicate <- str_split(mmo$sample, 'Rep') %>% lapply(., dplyr::last) %>% unlist() %>% as.numeric()
mmo$sample <- str_split(mmo$sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()

ggplot(mmo, aes(protocol, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by protocol')

ggplot(mmo, aes(nuclease, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by nuclease')
```

Summary plot including separated by all experimental factors.

```{r plot_by_sample, fig.width=16, fig.height=10}
ggplot(mmo, aes(sample, value, colour = protocol, shape = nuclease)) + 
    geom_point() +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90)) +
    facet_wrap(~metric, scales = 'free') +
    scale_colour_brewer(palette = 'Dark2') +
    ggtitle('by protocol')
```


## Statistical test results by protocol

For each metric, take the average of each replicate and perform a two-sided, unpaired T-test between protocols.

```{r tests_by_protocol}
stats <- NULL
metric_names <- unique(mmo$metric) %>% as.character()
for(metric_name in metric_names) {
    nano <- mmo[mmo$metric == metric_name & mmo$protocol == 'NanoSeq',]
    xgen <- mmo[mmo$metric == metric_name & mmo$protocol == 'xGen',]
    nano_vals <- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals <- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest <- t.test(nano_vals, xgen_vals)
    stats <- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```


Rerun tests removing outlier (xGEN rep1). The results are similar.

```{r test_by_protocol_remove_outlier}
stats <- NULL
for(metric_name in metric_names) {
    nano <- mmo[mmo$metric == metric_name & mmo$protocol == 'NanoSeq',]
    xgen <- mmo[mmo$metric == metric_name & mmo$protocol == 'xGen',]
    nano_vals <- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals <- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest <- t.test(nano_vals, xgen_vals)
    stats <- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```


## Two-way ANOVA analysis

We consider a two-way ANOVA, modelling the protocol, Mung Bean Unit and S1 Unit variables, as well as the interaction effect between the units and the protocol. 

```{r tests_by_nuclease}
stats <- NULL
metric_names <- unique(mm$metric) %>% as.character()
for(metric_name in metric_names) {
    x <- mm[mm$metric == metric_name,]
    x$MungBeanUnit <- as.factor(x$`Mung bean unit`)
    x$S1Unit <- as.factor(x$`S1 unit`)
    x <- x[,c('MungBeanUnit', 'S1Unit', 'protocol', 'nuclease', 'value')]
    x_aov <- aov(value ~ MungBeanUnit * protocol + S1Unit * protocol, data = x) %>% summary() %>% dplyr::first()
    stats <- rbind(stats,
                   data.frame(metric = metric_name,
                              variable = rownames(x_aov)[1:5],
                              pvalue = x_aov[['Pr(>F)']][1:5]))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```

We remove the outlier xGEN rep 1 and test again.

```{r tests_by_nuclease_remove_outlier}
stats <- NULL
metric_names <- unique(mmo$metric) %>% as.character()
for(metric_name in metric_names) {
    x <- mmo[mmo$metric == metric_name,]
    x$MungBeanUnit <- as.factor(x$`Mung bean unit`)
    x$S1Unit <- as.factor(x$`S1 unit`)
    x <- x[,c('MungBeanUnit', 'S1Unit', 'protocol', 'nuclease', 'value')]
    x_aov <- aov(value ~ MungBeanUnit * protocol + S1Unit * protocol, data = x) %>% summary() %>% dplyr::first()
    stats <- rbind(stats,
                   data.frame(metric = metric_name,
                              variable = rownames(x_aov)[1:5],
                              pvalue = x_aov[['Pr(>F)']][1:5]))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```


## Relationships between variables

```{r relationships, fig.width=10, fig.height=8}
tmp <- mmo[,c('sample', 'metric', 'value', 'protocol', 'nuclease', 'replicate')]
dm <- reshape2::dcast(mmo, sample + protocol + nuclease + replicate ~ metric)

cols <- c(brewer.pal(5, 'Greens')[2:5],
          brewer.pal(6, 'Blues')[2:6])
names(cols) <- as.factor(dm$sample) %>% levels()

ggplot(dm, aes(frac_singletons, drop_out_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Singletons vs. drop-out rate')

ggplot(dm, aes(efficiency, duplicate_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Efficiency vs. duplicate rate')

ggplot(dm, aes(efficiency, drop_out_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Efficiency vs. drop-out rate')

ggplot(dm, aes(efficiency, duplex_coverage_ratio, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Efficiency vs. duplex coverage ratio')

ggplot(dm, aes(duplicate_rate, duplex_coverage_ratio, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Duplicate rate vs. duplex coverage ratio')
```
Focus on relationship between efficiency, duplicate rate and drop-out rate.

```{r efficiency_comparison, fig.width=14, fig.height=5}
mt <- mm
mt$replicate <- str_split(mt$sample, 'Rep') %>% lapply(., dplyr::last) %>% unlist() %>% as.numeric()
mt$sample <- str_split(mt$sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()

mt <- mt[,c('sample', 'metric', 'value', 'protocol', 'nuclease', 'replicate')]
dm <- reshape2::dcast(mt, sample + protocol + nuclease + replicate ~ metric)

p1 <- ggplot(dm, aes(duplicate_rate, efficiency, colour=protocol, shape=nuclease)) +
    geom_point(size = 3) +
    theme_bw() +
    scale_colour_brewer(palette = 'Dark2') +
    ggtitle('Efficiency vs. duplicate rate')

p2 <- ggplot(dm, aes(drop_out_rate, efficiency, colour=protocol, shape=nuclease)) +
    geom_point(size = 3) +
    theme_bw() +
    scale_colour_brewer(palette = 'Dark2') +
    ggtitle('Efficiency vs. drop-out rate')
show(p1 + p2)
```

## Variant calls

Upset plot showing duplex variant calls. Variants were called in areas with at least 4x coverage with at least 2 supporting reads and a VAF of $\geq2$.

```{r variants, fig.width=12, fig.height=10}
ulist <- NULL
for(sample in sample_names) {
    ids <- var_df[var_df$sample %in% sample,]$id
    if (length(ids) > 0) {
        ulist[[gsub(pattern = '-HJK2GDSX3', replacement = '', sample)]] <- ids
    }
}

upset(fromList(ulist), order.by='freq', nsets=length(sample_names))
```


## Duplex coverage without requiring SSC

The pipeline was run only requiring a single read on each strand. Here we plot the difference in mean coverage. As we would expect, skipping SSC step increases duplex coverage. For some samples with disproportionately higher single-read families (NanoMB-S1), this increases duplex coverage significantly more.

```{r coverage_without_ssc, fig.width=10, fig.height=4}
ccov <- inner_join(qmap_cons_cov,
                   qmap_cons_cov_nossc,
                   by = 'Sample',
                   suffix = c('_ssc', '_nossc')) %>%
          inner_join(., qmap_cov, by = 'Sample')
ccov$sample <- str_split(ccov$Sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()
ccov$duplex_cov_ratio <- ccov$coverage / ccov$coverage_ssc
ccov$duplex_cov_ratio_noscc <- ccov$coverage / ccov$coverage_nossc
ccov <- left_join(ccov, distinct(mmo[,c('sample', 'protocol', 'nuclease')]), by = 'sample')

p1 <- ggplot(ccov, aes(coverage_ssc, coverage_nossc, colour = protocol, shape = nuclease)) +
  geom_point() +
  theme_bw() +
  xlim(0, 550) +
  ylim(0, 550) +
  xlab('with SSC') +
  ylab('without SSC') +
  geom_abline(slope = 1) +
  theme(legend.position = 'left') +
  scale_colour_brewer(palette = 'Dark2') +
  ggtitle('Mean duplex coverage')

p2 <- ggplot(ccov, aes(duplex_cov_ratio, duplex_cov_ratio_noscc, colour = protocol, shape = nuclease)) +
  geom_point() +
  theme_bw() +
  xlim(0, 100) +
  ylim(0, 100) +
  xlab('with SSC') +
  ylab('without SSC') +
  geom_abline(slope = 1) +
  theme(legend.position = 'right') +
  scale_colour_brewer(palette = 'Dark2') +
  ggtitle('Duplex coverage ratio')

p1 + p2
```

## Variant calls without SSC

Here we show the variant calls from the duplex sequences without SSC in the same Upset plot format.

```{r variants_nossc, fig.width=12, fig.height=10}

for(sample in sample_names) {
    ids <- var_df_nossc[var_df_nossc$sample %in% sample,]$id
    if (length(ids) > 0) {
        ulist[[sample]] <- ids
    }
}

upset(fromList(ulist), order.by='freq', nsets=length(sample_names))
```

## Input cells

Estimate the number of input cells using formula $d / e / c = n$ where d = mean duplex coverage, e = duplex efficiency, c = coverage per genome equivalent and n = number of cells.

```{r}
coverage_per_genome <- 10
qmap_cons_cov$Sample <- gsub('-HJK2GDSX3', '', qmap_cons_cov$Sample)
metrics <- inner_join(metrics, qmap_cons_cov, by = c('sample' = 'Sample'))
metrics$estimated_cells <- metrics$coverage / metrics$efficiency / coverage_per_genome

ggplot(metrics[!metrics$sample %in% 'xGEN-xGENRep1',], aes(sample, estimated_cells)) +
    geom_bar(stat = 'identity') + 
    theme_minimal() +
    coord_flip()
```

