---
title: "Compare sequencing metrics from E coli data"
output:
  workflowr::wflow_html:
      code_folding: hide
---

# Compare sequencing metrics from E coli data

These are extra stats that are not available in the MultiQC reports. These reports can be found below:

- [MultiQC pre-duplex](multiqc/multiqc_report.html)
- [MultiQC duplex](multiqc/multiqc_report_consensus.html)

```{r knitr, include = FALSE}
DOCNAME = "Ecoli_data"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 9,
                      fig.height     = 5,
                      dev            = c("png"),
                      message        = FALSE,
                      warning        = FALSE)
```

```{r libraries, cache = FALSE}
library(ggplot2)
library(data.table)
library(dplyr)
library(R.utils)
library(UpSetR)
library(here)
library(vcfR)
library(tibble)
library(stringr)
```

```{r functions}
load_data <- function(fdir, pattern, samples, read_func=read.delim) {
    df <- list.files(
        fdir,
        full.names = TRUE,
        recursive = TRUE,
        pattern = pattern) %>%
    lapply(., read_func)
    
    for (i in seq(length(samples))) {
        df[[i]]$Sample <- samples[i]
    }
    df <- rbindlist(df)
    return(df)
}

extract_std <- function(genome_results) {
    std <- genome_results[grep('std', genome_results$BamQC.report),] %>% 
            strsplit(., split='=') %>%
            last() %>% last() %>%
            gsub(' |X', '', .) %>% as.numeric()
    return(std)
}

load_cov_stats <- function(cov, qualimap_dir, samples) {
    cov_stats <- list.files(
        qualimap_dir,
        full.names = TRUE,
        recursive = TRUE,
        pattern = 'genome_results.txt') %>%
        lapply(., read.delim) %>%
        lapply(., extract_std) %>%
        unlist()
    
    cov_stats <- data.frame(cov_std=cov_stats, Sample=samples)
    cov_stats <- data.table(cov)[,mean(Coverage), by=Sample] %>%
                data.frame() %>% inner_join(., cov_stats, by='Sample')
    colnames(cov_stats)[2] <- 'cov_mean'
    
    return(cov_stats)
}

load_nanoseq_stats <- function(nanoseq_dir) {
    tsvs <- list.files(nanoseq_dir,
                   pattern = '.tsv',
                   full.names = TRUE) %>%
        grep('GC', ., value = TRUE, invert = TRUE) %>%
        lapply(., read.delim) 

    for (i in seq(length(samples))) {
        tsvs[[i]]$Sample <- samples[i]
        tsvs[[i]]$metric <- rownames(tsvs[[i]])
    }
    tsvs <- rbindlist(tsvs)
    colnames(tsvs)[1] <- 'value'
    tsvs <- tsvs[!is.na(tsvs$value)]
    
    # add GC deviation
    tmp <- tsvs[grep('GC', tsvs$metric)]
    gc <- tmp[tmp$metric == 'GC_SINGLE']
    gc$value <- abs(tmp[tmp$metric == 'GC_BOTH']$value - gc$value)
    gc$metric <- 'GC_DEVIATION'
    tsvs <- rbind(tsvs, gc)
    
    # extract protocol and nuclease labels
    tsvs$protocol <- 'NanoSeq'
    tsvs$protocol[grep('Nux', tsvs$Sample)] <- 'xGen'
    tsvs$nuclease <- str_split(tsvs$Sample, regex("N(uxg|an)|_")) %>%
                        lapply(., tail, 2) %>% lapply(., dplyr::first) %>% unlist()
    
    return(tsvs)
}

# plotting functions (for nanoseq stats)
plot_metric <- function(tsvs, metric, title) {
    tmp <- data.frame(tsvs)[grep(metric, tsvs$metric),]
    p <- ggplot(tmp, aes(Sample, value, fill = metric)) +
            geom_histogram(stat = 'identity', position = 'dodge') +
            theme_bw() +
            coord_flip() +
            ggtitle(title)
    return(p)
}

plot_metric_boxplot <- function(tsvs, x, metric, title) {
    tmp <- data.frame(tsvs)[grep(metric, tsvs$metric),]
    p <- ggplot(tmp, aes_string(x, 'value')) +
            geom_boxplot() +
            theme_bw() +
            ggtitle(title)
    return(p)
}
```


```{r load_data}
qualimap_dir <- here('data/ecoli/QC/qualimap/')
qualimap_cons_dir <- here('data/ecoli/QC/consensus/qualimap/')
variant_dir <- here('data/ecoli/variants')
family_size_stats <- here('data/ecoli/family_sizes.txt')
nanoseq_dir <- here('data/nanoseq_results')

samples <- list.files(qualimap_dir)

cov <- load_data(qualimap_dir, 'coverage_across_reference.txt', samples)
ccov <- load_data(qualimap_cons_dir, 'coverage_across_reference.txt', samples)
clip <- load_data(qualimap_dir, 'mapped_reads_clipping_profile', samples)
cclip <- load_data(qualimap_cons_dir, 'mapped_reads_clipping_profile', samples)
vars <- load_data(variant_dir, '.vcf', samples, read.table)
cov_stats <- load_cov_stats(cov, qualimap_dir, samples)
ccov_stats <- load_cov_stats(ccov, qualimap_cons_dir, samples)
fam <- read.delim(family_size_stats, sep='\t')
tsvs <- load_nanoseq_stats(nanoseq_dir)
```

## Coverage boxplot

Using coverage summary data from Qualimap (I assume these are summarised to 10kb windows, though I couldn't find this in the documentation).

```{r coverage_boxplots}
# order by median coverage
median_cov <- data.table(cov)[,median(Coverage), by=Sample]
cov$Sample <- factor(cov$Sample, levels=median_cov[order(median_cov$V1)]$Sample)
ggplot(cov, aes(Coverage, Sample)) + geom_boxplot() + theme_bw() + ggtitle('Pre-duplex coverage')

median_cov <- data.table(ccov)[,median(Coverage), by=Sample]
ccov$Sample <- factor(ccov$Sample, levels=median_cov[order(median_cov$V1)]$Sample)
ggplot(ccov, aes(Coverage, Sample)) + geom_boxplot() + theme_bw() + ggtitle('Duplex coverage')
```

## Coverage standard deviation bar plot

```{r cov_std}
ggplot(melt(cov_stats), aes(value, Sample, fill=variable)) +
    geom_bar(stat='identity', position = 'dodge') +
    theme_bw() +
    ggtitle('Coverage std & mean (pre-duplex)')

ggplot(melt(ccov_stats), aes(value, Sample, fill=variable)) +
    geom_bar(stat='identity', position = 'dodge') +
    theme_bw() +
    ggtitle('Coverage std & mean (duplex)')

cov_stats$cov_cv <- cov_stats$cov_std / cov_stats$cov_mean
ggplot(cov_stats, aes(cov_cv, Sample)) +
    geom_bar(stat='identity', position = 'dodge') +
    theme_bw() +
    ggtitle('Coverage CV (pre-duplex)')

ccov_stats$cov_cv <- ccov_stats$cov_std / ccov_stats$cov_mean
ggplot(ccov_stats, aes(cov_cv, Sample)) +
    geom_bar(stat='identity', position = 'dodge') +
    theme_bw() +
    ggtitle('Coverage CV (duplex)')
```

## Clipping profile

Pre-duplex reads prior to overlap clipping, but post-UMI removal.

```{r clipping_profile}
ggplot(clip, aes(X.Read.position..bp., Clipping.profile)) +
    geom_line() +
    theme_bw() +
    xlab('Read position') +
    facet_wrap(~Sample) +
    ggtitle('Pre-duplex clipping profile')
```

Duplex reads have been clipped to remove read overlap.

```{r clipping_profile_duplex}
ggplot(cclip, aes(X.Read.position..bp., Clipping.profile)) +
    geom_line() +
    theme_bw() +
    xlab('Read position') +
    facet_wrap(~Sample) +
    ggtitle('Duplex clipping profile')
```

## Family size stats (duplex statistics)

- frac_fam_gt1: the fraction of families where family size is greater than one.
- frac_fam_paired: the fraction of paired families (a family on each strand with the same UMI).
- frac_fam_paired_gt1: the fraction of families that are greater than one in size, and also paired.

```{r family_size}
ggplot(fam, aes(len, sample)) +
    geom_bar(stat='identity') +
    theme_bw() +
    ggtitle('Total family count')

mfam <- reshape2::melt(fam[,c('sample', 'min', 'max', 'mean', 'median')])
ggplot(mfam, aes(value, sample, colour=variable)) +
    geom_point() +
    theme_bw() +
    coord_trans(x='log2') +
    scale_x_continuous(breaks=seq(0, 25, 2)) +
    theme(axis.text.x = element_text(size=6)) +
    ggtitle('Family sizes')
    
mfam <- reshape2::melt(fam[,colnames(fam) %like% 'frac|sample'])
ggplot(mfam, aes(value, sample, fill=variable)) +
    geom_bar(stat='identity', position='dodge') +
    theme_bw() +
    ggtitle('Family reads + paired statistics')
```
### Compare single-read fraction  

```{r compare_famsize, fig.width = 6, fig.height = 5}
# extract protocol and nuclease labels
fam$protocol <- 'NanoSeq'
fam$protocol[grep('Nux', fam$sample)] <- 'xGen'
fam$nuclease <- str_split(fam$sample, regex("N(uxg|an)|_")) %>%
                    lapply(., tail, 2) %>% lapply(., dplyr::first) %>% unlist()

mfam <- reshape2::melt(fam[,colnames(fam) %like% 'fam_gt1|sample|protocol|nuc'])
ggplot(mfam, aes(protocol, value)) +
    geom_boxplot() +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    theme_bw() +
    ggtitle('Fraction families with size > 1 by protocol')

ggplot(mfam, aes(nuclease, value)) +
    geom_boxplot() +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    theme_bw() +
    ggtitle('Fraction families with size > 1 by nuclease')
```

## Variants

```{r variant_overlap}
vars$Sample <- strsplit(vars$Sample, '\\.') %>% lapply(., head, 1) %>% unlist()

ulist <- NULL
for(sample in samples) {
    ulist[[sample]] <- vars[vars$Sample %in% sample,]$V2
}

upset(fromList(ulist), order.by='freq', nsets=8)
```
Check how many variants have at least one reference base called.

```{r discordant_variants}
vcfs <- list.files(variant_dir,
                  full.names = TRUE) %>%
        lapply(., read.vcfR, verbose = FALSE) %>%
        lapply(., vcfR2tidy,
               format_fields = c('GT', 'AD', 'RD'),
               verbose = FALSE)

disc <- lapply(vcfs, function(vcf) {
          has_ref <- vcf$gt$gt_RD > 0
          row <- c(sum(has_ref), nrow(vcf$gt))
          return(row)
        })

disc <- data.frame(disc, row.names = c('has_ref_bases','total_variants')) %>% t() %>%
          data.frame(row.names = 1:nrow(.)) %>%
          add_column(sample=samples)

ggplot(melt(disc), aes(value, sample, fill = variable)) +
  geom_histogram(stat = 'identity', position = 'dodge') +
  theme_bw()

gts <- NULL
for(i in 1:length(samples)) {
    vcfs[[i]]$gt$sample <- samples[i]
    gts <- rbind(gts, vcfs[[i]]$gt)
}

ggplot(gts, aes(sample, fill = gt_GT)) +
    geom_histogram(stat = 'count', position = 'dodge') +
    theme_bw() +
    coord_flip() +
    ggtitle('Variant genotype')
```
## Statistics from NanoSeq pipeline

Stats obtained from [NanoSeq](https://github.com/cancerit/NanoSeq) pipeline.

```{r nanoseq_stats}
plot_metric(tsvs, 'READS$', 'Num reads')

plot_metric(tsvs, 'DUPLICATE', 'Duplicate rate (line = optimal)') +
    geom_hline(yintercept = 0.81)

plot_metric(tsvs, 'TOTAL', 'Total RBs (read barcodes)')

plot_metric(tsvs, 'PER', 'Reads per RB')

plot_metric(tsvs, 'OK', 'OK RBs (2 on each strand)')

plot_metric(tsvs, 'F-EFF',
             'F-EFF (drop out fraction, lines show optimal range)') +
    geom_hline(yintercept = c(0.1, 0.3))

plot_metric(tsvs, 'EFFICIENCY','Efficiency (line shows maximised value)') +
    geom_hline(yintercept = 0.07)

plot_metric(tsvs, '(GC_BOTH|GC_SINGLE)', 'GC per strand')
```
### Comparison of key stats

```{r compare_nanoseq, fig.width = 6, fig.height = 5}
# duplicate rate
plot_metric_boxplot(tsvs, 'protocol', 'DUPLICATE', 'Duplicate rate (line = optimal') +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    geom_hline(yintercept = 0.81)

plot_metric_boxplot(tsvs, 'nuclease', 'DUPLICATE', 'Duplicate rate (line = optimal') +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    geom_hline(yintercept = 0.81)

# efficiency
plot_metric_boxplot(tsvs, 'protocol', 'EFFICIENCY', 'Efficiency (line = optimal)') +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    geom_hline(yintercept = 0.07) 

plot_metric_boxplot(tsvs, 'nuclease', 'EFFICIENCY', 'Efficiency (line = optimal)') +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    geom_hline(yintercept = 0.07) 

# drop out rate
plot_metric_boxplot(tsvs, 'protocol', 'F-EFF', 'Strand drop-out fraction (lines = optimal range)') +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease)) +
    geom_hline(yintercept = c(0.1, 0.3))

plot_metric_boxplot(tsvs, 'nuclease', 'F-EFF', 'Strand drop-out fraction (lines = optimal range)') +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol)) +
    geom_hline(yintercept = c(0.1, 0.3))

# GC deviation between strands
plot_metric_boxplot(tsvs, 'protocol', 'GC_DEV', 'GC deviation (both strands vs. one)') +
    geom_jitter(width=0.1, aes(protocol, value, colour = nuclease))

plot_metric_boxplot(tsvs, 'nuclease', 'GC_DEV', 'GC deviation (both strands vs. one)') +
    geom_jitter(width=0.1, aes(nuclease, value, colour = protocol))
```

