---
title: "E. coli K12 analysis"
output:
  workflowr::wflow_html:
      code_folding: hide
---

# Metrics for E. coli K12 data

## Experimental setup

| End-prep | Mung Bean nuclease units | S1 Nuclease units | Protocol | PCR       | Samples | Key |
|-----------|:---------:|:---------:|-----------|-----------|-----------|-----------|
| MB1      |            1             |         0         | NanoSeq  | limited   | 2       | 1+0 |
| MB2      |            2             |         0         | NanoSeq  | limited   | 2       | 2+0 |
| MB3      |            3             |         0         | Nanoseq  | limited   | 2       | 3+0 |
| MB-S1    |            2             |         1         | Nanoseq  | limited   | 2       | 2+1 |
| MB1      |            1             |         0         | xGEN     | limited   | 2       | 1+0 |
| MB2      |            2             |         0         | xGEN     | limited   | 2       | 2+0 |
| MB3      |            3             |         0         | xGEN     | limited   | 2       | 3+0 |
| MB-S1    |            2             |         1         | xGEN     | limited   | 2       | 2+1 |
| xGEN     |            0             |         0         | xGEN     | limited   | 2       | 0+0 |
| xGEN     |            0             |         0         | xGEN     | unlimited | 1       | 0+0 |

## MultiQC reports:

-   [AGRF_CAGRF22029764_HJK2GDSX3 MultiQC pre-duplex](AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report.html)
-   [AGRF_CAGRF22029764_HJK2GDSX3 MultiQC duplex](AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report_consensus.html)
-   [AGRF_CAGRF22029764_HJK2GDSX3 MultiQC duplex without SSCs](AGRF_CAGRF22029764_HJK2GDSX3_multiqc/multiqc_report_consensus_nossc.html)

```{r setup, include=FALSE}
DOCNAME = "Analyse E. coli K12 results"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      cache.lazy     = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 7,
                      fig.height     = 5,
                      dev            = c("png"),
                      message        = FALSE,
                      warning        = FALSE)
```

```{r libraries, cache=FALSE, message=FALSE}
library(ggplot2)
library(data.table)
library(dplyr)
library(here)
library(tibble)
library(stringr)
library(Rsamtools)
library(GenomicRanges)
library(seqinr)
library(parallel)
library(readxl)
library(patchwork)
library(RColorBrewer)
library(UpSetR)
library(vcfR)
```

```{r source}
source(here('code/load_data.R'))
source(here('code/plot.R'))
source(here('code/efficiency_nanoseq_functions.R'))
```

```{r variables}
# Ecoli genome max size
# genome_max <- 4528118
genome_max <- c('2e914854fabb46b9_1' = 4661751,
                '2e914854fabb46b9_2' = 67365)
cores = 8
```

```{r paths}
genomeFile <- here('data/ref/Escherichia_coli_ATCC_10798.fasta')
rinfo_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/read_info')
markdup_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/mark_duplicates')
qualimap_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/qualimap')
qualimap_cons_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/consensus/qualimap')
qualimap_cons_nossc_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/QC/consensus/qualimap_nossc')
metadata_file <- here('data/metadata/NovaSeq data E coli.xlsx')
variant_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/variants')
variant_nossc_dir <- here('data/ecoli/AGRF_CAGRF22029764_HJK2GDSX3/variants_nossc')
variant_raw_dir <- here('data/ecoli/variant_calling')
```

```{r calculate_metrics, cache=FALSE}
sample_names <- list.files(rinfo_dir) %>%
                str_split('\\.txt.gz') %>%
                lapply(., dplyr::first) %>%
                unlist() %>%
                str_split('_') %>%
                lapply(., head, 2) %>%
                lapply(., paste, collapse='-') %>%
                unlist()

# load variant data
var_df <- load_variants(variant_dir, sample_names)
var_df_nossc <- load_variants(variant_nossc_dir, sample_names[-9])
var_df_raw <- tmp <- load_variants(variant_raw_dir, c('xGEN-xGENRep1_default_filter', 'xGEN-xGENRep1_minimal_filter'))

# load and fetch duplicate rate from MarkDuplicates output
mdup <- load_markdup_data(markdup_dir, sample_names)

# get mean coverage for pre and post-consensus reads
qmap_cov <- get_qmap_coverage(qualimap_dir, sample_names)
qmap_cons_cov <- get_qmap_coverage(qualimap_cons_dir, sample_names)
qmap_cons_cov_nossc <- get_qmap_coverage(qualimap_cons_nossc_dir, sample_names[-9])

# # uncomment below to calculate metrics
# # calculate metrics for nanoseq
# rlen <- 151; skips <- 5
# metrics_nano <- calc_metrics_new_rbs(rinfo_dir, pattern = 'Nano', cores = cores)
# 
# # calculate metrics for xGen
# rlen <- 151; skips <- 8
# metrics_xgen <- calc_metrics_new_rbs(rinfo_dir, pattern = 'xGEN', cores = cores)
# 
# metrics <- c(metrics_nano, metrics_xgen) %>% bind_rows()
# metrics$duplicate_rate <- as.numeric(mdup)
# metrics$duplex_coverage_ratio <- qmap_cov$coverage / qmap_cons_cov$coverage
# metrics$duplex_coverage_ratio[qmap_cons_cov$coverage < 1] <- 0 # fix when < 1 duplex cov
# metrics$sample <- gsub('-HJK2GDSX3', '', sample_names)

# cache metrics object
# saveRDS(metrics, file = here('data/ecoli_k12_metrics.rds'))
metrics <- readRDS(here('data/ecoli_k12_metrics.rds'))
metrics$single_family_fraction <- metrics$single_families / metrics$total_families

# load metadata
metadata <- read_excel(metadata_file)
metadata$`sample name` <- gsub('_', '-', metadata$`sample name`)

# prepare for plotting
mm <- data.frame(melt(metrics))
mm$protocol <- 'NanoSeq'
mm$protocol[grep('xGEN', mm$sample)] <- 'xGen'

mm <- inner_join(mm, metadata, by=c('sample' = 'sample name'))
colnames(mm)[2] <- 'metric'
mm$nuclease <- paste(mm$`Mung bean unit`, mm$`S1 unit`, sep='+')
```

## Metric comparison plots

### Duplicate rate

Fraction of duplicate reads calculated by [Picard's MarkDuplicates](https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-). This is based on barcode-aware aligned duplicates mapping to the same 5' positions for both read pairs. The [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq) states the optimal empirical duplicate rate is 75-76% (marked in the plot).

```{r duplicate_rate}
metric <- 'duplicate_rate'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = 0.81, alpha = 0.4)  +
        ggtitle(metric)
```

### Fraction of singleton reads

Shows the number of single-read families divided by the total number of reads. As suggested by [Stoler et al. 2016](https://doi.org/10.1186/s13059-016-1039-4), this metric can server as a proxy for error rate, as (uncorrected) barcode mismatches will manifest as single-read families. The lower the fraction of singletons, the better.

```{r singletons}
metric <- 'frac_singletons'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)
```

### Single family fraction

Similar to traction of singletons, this is the number of single read families, divided by the total families.

```{r}
metric <- 'single_family_fraction'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)
```

### Drop-out rate

This is the same calculation as F-EFF in the [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq):

"This shows the fraction of read bundles missing one of the two original strands beyond what would be expected under random sampling (assuming a binomial process). Good values are between 0.10-0.30, and larger values are likely due to DNA damage such as modified bases or internal nicks that prevent amplification of one of the two strands. Larger values do not impact the quality of the results, just reduce the efficiency of the protocol."

This is similar to the singleton fraction, but taking into account loss of pairs due to sampling. The optimal range is shown by the lines.

```{r drop_out_rate}
metric <- 'drop_out_rate'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.1, 0.3), alpha = 0.4)  +
        ggtitle(metric)
```

### Efficiency

Efficiency is the number of duplex bases divided by the number of sequenced bases. According the [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq), this value is maximised at \~0.07 when duplicate rates and strand drop-outs are optimal.

```{r efficiency}
metric <- 'efficiency'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = c(0.07), alpha = 0.4)  +
        ggtitle(metric)
```

### GC deviation

GC deviation is the absolute difference between `GC_BOTH` and `GC_SINGLE` calculated by the [NanoSeq Analysis pipeline](https://github.com/cancerit/NanoSeq). The lower this deviation, the better.

"GC_BOTH and GC_SINGLE: the GC content of RBs with both strands and with just one strand. The two values should be similar between them and similar to the genome average. If there are large deviations that is possibly due to biases during PCR amplification. If GC_BOTH is substantially larger than GC_SINGLE, DNA denaturation before dilution may have taken place."

```{r gc_deviation}
metric <- 'gc_deviation'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        ggtitle(metric)
```

### Duplex Coverage ratio

The mean sequence (pre-duplex) coverage divided by mean duplex coverage. Indicates the yield of how much duplex coverage we get at each sample's sequence coverage. [Abascal et al.](https://doi.org/10.1038/s41586-021-03477-4) report that their yield was approximately 30x (marked on the plot).

```{r duplex_coverage_ratio}
metric <- 'duplex_coverage_ratio'
ggplot(mm[mm$metric == metric,], aes(sample, value)) +
        geom_histogram(stat = 'identity', position = 'dodge') +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = 30, alpha = 0.4)  +
        ggtitle(metric)
```

## Family statistics

Comparison of family pair sizes between samples (these are calculated from total reads of paired AB and BA families).

```{r family_sizes, fig.width = 8, fig.height = 5}
ggplot(mm[mm$metric %like% 'family', ], aes(value, sample, colour = metric)) +
        geom_point() +
        coord_trans(x='log2') +
        scale_x_continuous(breaks=seq(0, 94, 8)) +
        theme(axis.text.x = element_text(size=5)) +
        theme_bw() +
        ggtitle('Family pair sizes')
```

The following plot shows:

-   families_gt1: number of family pairs where at least one family (AB or BA) has \> 1 reads.
-   single_families: number of single-read families.
-   paired_families: number of family pairs where both families (AB and BA) have \> 0 reads.
-   paired_and_gt1: number of family pairs where both families (AB and BA) have \> 1 reads.

```{r family_stats, fig.width = 9, fig.height = 6}
tmp <- data.table(mm)[,list(meanval = mean(value), minval = min(value), maxval= max(value)), by=c('sample', 'metric')] %>% data.frame()
tmp <- left_join(mm, tmp, by = c('sample', 'metric'))

ggplot(tmp[tmp$metric %like% 'pair|gt1|single_families', ], aes(sample, meanval, fill = metric)) +
        geom_bar(stat='identity', position='dodge') +
        geom_errorbar( aes(x = sample, ymin = minval, ymax = maxval), position = 'dodge', colour = 'grey') +
        theme_bw() +
        coord_flip() +
        scale_fill_brewer(palette = 'Dark2') +
        theme(legend.position = 'right')
```

## Compare metrics side-by-side

Compare protocols and nucleases directly, the first plot includes the outlier sample and the second removes it.

```{r metrics_comp, fig.width=14.5, fig.height=5.5}
metric_optimals <- list('duplicate_rate' = 0.81,
                        'frac_singletons' = 0,
                        'drop_out_rate' = c(0.1, 0.3),
                        'efficiency' = 0.07,
                        'gc_deviation' = 0,
                        'duplex_coverage_ratio' = 30)
                       
gg_prot <- list(geom_boxplot(outlier.shape = NA),
                geom_jitter(width = 0.1, size = 2, aes(colour = nuclease, shape = nuclease)),
                theme_bw(),
                theme(legend.position = 'bottom'))
gg_nuc <- list(geom_boxplot(outlier.shape = NA),
               geom_jitter(width = 0.1, size = 2, aes(colour = protocol, shape = protocol)),
               theme_bw(),
               theme(legend.position = 'bottom'))

mmt <- mm
mmt$replicate <- str_split(mmt$sample, 'Rep') %>% lapply(., dplyr::last) %>% unlist() %>% as.numeric()
mmt$sample <- str_split(mmt$sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()

for(metric in names(metric_optimals)) {
    # plot all samples
    threshold <- metric_optimals[metric][[1]]
    tmp <- mmt[mmt$metric %in% metric,]
    p1 <- ggplot(tmp, aes(sample, value)) +
        geom_point() +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = threshold, alpha = 0.4) +
        ggtitle(paste(metric, '(line = optimal)'))
    
    p2 <- ggplot(tmp, aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    p3 <- ggplot(tmp, aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    show(p1 + p2 + p3)
    
    # repeat with removed outlier
    tmp <- mmt[mmt$metric %in% metric & !(mmt$sample %in% 'xGEN-xGEN' & mmt$replicate == 1),]
    p1 <- ggplot(tmp, aes(sample, value)) +
        geom_point() +
        theme_bw() +
        coord_flip() +
        geom_hline(yintercept = threshold, alpha = 0.4) +
        ggtitle(paste(metric, '(line = optimal)'))
    
    p2 <- ggplot(tmp, aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    p3 <- ggplot(tmp, aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = threshold, alpha = 0.4) 
    
    show(p1 + p2 + p3)
}
```

```{r poster_plots, fig.width=8, fig.height=3, include=FALSE, eval=FALSE}
for(metric in c('efficiency', 'drop_out_rate')) {
    threshold <- metric_optimals[metric][[1]]
    tmp <- mmt[mmt$metric %in% metric,]
    
    p2 <- ggplot(tmp, aes(protocol, value)) +
        gg_prot + geom_hline(yintercept = threshold, alpha = 0.4) +
        theme(legend.position = 'left') +
        scale_color_brewer(palette = 'Dark2')
    
    p3 <- ggplot(tmp, aes(nuclease, value)) +
        gg_nuc + geom_hline(yintercept = threshold, alpha = 0.4) +
        theme(legend.position = 'right') +
        scale_color_brewer(palette = 'Set1')
    
    pdf(paste0(metric, '.pdf'), width = 8, height = 3)
    show(p2 + p3)
    dev.off()
}
```

## Facet summary plots

Facet boxplots by nuclease and protocol to show overall results.

```{r facet_summary, fig.width=9, fig.height=6}
ggplot(mm, aes(protocol, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by protocol')

ggplot(mm, aes(nuclease, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by nuclease')
```

Plots again removing the outlier xGEN rep 1.

```{r facet_summary_remove_outlier, fig.width=9, fig.height=6}
mmo <- mm[mm$sample != 'xGEN-xGENRep1',]
mmo$replicate <- str_split(mmo$sample, 'Rep') %>% lapply(., dplyr::last) %>% unlist() %>% as.numeric()
mmo$sample <- str_split(mmo$sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()

ggplot(mmo, aes(protocol, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by protocol')

ggplot(mmo, aes(nuclease, value)) + 
    geom_boxplot() +
    theme_bw() +
    facet_wrap(~metric, scales = 'free') +
    ggtitle('by nuclease')
```

Summary plot including separated by all experimental factors.

```{r plot_by_sample, fig.width=16, fig.height=10}
ggplot(mmo, aes(sample, value, colour = protocol, shape = nuclease)) + 
    geom_point() +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90)) +
    facet_wrap(~metric, scales = 'free') +
    scale_colour_brewer(palette = 'Dark2') +
    ggtitle('by protocol')
```

## Statistical test results by protocol

For each metric, take the average of each replicate and perform a two-sided, unpaired T-test between protocols.

```{r tests_by_protocol}
stats <- NULL
metric_names <- unique(mmo$metric) %>% as.character()
for(metric_name in metric_names) {
    nano <- mmo[mmo$metric == metric_name & mmo$protocol == 'NanoSeq',]
    xgen <- mmo[mmo$metric == metric_name & mmo$protocol == 'xGen',]
    nano_vals <- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals <- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest <- t.test(nano_vals, xgen_vals)
    stats <- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```

Paired T-test (pairing nucleases, removing standard xGEN).

```{r}
stats <- NULL
metric_names <- unique(mmo$metric) %>% as.character()
for(metric_name in metric_names) {
    nano <- mmo[mmo$metric == metric_name & mmo$protocol == 'NanoSeq',]
    xgen <- mmo[mmo$metric == metric_name & mmo$protocol == 'xGen' & mmo$nuclease != '0+0',]
    nano_vals <- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals <- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest <- t.test(nano_vals, xgen_vals, paired=TRUE)
    stats <- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```

Rerun tests removing outlier (xGEN rep1). The results are similar.

```{r test_by_protocol_remove_outlier}
stats <- NULL
for(metric_name in metric_names) {
    nano <- mmo[mmo$metric == metric_name & mmo$protocol == 'NanoSeq',]
    xgen <- mmo[mmo$metric == metric_name & mmo$protocol == 'xGen',]
    nano_vals <- data.table(nano)[, mean(value), by = nuclease]$V1
    xgen_vals <- data.table(xgen)[, mean(value), by = nuclease]$V1
    wtest <- t.test(nano_vals, xgen_vals)
    stats <- rbind(stats,
                   data.frame(metric = metric_name, pvalue = wtest$p.value))
}
stats$significant <- stats$pvalue < 0.05
print(stats)
```

## Two-way ANOVA analysis

We consider a two-way ANOVA, modelling the protocol, Mung Bean Unit and S1 Unit variables, as well as the interaction effect between the units and the protocol.

```{r tests_by_nuclease}
stats <- NULL
metric_names <- unique(mm$metric) %>% as.character()
for(metric_name in metric_names) {
    x <- mm[mm$metric == metric_name,]
    x$MungBeanUnit <- as.factor(x$`Mung bean unit`)
    x$S1Unit <- as.factor(x$`S1 unit`)
    x <- x[,c('MungBeanUnit', 'S1Unit', 'protocol', 'nuclease', 'value')]
    x_aov <- aov(value ~ MungBeanUnit * protocol + S1Unit * protocol, data = x) %>% summary()
    stats <- rbind(stats,
                   data.frame(metric = metric_name,
                              variable = rownames(x_aov[[1]])[1:5],
                              pvalue = x_aov[[1]][['Pr(>F)']][1:5]))
}
stats$q <- p.adjust(stats$pvalue, method = 'BH')
stats$significant <- stats$q < 0.05
print(stats)
```

We remove the outlier xGEN rep 1 and test again.

```{r tests_by_nuclease_remove_outlier}
stats <- NULL
metric_names <- unique(mmo$metric) %>% as.character()
for(metric_name in metric_names) {
    x <- mmo[mmo$metric == metric_name,]
    x$MungBeanUnit <- as.factor(x$`Mung bean unit`)
    x$S1Unit <- as.factor(x$`S1 unit`)
    x <- x[,c('MungBeanUnit', 'S1Unit', 'protocol', 'nuclease', 'value')]
    # x_aov <- aov(value ~ MungBeanUnit * protocol + S1Unit * protocol, data = x) %>% summary() %>% dplyr::first()
    x_aov <- aov(value ~ MungBeanUnit * S1Unit * protocol, data = x) %>% summary()
    stats <- rbind(stats,
                   data.frame(metric = metric_name,
                              variable = rownames(x_aov[[1]])[1:5],
                              pvalue = x_aov[[1]][['Pr(>F)']][1:5]))
}
stats$q <- p.adjust(stats$pvalue, method = 'BH')
stats$significant <- stats$q < 0.05
print(stats)
```

## Relationships between variables

```{r relationships, fig.width=10, fig.height=8}
tmp <- mmo[,c('sample', 'metric', 'value', 'protocol', 'nuclease', 'replicate')]
dm <- reshape2::dcast(mmo, sample + protocol + nuclease + replicate ~ metric)

cols <- c(brewer.pal(5, 'Greens')[2:5],
          brewer.pal(6, 'Blues')[2:6])
names(cols) <- as.factor(dm$sample) %>% levels()

ggplot(dm, aes(frac_singletons, drop_out_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Singletons vs. drop-out rate')

ggplot(dm, aes(efficiency, duplicate_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Efficiency vs. duplicate rate')

ggplot(dm, aes(efficiency, drop_out_rate, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Efficiency vs. drop-out rate')

ggplot(dm, aes(efficiency, duplex_coverage_ratio, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Efficiency vs. duplex coverage ratio')

ggplot(dm, aes(duplicate_rate, duplex_coverage_ratio, colour=sample)) +
    geom_point() +
    theme_bw() +
    scale_colour_manual(values = cols) +
    ggtitle('Duplicate rate vs. duplex coverage ratio')
```

Focus on relationship between efficiency, duplicate rate and drop-out rate.

```{r efficiency_comparison, fig.width=14, fig.height=5}
mt <- mm
mt$replicate <- str_split(mt$sample, 'Rep') %>% lapply(., dplyr::last) %>% unlist() %>% as.numeric()
mt$sample <- str_split(mt$sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()

mt <- mt[,c('sample', 'metric', 'value', 'protocol', 'nuclease', 'replicate')]
dm <- reshape2::dcast(mt, sample + protocol + nuclease + replicate ~ metric)

p1 <- ggplot(dm, aes(duplicate_rate, efficiency, colour=protocol, shape=nuclease)) +
    geom_point(size = 3) +
    theme_bw() +
    scale_colour_brewer(palette = 'Dark2') +
    ggtitle('Efficiency vs. duplicate rate')

p2 <- ggplot(dm, aes(drop_out_rate, efficiency, colour=protocol, shape=nuclease)) +
    geom_point(size = 3) +
    theme_bw() +
    scale_colour_brewer(palette = 'Dark2') +
    ggtitle('Efficiency vs. drop-out rate')
show(p1 + p2)
```

## Variant calls

Upset plot showing duplex variant calls. Variants were called in areas with at least 4x coverage with at least 2 supporting reads and a VAF of $\geq2$.

```{r variants, fig.width=12, fig.height=10}
ulist <- NULL
for(sample in sample_names) {
    ids <- var_df[var_df$sample %in% sample,]$id
    if (length(ids) > 0) {
        ulist[[gsub(pattern = '-HJK2GDSX3', replacement = '', sample)]] <- ids
    }
}

upset(fromList(ulist), order.by='freq', nsets=length(sample_names))
```

### Raw variant calls

In the xGEN-xGEN samples, we compare the unrestricted PCR sample variants (most representative of a typical NGS experiment), called using raw reads (not duplex consensus), compared with the xGEN-xGEN samples with standard-end repair that were PCR restricted.

```{r}
tmp <- rbind(var_df, var_df_raw)

ulist <- NULL
for(sample in c('xGEN-xGENRep2-HJK2GDSX3', 'xGEN-xGENRep3-HJK2GDSX3', 'xGEN-xGENRep1_default_filter', 'xGEN-xGENRep1_minimal_filter')) {
    ids <- tmp[tmp$sample %in% sample,]$id
    if (length(ids) > 0) {
        ulist[[gsub(pattern = '-HJK2GDSX3', replacement = '', sample)]] <- ids
    }
}

upset(fromList(ulist), order.by='freq', nsets=length(sample_names))
```

## Duplex coverage without requiring SSC

The pipeline was run only requiring a single read on each strand. Here we plot the difference in mean coverage. As we would expect, skipping SSC step increases duplex coverage. For some samples with disproportionately higher single-read families (NanoMB-S1), this increases duplex coverage significantly more.

```{r coverage_without_ssc, fig.width=10, fig.height=4}
ccov <- inner_join(qmap_cons_cov,
                   qmap_cons_cov_nossc,
                   by = 'Sample',
                   suffix = c('_ssc', '_nossc')) %>%
          inner_join(., qmap_cov, by = 'Sample')
ccov$sample <- str_split(ccov$Sample, 'Rep') %>% lapply(., dplyr::first) %>% unlist()
ccov$duplex_cov_ratio <- ccov$coverage / ccov$coverage_ssc
ccov$duplex_cov_ratio_noscc <- ccov$coverage / ccov$coverage_nossc
ccov <- left_join(ccov, distinct(mmo[,c('sample', 'protocol', 'nuclease')]), by = 'sample')

p1 <- ggplot(ccov, aes(coverage_ssc, coverage_nossc, colour = protocol, shape = nuclease)) +
  geom_point() +
  theme_bw() +
  xlim(0, 550) +
  ylim(0, 550) +
  xlab('with SSC') +
  ylab('without SSC') +
  geom_abline(slope = 1) +
  theme(legend.position = 'left') +
  scale_colour_brewer(palette = 'Dark2') +
  ggtitle('Mean duplex coverage')

p2 <- ggplot(ccov, aes(duplex_cov_ratio, duplex_cov_ratio_noscc, colour = protocol, shape = nuclease)) +
  geom_point() +
  theme_bw() +
  xlim(0, 100) +
  ylim(0, 100) +
  xlab('with SSC') +
  ylab('without SSC') +
  geom_abline(slope = 1) +
  theme(legend.position = 'right') +
  scale_colour_brewer(palette = 'Dark2') +
  ggtitle('Duplex coverage ratio')

p1 + p2
```

## Variant calls without SSC

Here we show the variant calls from the duplex sequences without SSC in the same Upset plot format.

```{r variants_nossc, fig.width=12, fig.height=10}

for(sample in sample_names) {
    ids <- var_df_nossc[var_df_nossc$sample %in% sample,]$id
    if (length(ids) > 0) {
        ulist[[sample]] <- ids
    }
}

upset(fromList(ulist), order.by='freq', nsets=length(sample_names))
```

## Input cells

Estimate the number of input cells using formula $d / e / c = n$ where d = mean duplex coverage, e = duplex efficiency, c = coverage per genome equivalent and n = number of cells.

```{r}
coverage_per_genome <- 10
qmap_cons_cov$Sample <- gsub('-HJK2GDSX3', '', qmap_cons_cov$Sample)
metrics <- inner_join(metrics, qmap_cons_cov, by = c('sample' = 'Sample'))
metrics$estimated_cells <- metrics$coverage / metrics$efficiency / coverage_per_genome

ggplot(metrics[!metrics$sample %in% 'xGEN-xGENRep1',], aes(sample, estimated_cells)) +
    geom_bar(stat = 'identity') + 
    theme_minimal() +
    coord_flip()
```

## Efficiency estimation

The NanoSeq paper ([Abascal et al.](https://doi.org/10.1038/s41586-021-03477-4)) provides a formula for the estimation of efficiency given the sequencing ratio:

`(ppois(q=2-0.1, lambda=seqratio/2, lower.tail=F)/(1-dpois(0, seqratio/2)))^2 / (seqratio/(1-exp(-seqratio)))`

This is the zero-truncated poisson probability of selecting >=2 reads squared (as we need two read families to form a duplex), divided by the zero-truncated mean (sequences per family). We can expect this to be an over-estimate of the true efficiency, as the reasons of drop-out beyond sampling are not accounted for.

To start, let's see how close the efficiency estimation is to the estimate. We will assume 600 cell equivalents were sequenced per E coli sample. Additionally, we estimate the sequencing ratio as follows:

`ESTIMATED_LIBRARY_SIZE / (READ_PAIRS_EXAMINED - READ_PAIR_OPTICAL_DUPLICATES)`

The above numbers can be obtained from the MarkDuplicates output. We should be able to estimate this before sequencing, but I'm using the empirical values for this experiment.

We can then estimate the raw coverage as the seqratio mutliplied by the number of cells. Here is what this looks like on this data set:

```{r estimate_raw_coverage, fig.width=11, fig.height=4.5}
# prepare efficiency stats and filter out unrestricted xGEN sample
eff <- filter(mm, metric == 'efficiency') %>% 
        select(c('sample', 'nuclease', 'protocol', 'value')) %>%
        mutate(raw_coverage = qmap_cov$coverage,
               dup_coverage = qmap_cons_cov$coverage) %>%
        rename(value = 'efficiency') %>%
        filter(sample != 'xGEN-xGENRep1')

# re-extract markdup data with library stats
md <- list.files(
        markdup_dir,
        full.names = TRUE,
        recursive = TRUE,
        pattern = 'txt') %>%
        paste('grep -E "Library|LIBRARY"', .) %>%
        lapply(., fread) %>%
        suppressMessages()
md <- md[-9] # remove xGEN-xGENRep1

# calculate sequencing ratio
eff$libsize <- lapply(md, select, ESTIMATED_LIBRARY_SIZE) %>% unlist() %>% as.numeric() 
eff$total_reads <- lapply(md, function(x){x$READ_PAIRS_EXAMINED - x$READ_PAIR_OPTICAL_DUPLICATES}) %>% as.numeric()
eff$seqratio <- eff$total_reads / eff$libsize

# estimate duplex coverage from seqratio
eff$cells <- 600
eff$est_raw_coverage <- eff$seqratio * eff$cells

p1 <- ggplot(eff, aes(raw_coverage, est_raw_coverage, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated raw coverage') +
    xlab('Raw coverage') +
    ggtitle('Estimated vs. actual\nraw coverage (600 cells)') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 10000) + ylim(0, 10000)

eff$cells <- 2000
eff$est_raw_coverage <- eff$seqratio * eff$cells
p2 <- ggplot(eff, aes(raw_coverage, est_raw_coverage, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated raw coverage') +
    xlab('Raw coverage') +
    ggtitle('Estimated vs. actual\nraw coverage (2000 cells)') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 10000) + ylim(0, 10000)

p1 + p2
```

We can see that the coverage is under-estimated, which may be due to an under-estimation of the number of input cells, but I am not entirely sure.

We should be able to use estimated raw coverage, together with estimated efficiency to calculate the estimated duplex coverage. But because our estimated raw coverage is so off, we'll just use the calculated coverage for now.

Given that the efficiency model is quite simplistic, adjusting it by the drop-out rate gives a more accurate estimation. To do this, we multiply the estimated duplex coverage by one minus the drop-out rate.

```{r estimate_efficiency, fig.width=11, fig.height=4.5}
# calculate efficiency and estimated dup coverage
eff$est_efficiency <- (ppois(q=2-0.1, lambda=eff$seqratio/2, lower.tail=F)/(1-dpois(0, eff$seqratio/2)))^2 / (eff$seqratio/(1-exp(-eff$seqratio)))
eff$est_dup_coverage <- eff$raw_coverage * eff$est_efficiency

# here we add drop-out rate to the mix
eff <- filter(mm, metric == 'drop_out_rate' & sample != 'xGEN-xGENRep1') %>%
        select(c('sample', 'value')) %>%
        rename(value = 'drop_out_rate') %>%
        left_join(eff, ., by = 'sample')

p1 <- ggplot(eff, aes(efficiency, est_efficiency, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated efficiency') +
    xlab('Efficiency') +
    ggtitle('Estimated vs. actual efficiency') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 0.15) + ylim(0, 0.15)

p2 <- ggplot(eff, aes(efficiency, est_efficiency * (1 - drop_out_rate), colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated efficiency') +
    xlab('Efficiency') +
    ggtitle('Estimated vs. actual efficiency with drop out') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 0.15) + ylim(0, 0.15)

p1 + p2
```

Here we can see the effect on the duplex coverage. We also halve the estimated duplex coverage to account for read filtering, which further reduces efficiency but is not accounted for in the calculation.

```{r estimate_coverage_with_dropout, fig.width=16.5, fig.height=4.5}
eff$est_dup_coverage_wdo <- eff$est_dup_coverage * (1 - eff$drop_out_rate)
eff$est_dup_coverage_wdo_cor <- eff$est_dup_coverage_wdo * 0.5
p1 <- ggplot(eff, aes(dup_coverage, est_dup_coverage, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated duplex coverage') +
    xlab('Duplex coverage') +
    ggtitle('Estimated vs. actual duplex coverage') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 1200) + ylim(0, 1200)

p2 <- ggplot(eff, aes(dup_coverage, est_dup_coverage_wdo, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated duplex coverage') +
    xlab('Duplex coverage') +
    ggtitle('Estimated vs. actual duplex coverage with drop-out') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 1200) + ylim(0, 1200)

p3 <- ggplot(eff, aes(dup_coverage, est_dup_coverage_wdo_cor, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated duplex coverage') +
    xlab('Duplex coverage') +
    ggtitle('Estimated vs. actual duplex\ncoverage with drop-out and correction') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 1200) + ylim(0, 1200)
p1 + p2 + p3
```

Which gives something much closer to the mark. I'm using the drop-out rates calculated from the data, however. Here's what happens if we take a mean drop-out rate (`0.448` in these data).

```{r estimate_coverage_with_mean_dropout, fig.width=5.5, fig.height=4.5}
eff$est_dup_coverage_wdo <- eff$est_dup_coverage * (1 - mean(eff$drop_out_rate))

ggplot(eff, aes(dup_coverage, est_dup_coverage_wdo, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated duplex coverage') +
    xlab('Duplex coverage') +
    ggtitle('Estimated vs. actual duplex coverage with mean drop-out') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 600) + ylim(0, 600)
```
Now we're starting to see the efficiency differences between the protocols making a difference. 

Here's the same data for NanoSeq only, using the average NanoSeq drop-out rate of 0.285.

```{r estimate_coverage_with_mean_dropout_nano, fig.width=5.5, fig.height=4.5}
eff_nano <- filter(eff, protocol == 'NanoSeq')
eff_nano$est_dup_coverage_wdo <- eff_nano$est_dup_coverage * (1 - mean(eff_nano$drop_out_rate))

ggplot(eff_nano, aes(dup_coverage, est_dup_coverage_wdo, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Estimated duplex coverage') +
    xlab('Duplex coverage') +
    ggtitle('NanoSeq estimated vs. actual duplex\ncoverage with mean drop-out') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 1000) + ylim(0, 1000)
```
There's still an over-estimate here, however, the relationship is pretty linear, which means we can adjust for this.

### Coverage efficiency

We can also calculate duplex efficiency using the mean duplex coverage over the mean raw coverage. This provides more of a picture of the real duplex output efficiency, as it also accounts for duplex reads that have been filtered out downstream of the family counts (the regular efficiency metric is based on the family counts only). We can see that the coverage efficiency is lower than the efficiency calculated via the family counts. This provides another piece of evidence for why we over-estimate duplex coverage. On average, the real (coverage) efficiency is only ~62\% of the pre-filtered family efficiency. We can likely rescue some of these reads by considering single family pairs, or reducing filtering stringency.

```{r adjusted_efficiency, fig.width=5.5, fig.height=4.5}
eff$real_efficiency <- eff$dup_coverage / eff$raw_coverage

ggplot(eff, aes(efficiency, real_efficiency, colour=protocol, shape=nuclease)) +
    geom_point() +
    theme_minimal() +
    geom_abline(slope = 1) +
    ylab('Coverage efficiency') +
    xlab('Family efficiency') +
    ggtitle('Coverage vs. family efficiency ') +
    scale_colour_brewer(palette = 'Dark2') +
    xlim(0, 0.15) + ylim(0, 0.15)
```

