---
title: "E. coli in silico mixture results"
output:
  workflowr::wflow_html:
      code_folding: hide
---

In this experiment, the duplex reads from NanoSeq MB2 rep 1 from E coli K12 was mixed in pairs (in silico) at four depths (0.1x, 1x, 5x and 10x) with the duplex reads from NanoSeq MB2 (MGI) from E coli BL21 at four depths (99.9x, 99x, 95x and 90x). Variants called required a coverage of at least 4 and at least 1 supporting variant read. There are >33k SNP and INDEL differences between the E coli species. Given the mixture and total depth we can calculate the number of SNPs that we expect to find.

```{r setup, include=FALSE}
DOCNAME = "Ecoli in silico mixture results"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      cache.lazy     = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 7,
                      fig.height     = 7,
                      dev            = c("png"),
                      message        = FALSE,
                      warning        = FALSE)
```


```{r libraries, cache=FALSE, message=FALSE}
library(ggplot2)
library(data.table)
library(dplyr)
library(here)
library(tibble)
library(stringr)
library(Rsamtools)
library(GenomicRanges)
library(seqinr)
library(readxl)
library(patchwork)
library(RColorBrewer)
library(UpSetR)
library(vcfR)
```


```{r source}
source(here('code/load_data.R'))

calculate_vafs <- function(var_df, freq_filter = 0.3) {
    tmp <- var_df$Sample1 %>% str_split(':') %>% lapply(., function(x){as.numeric(x[5:6])}) 
    var_df$RD <- lapply(tmp, head, 1) %>% unlist()
    var_df$AD <- lapply(tmp, tail, 1) %>% unlist()
    
    var_df$VAF <- var_df$AD / (var_df$AD + var_df$RD)
    var_df <- var_df[var_df$VAF < freq_filter,]
    
    return(var_df)
}
```


```{r paths}
variant_dir <- here('data/mixtures/min_1_read')
```


```{r load_data}
var_df <- load_variants(variant_dir, c('NanoMB2-0.001', 'NanoMB2-0.01', 'NanoMB2-0.05', 'NanoMB2-0.10')) %>% calculate_vafs()
var_df$sample <- factor(var_df$sample, levels = c('NanoMB2-0.001', 'NanoMB2-0.01', 'NanoMB2-0.05', 'NanoMB2-0.10'))
```


## VAF mixture distributions

NanoSeq MB2 mixes:

- 0.1:99.9x KL12:BL21 (0.001 mix)
- 1:99x K12:BL21 (0.01 mix)
- 5:95x K12:BL21 (0.05 mix)
- 10:90x K12:BL21 (0.10 mix)


```{r plot_vafs}
# NOTE: variant number obtained from nucdiff difference between two genomes
# cat G000204/workflow/jafarJ_201021_duplex/duplex-pipe/ref/nucdiff/ecoli_BL21_vs_ATCCecoli_BL21_vs_ATCC_1.snps | cut -f 1-2 | grep -v "\." | cut -f 1 | sort | uniq | wc -l

vaf_sm <- data.table(var_df)[, list(VAF_mean = mean(VAF), nvars = length(POS)), by=sample] %>%
            mutate(VAF_mix = as.character(sample) %>% strsplit('-') %>% lapply(last) %>% unlist() %>% as.numeric())
vaf_sm$expected <- (1 - (1 - vaf_sm$VAF_mix) ^ 100) * 33655
print(vaf_sm)

ggplot(var_df, aes(VAF)) +
    geom_histogram(bins = 50) +
    theme_minimal() +
    facet_wrap(~sample) +
    scale_x_continuous(breaks = seq(0, 1, 0.1), limits = c(0, 1))
```

