---
title: "Rare mutation detection model"
output:
  workflowr::wflow_html:
      code_folding: hide
---

```{r knitr, include = FALSE}
DOCNAME = "Simulation_Benchmarking"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 9,
                      fig.height     = 5,
                      dev            = c("png"),
                      message        = FALSE,
                      warning        = FALSE)
```

```{r libraries, cache = FALSE}
library(ggplot2)
```

# Model

Q: how many cells (or "cell equivalents") do we need to sequence to detect variants down to a VAF of X%

## Assumptions

* We consider a diploid genome without SCNAs
* The mutation(s) are heterozygous
* We're not considering DNA extraction efficiency or ligation efficiency

## Probability of sequencing a mutant cell

Let $p$ be the probability of selecting a mutant cell from a pool of $n$ cells.

$p = v / 2$

Where:

* $v$ = target VAF

We assume the probability of selecting a mutant cell is binomially distributed. We want to know the probability of selecting at least one mutant:

$P(Bin(n, p)) > 0)$ = 0.95

This is equivalent to:

$P(Bin(n, p)) = 0)$ = 0.05

Where $n$ is the number of sequenced cells. We note that the number of mutant cells will, on average, will be $2nv$. 

Let's consider how many input cells ($n$) are required to select at least one mutant cell at 1% VAF (95% confidence).

We'll consider 100 cell increments from $n = \{100, 200..2000\}$. Using these values, we can plot the probability selecting of sequencing at least one cell.

```{r cells_plot}
v = 0.01
n = seq(100, 2000, 100)

vafs <- data.frame(vaf=v,
                   p=pbinom(0, n, v / 2),
                   input_cells=n,
                   mutant_cells=(n * v * 2))

ggplot(vafs, aes(input_cells, p)) +
    geom_point() +
    theme_bw() +
    geom_hline(yintercept=0.05, alpha=0.4)
```

We can also plot this as mutant cells instead of VAF:

```{r mutant_cells_plot}
ggplot(vafs, aes(mutant_cells, p)) +
    geom_point() +
    theme_bw() +
    geom_hline(yintercept=0.05, alpha=0.4)
```

This shows that for a VAF target of 1%, we should sequence at least 600 cells (12 are mutants).

```{r cells_at_0.05}
deviation <- abs(0.05 - vafs$p)
print(vafs[which(deviation == min(deviation)),])
```

## Varying the VAF

If we select 1000 input cells, what's the lowest VAF we can sequence down to, where $v = {0.0002, 0.0004..0.01}$.

```{r varying_vaf}
v = seq(0.0002, 0.01, 0.0002)
n = 1000

vafs <- data.frame(vaf=v,
                   p=pbinom(0, n, v / 2),
                   input_cells=n,
                   mutant_cells=(n * v * 2))

ggplot(vafs, aes(vaf, p)) +
    geom_point() +
    theme_bw() +
    geom_hline(yintercept=0.05, alpha=0.4)
```

We can sequence down to approx 0.6% VAF. 

```{r vaf_at_1k_cells}
deviation <- abs(0.05 - vafs$p)
print(vafs[which(deviation == min(deviation)),])
```

We can look at the relationship between input cells and allele frequency: 

```{r cells_vs_vaf_plot, fig.height=6.5, fig.width=9}
cells_vs_vaf = NULL
n = seq(100, 10000, 100)
V = seq(0.001, 0.01, 0.001)
for (v in V) {
    toadd <- data.frame(
        vaf=as.factor(v),
        p=pbinom(0, n, v / 2),
        total_cells=n
    )
    cells_vs_vaf <- rbind(cells_vs_vaf, toadd)
}

ggplot(cells_vs_vaf, aes(total_cells, p, colour=vaf)) +
    geom_line() +
    theme_bw() +
    theme(legend.position = 'bottom') +
    geom_hline(yintercept=0.05, alpha=0.4)
```

We can define an equation based on the binomial probability calculation, of obtaining the number of cells to sequence, to be 95% confident of sequencing the mutation based on the target VAF:

$(1 â€“ v / 2) ^ n = 0.05$

## Coverage

The above thought experiment won't tell us the probability of sequencing the variant, as sequencing involves a whole host of other factors. Coverage is one of the most important. Given that approximately 30x duplex coverage is required to yield 1x of high quality duplex coverage, we can look at the relationship between coverage and VAF (need the variant to be at least 1x to detect).

Here we can see that at VAF = 0.01 (line), we need to sequence to approx 3000x, and this gets exponentially deeper as we go down in target VAF.

```{r coverage}
v = seq(0.001, 0.1, 0.001)
ccov <- data.frame(d = 30 / v,
                   v = v)
ggplot(ccov, aes(d, v)) +
    geom_line() +
    theme_bw() +
    geom_hline(yintercept=0.01, alpha=0.4)
```
